Index: firefox-104.0.2/browser/base/content/browser-sidebar.js
===================================================================
--- firefox-104.0.2.orig/browser/base/content/browser-sidebar.js
+++ firefox-104.0.2/browser/base/content/browser-sidebar.js
@@ -41,6 +41,8 @@ var SidebarUI = {
           triggerButtonId: "appMenuViewHistorySidebar",
         }),
       ],
+/* MagIns - Synced Tabs depends on device Push, which is not yet implemented. Thus, remove
+            the Synced Tabs entry from the list of sidebars
       [
         "viewTabsSidebar",
         makeSidebar({
@@ -48,7 +50,7 @@ var SidebarUI = {
           url: "chrome://browser/content/syncedtabs/sidebar.xhtml",
           menuId: "menu_tabsSidebar",
         }),
-      ],
+      ],*/
     ]));
   },
 
Index: firefox-104.0.2/browser/base/content/browser-sync.js
===================================================================
--- firefox-104.0.2.orig/browser/base/content/browser-sync.js
+++ firefox-104.0.2/browser/base/content/browser-sync.js
@@ -658,10 +658,10 @@ var gSync = {
       title = PluralForm.get(tabCount, string).replace("#1", tabCount);
     }
 
-    PanelMultiView.getViewNode(
+/*    PanelMultiView.getViewNode(
       document,
       "PanelUI-fxa-menu-sendtab-button"
-    ).setAttribute("label", title);
+    ).setAttribute("label", title);*/
   },
 
   showSendToDeviceView(anchor) {
@@ -812,12 +812,14 @@ var gSync = {
 
     this.enableSendTabIfValidTab();
 
+/* MagIns - With the other removed elements, we must comment this out for the FxA panel
+            to function properly.
     if (!this.getSendTabTargets().length) {
       PanelMultiView.getViewNode(
         document,
         "PanelUI-fxa-menu-sendtab-button"
       ).hidden = true;
-    }
+    }*/
 
     if (anchor.getAttribute("open") == "true") {
       PanelUI.hide();
@@ -846,10 +848,12 @@ var gSync = {
       "fxa-menu-header-description"
     );
 
+/* MagIns - Device Push is not yet implemented. Thus, we remove the "Connect Another Device"
+            option in the FxAccounts panel
     const cadButtonEl = PanelMultiView.getViewNode(
       document,
       "PanelUI-fxa-menu-connect-device-button"
-    );
+    );*/
 
     const syncSetupButtonEl = PanelMultiView.getViewNode(
       document,
@@ -873,7 +877,8 @@ var gSync = {
       "fxa-menu-turn-on-sync-default"
     );
 
-    cadButtonEl.setAttribute("disabled", true);
+/* MagIns - Remove reference to cadButtonEl
+    cadButtonEl.setAttribute("disabled", true);*/
     syncNowButtonEl.hidden = true;
     fxaMenuAccountButtonEl.classList.remove("subviewbutton-nav");
     fxaMenuAccountButtonEl.removeAttribute("closemenu");
@@ -916,7 +921,7 @@ var gSync = {
         mainWindowEl.style.removeProperty("--avatar-image-url");
       }
 
-      cadButtonEl.removeAttribute("disabled");
+//      cadButtonEl.removeAttribute("disabled");
 
       if (state.syncEnabled) {
         syncNowButtonEl.removeAttribute("hidden");
@@ -941,6 +946,8 @@ var gSync = {
   },
 
   enableSendTabIfValidTab() {
+/* MagIns - This function references the "Send Tab" option in the FxAccounts panel. Since
+            Synced Tabs are currently disabled, we comment this function for compatibility.
     // All tabs selected must be sendable for the Send Tab button to be enabled
     // on the FxA menu.
     let canSendAllURIs = gBrowser.selectedTabs.every(t =>
@@ -950,7 +957,7 @@ var gSync = {
     PanelMultiView.getViewNode(
       document,
       "PanelUI-fxa-menu-sendtab-button"
-    ).hidden = !canSendAllURIs;
+    ).hidden = !canSendAllURIs;*/
   },
 
   emitFxaToolbarTelemetry(type, panel) {
@@ -1545,8 +1552,9 @@ var gSync = {
     const enabled = !this.sendTabConfiguredAndLoading && hasASendableURI;
     const hideItems = this.shouldHideSendContextMenuItems(enabled);
 
-    let sendTabsToDevice = document.getElementById("context_sendTabToDevice");
-    sendTabsToDevice.disabled = !enabled;
+/* MagIns - Remove instance of context_sendTabToDevice
+      let sendTabsToDevice = document.getElementById("context_sendTabToDevice");
+    sendTabsToDevice.disabled = !enabled;*/
 
     if (hideItems || !hasASendableURI) {
       sendTabsToDevice.hidden = true;
Index: firefox-104.0.2/browser/base/content/browser-menubar.inc
===================================================================
--- firefox-104.0.2.orig/browser/base/content/browser-menubar.inc
+++ firefox-104.0.2/browser/base/content/browser-menubar.inc
@@ -149,10 +149,11 @@
                               type="checkbox"
                               key="key_gotoHistory"
                               oncommand="SidebarUI.toggle('viewHistorySidebar');" data-l10n-id="menu-view-history-button"/>
+<!-- MagIns - Remove Synced Tabs from the View -> Sidebar menu
                     <menuitem id="menu_tabsSidebar"
                               type="checkbox"
                               class="sync-ui-item"
-                              oncommand="SidebarUI.toggle('viewTabsSidebar');" data-l10n-id="menu-view-synced-tabs-sidebar"/>
+                              oncommand="SidebarUI.toggle('viewTabsSidebar');" data-l10n-id="menu-view-synced-tabs-sidebar"/>-->
                   </menupopup>
                 </menu>
                 <menuseparator/>
Index: firefox-104.0.2/browser/base/content/browser-context.inc
===================================================================
--- firefox-104.0.2.orig/browser/base/content/browser-context.inc
+++ firefox-104.0.2/browser/base/content/browser-context.inc
@@ -111,13 +111,14 @@
       <menuitem id="context-copylink"
                 data-l10n-id="main-context-menu-copy-link-simple"
                 oncommand="gContextMenu.copyLink();"/>
+<!-- MagIns - Remove "Send Link to Device" option from context menu
       <menu id="context-sendlinktodevice"
             class="sync-ui-item"
             data-l10n-id="main-context-menu-link-send-to-device"
             hidden="true">
             <menupopup id="context-sendlinktodevice-popup"
                   onpopupshowing="gSync.populateSendTabToDevicesMenu(event.target, gContextMenu.linkURL, gContextMenu.linkTextStr);"/>
-      </menu>
+      </menu>-->
       <menuseparator id="context-sep-sendlinktodevice" class="sync-ui-item"
                 hidden="true"/>
       <menuseparator id="context-sep-copylink"/>
@@ -261,13 +262,14 @@
       <menuitem id="context-pocket"
                 data-l10n-id="main-context-menu-save-to-pocket"
                 oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
+<!-- MagIns - Remove "Send Page to Device" option from context menu
       <menu id="context-sendpagetodevice"
             class="sync-ui-item"
             data-l10n-id="main-context-menu-send-to-device"
             hidden="true">
         <menupopup id="context-sendpagetodevice-popup"
                    onpopupshowing="(() => { gSync.populateSendTabToDevicesMenu(event.target, gBrowser.currentURI.spec, gBrowser.contentTitle); })()"/>
-      </menu>
+      </menu>-->
       <menu id="fill-login" hidden="true">
         <menupopup id="fill-login-popup" />
       </menu>
Index: firefox-104.0.2/browser/components/customizableui/CustomizableWidgets.jsm
===================================================================
--- firefox-104.0.2.orig/browser/components/customizableui/CustomizableWidgets.jsm
+++ firefox-104.0.2/browser/components/customizableui/CustomizableWidgets.jsm
@@ -477,6 +477,7 @@ const CustomizableWidgets = [
   },
 ];
 
+/* MagIns - Remove the Synced Tabs item from the Customize page
 if (Services.prefs.getBoolPref("identity.fxaccounts.enabled")) {
   CustomizableWidgets.push({
     id: "sync-button",
@@ -549,7 +550,7 @@ if (!lazy.screenshotsDisabled) {
       }
     },
   });
-}
+}*/
 
 let preferencesButton = {
   id: "preferences-button",
Index: firefox-104.0.2/browser/components/preferences/sync.inc.xhtml
===================================================================
--- firefox-104.0.2.orig/browser/components/preferences/sync.inc.xhtml
+++ firefox-104.0.2/browser/components/preferences/sync.inc.xhtml
@@ -196,10 +196,10 @@
                     <image class="sync-engine-image sync-engine-history"/>
                     <label data-l10n-id="sync-currently-syncing-history"/>
                   </html:div>
-                  <html:div engine_preference="services.sync.engine.tabs">
+<!--                  <html:div engine_preference="services.sync.engine.tabs">
                     <image class="sync-engine-image sync-engine-tabs"/>
                     <label data-l10n-id="sync-currently-syncing-tabs"/>
-                  </html:div>
+                  </html:div>-->
                   <html:div engine_preference="services.sync.engine.passwords">
                     <image class="sync-engine-image sync-engine-passwords"/>
                     <label data-l10n-id="sync-currently-syncing-logins-passwords"/>
@@ -233,12 +233,13 @@
         </groupbox>
       </vbox>
     </hbox>
+<!-- MagIns - Remove the Connect Another Device link from the Sync preferences page
     <vbox align="start">
       <label id="connect-another-device"
              is="text-link"
              class="fxaMobilePromo"
              data-l10n-id="sync-connect-another-device"/>
-    </vbox>
+    </vbox>-->
   </vbox>
 </deck>
 </html:template>
Index: firefox-104.0.2/browser/components/preferences/dialogs/syncChooseWhatToSync.js
===================================================================
--- firefox-104.0.2.orig/browser/components/preferences/dialogs/syncChooseWhatToSync.js
+++ firefox-104.0.2/browser/components/preferences/dialogs/syncChooseWhatToSync.js
@@ -8,7 +8,7 @@ Preferences.addAll([
   { id: "services.sync.engine.addons", type: "bool" },
   { id: "services.sync.engine.bookmarks", type: "bool" },
   { id: "services.sync.engine.history", type: "bool" },
-  { id: "services.sync.engine.tabs", type: "bool" },
+//  { id: "services.sync.engine.tabs", type: "bool" },
   { id: "services.sync.engine.prefs", type: "bool" },
   { id: "services.sync.engine.passwords", type: "bool" },
   { id: "services.sync.engine.addresses", type: "bool" },
Index: firefox-104.0.2/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
===================================================================
--- firefox-104.0.2.orig/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
+++ firefox-104.0.2/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
@@ -34,10 +34,11 @@
       <checkbox data-l10n-id="sync-engine-history"
                 preference="services.sync.engine.history"/>
     </html:div>
+<!-- MagIns - Remove the "Open Tabs" (Synced Tabs) option from Sync preferences
     <html:div class="sync-engine-tabs">
       <checkbox data-l10n-id="sync-engine-tabs"
                 preference="services.sync.engine.tabs"/>
-    </html:div>
+    </html:div>-->
     <html:div class="sync-engine-passwords">
       <checkbox data-l10n-id="sync-engine-logins-passwords"
                 preference="services.sync.engine.passwords"/>
Index: firefox-104.0.2/modules/libpref/init/all.js
===================================================================
--- firefox-104.0.2.orig/modules/libpref/init/all.js
+++ firefox-104.0.2/modules/libpref/init/all.js
@@ -4232,7 +4232,9 @@ pref("services.common.log.logger.tokense
   pref("services.sync.engine.history", true);
   pref("services.sync.engine.passwords", true);
   pref("services.sync.engine.prefs", true);
-  pref("services.sync.engine.tabs", true);
+/* MagIns - Disable tab syncing
+  pref("services.sync.engine.tabs", true);*/
+  pref("services.sync.engine.tabs", false);
   pref("services.sync.engine.tabs.filteredSchemes", "about|resource|chrome|file|blob|moz-extension|data");
 
   // The addresses and CC engines might not actually be available at all.
Index: firefox-104.0.2/browser/components/preferences/sync.js
===================================================================
--- firefox-104.0.2.orig/browser/components/preferences/sync.js
+++ firefox-104.0.2/browser/components/preferences/sync.js
@@ -106,11 +106,11 @@ var gSyncPane = {
 
     FxAccounts.config
       .promiseConnectDeviceURI(this._getEntryPoint())
-      .then(connectURI => {
+/*      .then(connectURI => {
         document
           .getElementById("connect-another-device")
           .setAttribute("href", connectURI);
-      });
+      })*/;
     // Links for mobile devices.
     for (let platform of ["android", "ios"]) {
       let url =
Index: firefox-104.0.2/browser/base/content/appmenu-viewcache.inc.xhtml
===================================================================
--- firefox-104.0.2.orig/browser/base/content/appmenu-viewcache.inc.xhtml
+++ firefox-104.0.2/browser/base/content/appmenu-viewcache.inc.xhtml
@@ -577,15 +577,17 @@
                       data-l10n-id="appmenu-fxa-setup-sync"
                       oncommand="gSync.openPrefsFromFxaMenu('sync_settings', this);"/>
       <!-- The `Connect Another Device` button is disabled by default until the user logs into Sync. -->
+<!-- MagIns - Remove the Connect Another Device button from the FxAccounts Panel
       <toolbarbutton id="PanelUI-fxa-menu-connect-device-button"
                       class="subviewbutton"
                       data-l10n-id="fxa-menu-connect-another-device"
                       disabled="true"
-                      oncommand="gSync.openConnectAnotherDeviceFromFxaMenu(this);"/>
+                      oncommand="gSync.openConnectAnotherDeviceFromFxaMenu(this);"/>-->
+<!-- MagIns - Remove the Send Tab to Device button from the FxAccounts Panel
       <toolbarbutton id="PanelUI-fxa-menu-sendtab-button"
                       class="subviewbutton subviewbutton-nav"
                       closemenu="none"
-                      oncommand="gSync.showSendToDeviceViewFromFxaMenu(this);"/>
+                      oncommand="gSync.showSendToDeviceViewFromFxaMenu(this);"/>-->
       <toolbarbutton id="PanelUI-fxa-menu-sync-prefs-button"
                       class="subviewbutton"
                       data-l10n-id="fxa-menu-sync-settings"
Index: firefox-104.0.2/browser/base/content/main-popupset.inc.xhtml
===================================================================
--- firefox-104.0.2.orig/browser/base/content/main-popupset.inc.xhtml
+++ firefox-104.0.2/browser/base/content/main-popupset.inc.xhtml
@@ -57,13 +57,15 @@
                   oncommand="gBrowser.replaceTabsWithWindow(TabContextMenu.contextTab);"/>
       </menupopup>
     </menu>
+<!-- MagIns - Synced Tabs depends on device Push, which is not yet implemented. Thus, remove
+              the "Send Tab to Device" option from the tab context menu
     <menu id="context_sendTabToDevice"
           class="sync-ui-item"
           data-lazy-l10n-id="tab-context-send-tabs-to-device"
           data-l10n-args='{"tabCount": 1}'>
       <menupopup id="context_sendTabToDevicePopupMenu"
                  onpopupshowing="gSync.populateSendTabToDevicesMenu(event.target, TabContextMenu.contextTab.linkedBrowser.currentURI.spec, TabContextMenu.contextTab.linkedBrowser.contentTitle, TabContextMenu.contextTab.multiselected);"/>
-    </menu>
+    </menu>-->
     <menu id="context_reopenInContainer"
           data-lazy-l10n-id="tab-context-open-in-new-container-tab"
           hidden="true">
@@ -302,10 +304,11 @@
                    class="subviewbutton"
                    key="key_gotoHistory"
                    oncommand="SidebarUI.show('viewHistorySidebar');"/>
+<!-- MagIns - Remove Synced Tabs from the sidebar drop-down menu
     <toolbarbutton id="sidebar-switcher-tabs"
                    data-l10n-id="sidebar-menu-synced-tabs"
                    class="subviewbutton sync-ui-item"
-                   oncommand="SidebarUI.show('viewTabsSidebar');"/>
+                   oncommand="SidebarUI.show('viewTabsSidebar');"/>-->
     <toolbarseparator/>
     <!-- Extension toolbarbuttons go here. -->
     <toolbarseparator id="sidebar-extensions-separator"/>
@@ -539,7 +542,7 @@
   <tooltip id="dynamic-shortcut-tooltip"
            onpopupshowing="UpdateDynamicShortcutTooltipText(this);"/>
 
-  <menupopup id="SyncedTabsSidebarContext">
+<!--  <menupopup id="SyncedTabsSidebarContext">
     <menuitem data-lazy-l10n-id="synced-tabs-context-open"
               id="syncedTabsOpenSelected" where="current"/>
     <menuitem data-lazy-l10n-id="synced-tabs-context-open-in-tab"
@@ -587,7 +590,7 @@
     <menuseparator/>
     <menuitem data-lazy-l10n-id="synced-tabs-context-sync-now"
               id="syncedTabsRefreshFilter"/>
-  </menupopup>
+  </menupopup>-->
 
   <hbox id="statuspanel" inactive="true">
     <hbox id="statuspanel-inner">
