Index: firefox-84.0.2/browser/base/content/browser-sidebar.js
===================================================================
--- firefox-84.0.2.orig/browser/base/content/browser-sidebar.js
+++ firefox-84.0.2/browser/base/content/browser-sidebar.js
@@ -34,6 +34,8 @@ var SidebarUI = {
           triggerButtonId: "appMenuViewHistorySidebar",
         },
       ],
+/* MagIns - Synced Tabs depends on device Push, which is not yet implemented. Thus, remove
+            the Synced Tabs entry from the list of sidebars
       [
         "viewTabsSidebar",
         {
@@ -45,7 +47,7 @@ var SidebarUI = {
           buttonId: "sidebar-switcher-tabs",
           triggerButtonId: "PanelUI-remotetabs-view-sidebar",
         },
-      ],
+      ],*/
     ]));
   },
 
Index: firefox-84.0.2/browser/base/content/browser-sync.js
===================================================================
--- firefox-84.0.2.orig/browser/base/content/browser-sync.js
+++ firefox-84.0.2/browser/base/content/browser-sync.js
@@ -301,10 +301,10 @@ var gSync = {
       title = PluralForm.get(tabCount, string).replace("#1", tabCount);
     }
 
-    PanelMultiView.getViewNode(
+/*    PanelMultiView.getViewNode(
       document,
       "PanelUI-fxa-menu-sendtab-button"
-    ).setAttribute("label", title);
+    ).setAttribute("label", title);*/
   },
 
   showSendToDeviceView(anchor) {
@@ -462,10 +462,12 @@ var gSync = {
       "fxa-menu-header-description"
     );
 
+/* MagIns - Device Push is not yet implemented. Thus, we remove the "Connect Another Device"
+            option in the FxAccounts panel and comment out references to cadButtonEl
     const cadButtonEl = PanelMultiView.getViewNode(
       document,
       "PanelUI-fxa-menu-connect-device-button"
-    );
+    );*/
 
     const syncSetupButtonEl = PanelMultiView.getViewNode(
       document,
@@ -501,7 +503,7 @@ var gSync = {
     let panelTitle = this.fxaStrings.GetStringFromName("account.title");
 
     fxaMenuPanel.removeAttribute("title");
-    cadButtonEl.setAttribute("disabled", true);
+//    cadButtonEl.setAttribute("disabled", true);
     syncNowButtonEl.setAttribute("hidden", true);
     fxaMenuAccountButtonEl.classList.remove("subviewbutton-nav");
     fxaMenuAccountButtonEl.removeAttribute("closemenu");
@@ -542,7 +544,7 @@ var gSync = {
         mainWindowEl.style.removeProperty("--avatar-image-url");
       }
 
-      cadButtonEl.removeAttribute("disabled");
+//      cadButtonEl.removeAttribute("disabled");
 
       if (state.syncEnabled) {
         syncNowButtonEl.removeAttribute("hidden");
@@ -570,6 +572,8 @@ var gSync = {
   },
 
   enableSendTabIfValidTab() {
+/* MagIns - This function references the "Send Tab" option in the FxAccounts panel. Since
+            Synced Tabs are currently disabled, we comment this function for compatibility.
     // All tabs selected must be sendable for the Send Tab button to be enabled
     // on the FxA menu.
     let canSendAllURIs = gBrowser.selectedTabs.every(t =>
@@ -586,7 +590,7 @@ var gSync = {
         document,
         "PanelUI-fxa-menu-sendtab-button"
       ).setAttribute("disabled", true);
-    }
+    }*/
   },
 
   emitFxaToolbarTelemetry(type, panel) {
@@ -1212,8 +1216,8 @@ var gSync = {
     }
     const enabled = !this.sendTabConfiguredAndLoading && hasASendableURI;
 
-    let sendTabsToDevice = document.getElementById("context_sendTabToDevice");
-    sendTabsToDevice.disabled = !enabled;
+/*    let sendTabsToDevice = document.getElementById("context_sendTabToDevice");
+    sendTabsToDevice.disabled = !enabled;*/
 
     let tabCount = aTargetTab.multiselected
       ? gBrowser.multiSelectedTabsCount
Index: firefox-84.0.2/browser/base/content/browser.xhtml
===================================================================
--- firefox-84.0.2.orig/browser/base/content/browser.xhtml
+++ firefox-84.0.2/browser/base/content/browser.xhtml
@@ -182,11 +182,13 @@
                     oncommand="gBrowser.replaceTabsWithWindow(TabContextMenu.contextTab);"/>
         </menupopup>
       </menu>
+<!-- MagIns - Synced Tabs depends on device Push, which is not yet implemented. Thus, remove
+              the "Send Tab to Device" option from the tab context menu
       <menu id="context_sendTabToDevice"
             class="sync-ui-item">
         <menupopup id="context_sendTabToDevicePopupMenu"
                    onpopupshowing="gSync.populateSendTabToDevicesMenu(event.target, TabContextMenu.contextTab.linkedBrowser.currentURI.spec, TabContextMenu.contextTab.linkedBrowser.contentTitle, TabContextMenu.contextTab.multiselected);"/>
-      </menu>
+      </menu>-->
       <menuseparator/>
       <menu id="context_closeTabOptions"
             data-lazy-l10n-id="tab-context-close-multiple-tabs">
@@ -410,11 +412,12 @@
                      class="subviewbutton subviewbutton-iconic"
                      key="key_gotoHistory"
                      oncommand="SidebarUI.show('viewHistorySidebar');"/>
+<!-- MagIns - Remove Synced Tabs from the sidebar drop-down menu
       <toolbarbutton id="sidebar-switcher-tabs"
                      type="checkbox"
                      data-l10n-id="sidebar-menu-synced-tabs"
                      class="subviewbutton subviewbutton-iconic sync-ui-item"
-                     oncommand="SidebarUI.show('viewTabsSidebar');"/>
+                     oncommand="SidebarUI.show('viewTabsSidebar');"/>-->
       <toolbarseparator/>
       <!-- Extension toolbarbuttons go here. -->
       <toolbarseparator id="sidebar-extensions-separator"/>
@@ -646,7 +649,7 @@
     <tooltip id="dynamic-shortcut-tooltip"
              onpopupshowing="UpdateDynamicShortcutTooltipText(this);"/>
 
-    <menupopup id="SyncedTabsSidebarContext">
+<!--    <menupopup id="SyncedTabsSidebarContext">
       <menuitem data-lazy-l10n-id="synced-tabs-context-open"
                 id="syncedTabsOpenSelected" where="current"/>
       <menuitem data-lazy-l10n-id="synced-tabs-context-open-in-new-tab"
@@ -688,7 +691,7 @@
       <menuseparator/>
       <menuitem data-lazy-l10n-id="synced-tabs-context-sync-now"
                 id="syncedTabsRefreshFilter"/>
-    </menupopup>
+    </menupopup>-->
 
     <hbox id="statuspanel" inactive="true">
       <hbox id="statuspanel-inner">
@@ -1355,18 +1358,20 @@
           </vbox>
         </toolbarbutton>
         <toolbarseparator/>
+<!-- MagIns - Remove the Send Tab to Device button from the FxAccounts Panel
         <toolbarbutton id="PanelUI-fxa-menu-sendtab-button"
                         class="subviewbutton subviewbutton-iconic subviewbutton-nav"
                         closemenu="none"
                         oncommand="gSync.showSendToDeviceViewFromFxaMenu(this);"/>
-        <toolbarseparator/>
+        <toolbarseparator/>-->
         <!-- The `Connect Another Device` button is disabled by default until the user logs into Sync. -->
+<!-- MagIns - Remove the Connect Another Device button from the FxAccounts Panel
         <toolbarbutton id="PanelUI-fxa-menu-connect-device-button"
                         class="subviewbutton subviewbutton-iconic"
                         label="&fxa.menu.connectAnotherDevice2.label;"
                         disabled="true"
                         oncommand="gSync.openConnectAnotherDeviceFromFxaMenu(this);"/>
-        <toolbarseparator/>
+        <toolbarseparator/>-->
         <!-- The `Sync Now` button is hidden by default until the user logs into Sync. -->
         <toolbarbutton id="PanelUI-fxa-menu-syncnow-button"
                         data-l10n-id="fxa-toolbar-sync-now"
@@ -1380,11 +1385,12 @@
                         class="subviewbutton subviewbutton-iconic"
                         label="&fxa.menu.setupSync.label;"
                         oncommand="gSync.openPrefsFromFxaMenu('sync_settings', this);"/>
+<!-- MagIns - Remove the Synced Tabs button from the FxAccounts Panel
         <toolbarbutton id="PanelUI-fxa-menu-remotetabs-button"
                         class="subviewbutton subviewbutton-iconic subviewbutton-nav"
                         label="&appMenuRemoteTabs.label;"
                         closemenu="none"
-                        oncommand="gSync.showRemoteTabsFromFxaMenu(this);"/>
+                        oncommand="gSync.showRemoteTabsFromFxaMenu(this);"/>-->
         <toolbarbutton id="PanelUI-fxa-menu-sync-prefs-button"
                         class="subviewbutton subviewbutton-iconic"
                         label="&fxa.menu.syncSettings2.label;"
@@ -1479,11 +1485,12 @@
                        label="&libraryDownloads.label;"
                        closemenu="none"
                        oncommand="DownloadsSubview.show(this);"/>
+<!-- MagIns - Remove the Synced Tabs option from the Library menu
         <toolbarbutton id="appMenu-library-remotetabs-button"
                        class="subviewbutton subviewbutton-iconic subviewbutton-nav sync-ui-item"
                        label="&appMenuRemoteTabs.label;"
                        closemenu="none"
-                       oncommand="PanelUI.showSubView('PanelUI-remotetabs', this)"/>
+                       oncommand="PanelUI.showSubView('PanelUI-remotetabs', this)"/>-->
         <toolbarseparator hidden="true"/>
         <label value="&appMenuRecentHighlights.label;"
                hidden="true"
Index: firefox-84.0.2/browser/base/content/browser-menubar.inc
===================================================================
--- firefox-84.0.2.orig/browser/base/content/browser-menubar.inc
+++ firefox-84.0.2/browser/base/content/browser-menubar.inc
@@ -161,10 +161,11 @@
                               type="checkbox"
                               key="key_gotoHistory"
                               oncommand="SidebarUI.toggle('viewHistorySidebar');" data-l10n-id="menu-view-history-button"/>
+<!-- MagIns - Remove Synced Tabs from the View -> Sidebar menu
                     <menuitem id="menu_tabsSidebar"
                               type="checkbox"
                               class="sync-ui-item"
-                              oncommand="SidebarUI.toggle('viewTabsSidebar');" data-l10n-id="menu-view-synced-tabs-sidebar"/>
+                              oncommand="SidebarUI.toggle('viewTabsSidebar');" data-l10n-id="menu-view-synced-tabs-sidebar"/>-->
                   </menupopup>
                 </menu>
                 <menuseparator/>
Index: firefox-84.0.2/browser/base/content/browser-context.inc
===================================================================
--- firefox-84.0.2.orig/browser/base/content/browser-context.inc
+++ firefox-84.0.2/browser/base/content/browser-context.inc
@@ -219,6 +219,7 @@
       <menuitem id="context-pocket"
                 data-l10n-id="main-context-menu-save-to-pocket"
                 oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
+<!-- MagIns - Remove "Send Page to Device" option from context menu
       <menuseparator id="context-sep-sendpagetodevice" class="sync-ui-item"
                      hidden="true"/>
       <menu id="context-sendpagetodevice"
@@ -227,7 +228,7 @@
             hidden="true">
         <menupopup id="context-sendpagetodevice-popup"
                    onpopupshowing="(() => { gSync.populateSendTabToDevicesMenu(event.target, gBrowser.currentURI.spec, gBrowser.contentTitle); })()"/>
-      </menu>
+      </menu>-->
       <menuseparator id="context-sep-viewbgimage"/>
       <menuitem id="context-viewbgimage"
                 data-l10n-id="main-context-menu-view-background-image"
@@ -287,6 +288,7 @@
                 oncommand="BrowserSearch.loadSearchFromContext(this.searchTerms, this.usePrivate, this.principal, this.csp);"/>
       <menuitem id="context-searchselect-private"
                 oncommand="BrowserSearch.loadSearchFromContext(this.searchTerms, true, this.principal, this.csp);"/>
+<!-- MagIns - Remove "Send Link to Device" option from context menu
       <menuseparator id="context-sep-sendlinktodevice" class="sync-ui-item"
                      hidden="true"/>
       <menu id="context-sendlinktodevice"
@@ -295,7 +297,7 @@
             hidden="true">
         <menupopup id="context-sendlinktodevice-popup"
                    onpopupshowing="gSync.populateSendTabToDevicesMenu(event.target, gContextMenu.linkURL, gContextMenu.linkTextStr);"/>
-      </menu>
+      </menu>-->
       <menuseparator id="frame-sep"/>
       <menu id="frame" data-l10n-id="main-context-menu-frame">
         <menupopup>
Index: firefox-84.0.2/browser/components/customizableui/CustomizableWidgets.jsm
===================================================================
--- firefox-84.0.2.orig/browser/components/customizableui/CustomizableWidgets.jsm
+++ firefox-84.0.2/browser/components/customizableui/CustomizableWidgets.jsm
@@ -572,6 +572,7 @@ const CustomizableWidgets = [
   },
 ];
 
+/* MagIns - Remove the Synced Tabs item from the Customize page
 if (Services.prefs.getBoolPref("identity.fxaccounts.enabled")) {
   CustomizableWidgets.push({
     id: "sync-button",
@@ -835,7 +836,7 @@ if (Services.prefs.getBoolPref("identity
       return showAllItem;
     },
   });
-}
+}*/
 
 let preferencesButton = {
   id: "preferences-button",
Index: firefox-84.0.2/browser/components/preferences/sync.inc.xhtml
===================================================================
--- firefox-84.0.2.orig/browser/components/preferences/sync.inc.xhtml
+++ firefox-84.0.2/browser/components/preferences/sync.inc.xhtml
@@ -196,10 +196,10 @@
                     <image class="sync-engine-image sync-engine-history"/>
                     <label data-l10n-id="sync-currently-syncing-history"/>
                   </html:div>
-                  <html:div engine_preference="services.sync.engine.tabs">
+<!--                  <html:div engine_preference="services.sync.engine.tabs">
                     <image class="sync-engine-image sync-engine-tabs"/>
                     <label data-l10n-id="sync-currently-syncing-tabs"/>
-                  </html:div>
+                  </html:div>-->
                   <html:div engine_preference="services.sync.engine.passwords">
                     <image class="sync-engine-image sync-engine-passwords"/>
                     <label data-l10n-id="sync-currently-syncing-logins-passwords"/>
@@ -233,12 +233,13 @@
         </groupbox>
       </vbox>
     </hbox>
+<!-- MagIns - Remove the Connect Another Device link from the Sync preferences page
     <vbox align="start">
       <label id="connect-another-device"
              is="text-link"
              class="fxaMobilePromo"
              data-l10n-id="sync-connect-another-device"/>
-    </vbox>
+    </vbox>-->
   </vbox>
 </deck>
 </html:template>
Index: firefox-84.0.2/browser/components/preferences/dialogs/syncChooseWhatToSync.js
===================================================================
--- firefox-84.0.2.orig/browser/components/preferences/dialogs/syncChooseWhatToSync.js
+++ firefox-84.0.2/browser/components/preferences/dialogs/syncChooseWhatToSync.js
@@ -10,7 +10,7 @@ Preferences.addAll([
   { id: "services.sync.engine.addons", type: "bool" },
   { id: "services.sync.engine.bookmarks", type: "bool" },
   { id: "services.sync.engine.history", type: "bool" },
-  { id: "services.sync.engine.tabs", type: "bool" },
+//  { id: "services.sync.engine.tabs", type: "bool" },
   { id: "services.sync.engine.prefs", type: "bool" },
   { id: "services.sync.engine.passwords", type: "bool" },
   { id: "services.sync.engine.addresses", type: "bool" },
Index: firefox-84.0.2/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
===================================================================
--- firefox-84.0.2.orig/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
+++ firefox-84.0.2/browser/components/preferences/dialogs/syncChooseWhatToSync.xhtml
@@ -34,10 +34,11 @@
       <checkbox data-l10n-id="sync-engine-history"
                 preference="services.sync.engine.history"/>
     </html:div>
+<!-- MagIns - Remove the "Open Tabs" (Synced Tabs) option from Sync preferences
     <html:div class="sync-engine-tabs">
       <checkbox data-l10n-id="sync-engine-tabs"
                 preference="services.sync.engine.tabs"/>
-    </html:div>
+    </html:div>-->
     <html:div class="sync-engine-passwords">
       <checkbox data-l10n-id="sync-engine-logins-passwords"
                 preference="services.sync.engine.passwords"/>
Index: firefox-84.0.2/browser/modules/PageActions.jsm
===================================================================
--- firefox-84.0.2.orig/browser/modules/PageActions.jsm
+++ firefox-84.0.2/browser/modules/PageActions.jsm
@@ -1263,6 +1263,7 @@ var gBuiltInActions = [
 ];
 
 // send to device
+/* MagIns - Remove "Send Tab To Device" option from Page Actions (3 dots in URL bar)
 if (Services.prefs.getBoolPref("identity.fxaccounts.enabled")) {
   gBuiltInActions.push({
     id: "sendToDevice",
@@ -1290,7 +1291,7 @@ if (Services.prefs.getBoolPref("identity
       );
     },
   });
-}
+}*/
 
 if (SiteSpecificBrowserService.isEnabled) {
   gBuiltInActions.push({
Index: firefox-84.0.2/modules/libpref/init/all.js
===================================================================
--- firefox-84.0.2.orig/modules/libpref/init/all.js
+++ firefox-84.0.2/modules/libpref/init/all.js
@@ -4516,7 +4516,7 @@ pref("services.common.log.logger.tokense
   pref("services.sync.engine.history", true);
   pref("services.sync.engine.passwords", true);
   pref("services.sync.engine.prefs", true);
-  pref("services.sync.engine.tabs", true);
+  pref("services.sync.engine.tabs", false);
   pref("services.sync.engine.tabs.filteredUrls", "^(about:.*|resource:.*|chrome:.*|wyciwyg:.*|file:.*|blob:.*|moz-extension:.*)$");
 
   // The addresses and CC engines might not actually be available at all.
Index: firefox-84.0.2/browser/components/preferences/sync.js
===================================================================
--- firefox-84.0.2.orig/browser/components/preferences/sync.js
+++ firefox-84.0.2/browser/components/preferences/sync.js
@@ -116,11 +116,11 @@ var gSyncPane = {
 
     FxAccounts.config
       .promiseConnectDeviceURI(this._getEntryPoint())
-      .then(connectURI => {
+/*      .then(connectURI => {
         document
           .getElementById("connect-another-device")
           .setAttribute("href", connectURI);
-      });
+      })*/;
     // Links for mobile devices.
     for (let platform of ["android", "ios"]) {
       let url =
