Index: firefox-100.0.2/browser/components/preferences/sync.inc.xhtml
===================================================================
--- firefox-100.0.2.orig/browser/components/preferences/sync.inc.xhtml
+++ firefox-100.0.2/browser/components/preferences/sync.inc.xhtml
@@ -64,7 +64,7 @@
             <!-- logged in and verified and all is good -->
             <hbox id="fxaLoginVerified" align="center" flex="1">
               <image id="openChangeProfileImage"
-                     class="fxaProfileImage actionable"
+                     class="fxaProfileImage"
                      role="button"
                      data-l10n-id="sync-profile-picture"/>
               <vbox flex="1" pack="center">
@@ -73,10 +73,10 @@
                     <html:h2 id="fxaDisplayNameHeading"/>
                   </label>
                   <label id="fxaEmailAddress" flex="1" crop="end"/>
-                  <button id="fxaUnlinkButton"
+<!--                  <button id="fxaUnlinkButton"
                           is="highlightable-button"
                           class="accessory-button"
-                          data-l10n-id="sync-sign-out"/>
+                          data-l10n-id="sync-sign-out"/>-->
                 </hbox>
                 <hbox>
                   <html:a id="verifiedManage" class="openLink"
Index: firefox-100.0.2/browser/components/preferences/sync.js
===================================================================
--- firefox-100.0.2.orig/browser/components/preferences/sync.js
+++ firefox-100.0.2/browser/components/preferences/sync.js
@@ -217,9 +217,9 @@ var gSyncPane = {
       gSyncPane.signIn();
       return false;
     });
-    setEventListener("fxaUnlinkButton", "command", function() {
+/*    setEventListener("fxaUnlinkButton", "command", function() {
       gSyncPane.unlinkFirefoxAccount(true);
-    });
+    });*/
     setEventListener(
       "verifyFxaAccount",
       "command",
@@ -477,7 +477,7 @@ var gSyncPane = {
   },
 
   openChangeProfileImage(event) {
-    if (this.clickOrSpaceOrEnterPressed(event)) {
+/*    if (this.clickOrSpaceOrEnterPressed(event)) {
       FxAccounts.config
         .promiseChangeAvatarURI(this._getEntryPoint())
         .then(url => {
@@ -488,7 +488,7 @@ var gSyncPane = {
         });
       // Prevent page from scrolling on the space key.
       event.preventDefault();
-    }
+    }*/
   },
 
   openManageFirefoxAccount(event) {
Index: firefox-100.0.2/browser/components/newtab/lib/DefaultSites.jsm
===================================================================
--- firefox-100.0.2.orig/browser/components/newtab/lib/DefaultSites.jsm
+++ firefox-100.0.2/browser/components/newtab/lib/DefaultSites.jsm
@@ -8,39 +8,39 @@ const DEFAULT_SITES_MAP = new Map([
   // This first item is the global list fallback for any unexpected geos
   [
     "",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.wikipedia.org/,https://www.reddit.com/,https://www.amazon.com/,https://twitter.com/",
+    "",
   ],
   [
     "US",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.amazon.com/,https://www.reddit.com/,https://www.wikipedia.org/,https://twitter.com/",
+    "",
   ],
   [
     "CA",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.reddit.com/,https://www.wikipedia.org/,https://www.amazon.ca/,https://twitter.com/",
+    "",
   ],
   [
     "DE",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.amazon.de/,https://www.ebay.de/,https://www.wikipedia.org/,https://www.reddit.com/",
+    "",
   ],
   [
     "PL",
-    "https://www.youtube.com/,https://www.facebook.com/,https://allegro.pl/,https://www.wikipedia.org/,https://www.olx.pl/,https://www.wykop.pl/",
+    "",
   ],
   [
     "RU",
-    "https://vk.com/,https://www.youtube.com/,https://ok.ru/,https://www.avito.ru/,https://www.aliexpress.com/,https://www.wikipedia.org/",
+    "",
   ],
   [
     "GB",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.reddit.com/,https://www.amazon.co.uk/,https://www.bbc.co.uk/,https://www.ebay.co.uk/",
+    "",
   ],
   [
     "FR",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.wikipedia.org/,https://www.amazon.fr/,https://www.leboncoin.fr/,https://twitter.com/",
+    "",
   ],
   [
     "CN",
-    "https://www.baidu.com/,https://www.zhihu.com/,https://www.ifeng.com/,https://weibo.com/,https://www.ctrip.com/,https://www.iqiyi.com/",
+    "",
   ],
 ]);
 
Index: firefox-100.0.2/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
===================================================================
--- firefox-100.0.2.orig/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
+++ firefox-100.0.2/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
@@ -252,11 +252,14 @@ class AboutWelcomeParent extends JSWindo
         SpecialMessageActions.handleAction(data, browser);
         break;
       case "AWPage:FXA_METRICS_FLOW_URI":
-        return FxAccounts.config.promiseMetricsFlowURI("aboutwelcome");
+/* MagIns - We do not collect metrics, thus do not provide an endpoint for about:welcome metrics
+        return FxAccounts.config.promiseMetricsFlowURI("aboutwelcome");*/
+        break;
       case "AWPage:IMPORTABLE_SITES":
         return getImportableSites();
       case "AWPage:TELEMETRY_EVENT":
-        Telemetry.sendTelemetry(data);
+/* MagIns - Disable telemetry sending
+        Telemetry.sendTelemetry(data);*/
         break;
       case "AWPage:GET_ATTRIBUTION_DATA":
         let attributionData = await AboutWelcomeDefaults.getAttributionContent();
Index: firefox-100.0.2/browser/components/preferences/preferences.js
===================================================================
--- firefox-100.0.2.orig/browser/components/preferences/preferences.js
+++ firefox-100.0.2/browser/components/preferences/preferences.js
@@ -246,7 +246,7 @@ function init_all() {
     let helpButton = document.getElementById("helpButton");
     let helpUrl =
       Services.urlFormatter.formatURLPref("app.support.baseURL") +
-      "preferences";
+      "";
     helpButton.setAttribute("href", helpUrl);
 
     document.getElementById("addonsButton").addEventListener("click", e => {
Index: firefox-100.0.2/browser/app/profile/firefox.js
===================================================================
--- firefox-100.0.2.orig/browser/app/profile/firefox.js
+++ firefox-100.0.2/browser/app/profile/firefox.js
@@ -293,16 +293,12 @@ pref("browser.shell.setDefaultPDFHandler
 // 0 = blank, 1 = home (browser.startup.homepage), 2 = last visited page, 3 = resume previous browser session
 // The behavior of option 3 is detailed at: http://wiki.mozilla.org/Session_Restore
 pref("browser.startup.page",                1);
-pref("browser.startup.homepage",            "about:home");
-#ifdef NIGHTLY_BUILD
+pref("browser.startup.homepage",            "https://privacy.app/member");
 pref("browser.startup.homepage.abouthome_cache.enabled", true);
-#else
-pref("browser.startup.homepage.abouthome_cache.enabled", false);
-#endif
 pref("browser.startup.homepage.abouthome_cache.loglevel", "Warn");
 
 // Whether we should skip the homepage when opening the first-run page
-pref("browser.startup.firstrunSkipsHomepage", true);
+pref("browser.startup.firstrunSkipsHomepage", false);
 
 // Whether we should show the session-restore infobar on startup
 pref("browser.startup.couldRestoreSession.count", 0);
@@ -2283,7 +2279,9 @@ pref("toolkit.coverage.endpoint.base", "
 pref("toolkit.coverage.opt-out", true);
 
 // Discovery prefs
-pref("browser.discovery.enabled", true);
+/* MagIns - Disable personalized recommendations in about:addons (Arkenfox 0322)
+pref("browser.discovery.enabled", true);*/
+pref("browser.discovery.enabled", false);
 pref("browser.discovery.containers.enabled", true);
 pref("browser.discovery.sites", "addons.mozilla.org");
 
@@ -2786,6 +2784,13 @@ pref("browser.snapshots.score.IsOverlapp
 pref("browser.snapshots.score.IsUserPersisted", 1);
 pref("browser.snapshots.score.IsUsedRemoved", -10);
 
+/* MagIns - Disable Recommendation pane in about:addons (Arkenfox 0320)*/
+pref("extensions.getAddons.showPane", false);
+
+/* MagIns - Disable recommendations in about:addons Extensions and Themes panes
+            (Arkenfox 0321)*/
+pref("extensions.htmlaboutaddons.recommendations.enabled", false);
+
 /* MagIns - The browser milestone records the version number of the last successful
             startup. After updates/upgrades, the user's homepage is changed to show
             a promotional page about the update. Setting this to "ignore", the value
Index: firefox-100.0.2/toolkit/content/jar.mn
===================================================================
--- firefox-100.0.2.orig/toolkit/content/jar.mn
+++ firefox-100.0.2/toolkit/content/jar.mn
@@ -51,8 +51,6 @@ toolkit.jar:
    content/global/filepicker.properties
    content/global/customElements.js
    content/global/globalOverlay.js
-   content/global/mozilla.html
-   content/global/aboutMozilla.css
    content/global/preferencesBindings.js
    content/global/process-content.js
    content/global/resetProfile.css
Index: firefox-100.0.2/browser/components/newtab/aboutwelcome/content/aboutwelcome.bundle.js
===================================================================
--- firefox-100.0.2.orig/browser/components/newtab/aboutwelcome/content/aboutwelcome.bundle.js
+++ firefox-100.0.2/browser/components/newtab/aboutwelcome/content/aboutwelcome.bundle.js
@@ -1396,10 +1396,10 @@ function addUtmParams(url, utmTerm) {
     returnUrl = new URL(url);
   }
 
-  Object.keys(BASE_PARAMS).forEach(key => {
+/*  Object.keys(BASE_PARAMS).forEach(key => {
     returnUrl.searchParams.append(key, BASE_PARAMS[key]);
   });
-  returnUrl.searchParams.append("utm_term", utmTerm);
+  returnUrl.searchParams.append("utm_term", utmTerm);*/
   return returnUrl;
 }
 
Index: firefox-100.0.2/browser/base/content/browser-menubar.inc
===================================================================
--- firefox-100.0.2.orig/browser/base/content/browser-menubar.inc
+++ firefox-100.0.2/browser/base/content/browser-menubar.inc
@@ -461,10 +461,11 @@
 #else
                           />
 #endif
+<!-- MagIns - Remove the "Submit Feedback" option in the Help menu
                 <menuitem id="feedbackPage"
                           oncommand="openFeedbackPage()"
                           data-l10n-id="menu-help-share-ideas"
-                          appmenu-data-l10n-id="appmenu-help-share-ideas"/>
+                          appmenu-data-l10n-id="appmenu-help-share-ideas"/>-->
                 <menuitem id="helpSafeMode"
                           oncommand="safeModeRestart();"
                           data-l10n-id="menu-help-enter-troubleshoot-mode2"
Index: firefox-100.0.2/browser/components/newtab/lib/SearchShortcuts.jsm
===================================================================
--- firefox-100.0.2.orig/browser/components/newtab/lib/SearchShortcuts.jsm
+++ firefox-100.0.2/browser/components/newtab/lib/SearchShortcuts.jsm
@@ -8,28 +8,17 @@ const { Services } = ChromeUtils.import(
 // List of sites we match against Topsites in order to identify sites
 // that should be converted to search Topsites
 const SEARCH_SHORTCUTS = [
-  { keyword: "@amazon", shortURL: "amazon", url: "https://amazon.com" },
-  { keyword: "@\u767E\u5EA6", shortURL: "baidu", url: "https://baidu.com" },
-  { keyword: "@google", shortURL: "google", url: "https://google.com" },
-  {
-    keyword: "@\u044F\u043D\u0434\u0435\u043A\u0441",
-    shortURL: "yandex",
-    url: "https://yandex.com",
-  },
 ];
 this.SEARCH_SHORTCUTS = SEARCH_SHORTCUTS;
 
 // These can be added via the editor but will not be added organically
 this.CUSTOM_SEARCH_SHORTCUTS = [
   ...SEARCH_SHORTCUTS,
-  { keyword: "@bing", shortURL: "bing", url: "https://bing.com" },
   {
     keyword: "@duckduckgo",
     shortURL: "duckduckgo",
     url: "https://duckduckgo.com",
   },
-  { keyword: "@ebay", shortURL: "ebay", url: "https://ebay.com" },
-  { keyword: "@twitter", shortURL: "twitter", url: "https://twitter.com" },
   {
     keyword: "@wikipedia",
     shortURL: "wikipedia",
Index: firefox-100.0.2/browser/components/newtab/lib/ActivityStream.jsm
===================================================================
--- firefox-100.0.2.orig/browser/components/newtab/lib/ActivityStream.jsm
+++ firefox-100.0.2/browser/components/newtab/lib/ActivityStream.jsm
@@ -317,7 +317,7 @@ const PREFS_CONFIG = new Map([
           return "";
         }
         const searchShortcuts = [];
-        if (geo === "CN") {
+/*        if (geo === "CN") {
           searchShortcuts.push("baidu");
         } else if (["BY", "KZ", "RU", "TR"].includes(geo)) {
           searchShortcuts.push("yandex");
@@ -326,7 +326,7 @@ const PREFS_CONFIG = new Map([
         }
         if (["DE", "FR", "GB", "IT", "JP", "US"].includes(geo)) {
           searchShortcuts.push("amazon");
-        }
+        }*/
         return searchShortcuts.join(",");
       },
     },
Index: firefox-100.0.2/browser/base/content/utilityOverlay.js
===================================================================
--- firefox-100.0.2.orig/browser/base/content/utilityOverlay.js
+++ firefox-100.0.2/browser/base/content/utilityOverlay.js
@@ -1032,9 +1032,11 @@ function openFeedbackPage() {
 }
 
 function buildHelpMenu() {
+/* MagIns - The Feedback entry is removed in browser-menubar.inc, comment out for
+            compatibility
   document.getElementById(
     "feedbackPage"
-  ).disabled = !Services.policies.isAllowed("feedbackCommands");
+  ).disabled = !Services.policies.isAllowed("feedbackCommands");*/
 
   document.getElementById(
     "helpSafeMode"
Index: firefox-100.0.2/services/settings/dumps/main/top-sites.json
===================================================================
--- firefox-100.0.2.orig/services/settings/dumps/main/top-sites.json
+++ firefox-100.0.2/services/settings/dumps/main/top-sites.json
@@ -1,400 +1,5 @@
 {
   "data": [
-    {
-      "url": "https://www.wikipedia.org/",
-      "order": 0,
-      "title": "Wikipedia",
-      "schema": 1646585618139,
-      "include_regions": [
-        "RU",
-        "BY",
-        "KZ",
-        "UA",
-        "UZ",
-        "TR"
-      ],
-      "id": "060ad5fe-40c3-490c-92bc-1ba5fc14a451",
-      "last_modified": 1646597012924
-    },
-    {
-      "url": "https://www.wikipedia.org/",
-      "order": 30,
-      "title": "Wikipedia",
-      "schema": 1646445825860,
-      "exclude_regions": [
-        "GB",
-        "CN",
-        "RU",
-        "BY",
-        "KZ",
-        "UA",
-        "UZ",
-        "TR"
-      ],
-      "id": "c109ad2c-eae5-4b47-bac7-ac2eb9eb9aab",
-      "last_modified": 1646597012919
-    },
-    {
-      "url": "https://www.amazon.ca/",
-      "order": 50,
-      "title": "Amazon",
-      "schema": 1624370092747,
-      "include_regions": [
-        "CA"
-      ],
-      "search_shortcut": false,
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "id": "82386dfa-cc7a-4fcb-9b1e-c9343d3da79d",
-      "last_modified": 1624376393938
-    },
-    {
-      "url": "https://www.amazon.com/",
-      "order": 50,
-      "title": "Amazon",
-      "schema": 1623956287723,
-      "include_locales": [],
-      "include_regions": [
-        "ES"
-      ],
-      "include_experiments": [
-        "ebay-2020-1_fallback"
-      ],
-      "id": "354690ae-3d9d-4745-a82b-7ac43e4b3213",
-      "last_modified": 1623959042903
-    },
-    {
-      "url": "https://www.amazon.fr/",
-      "order": 40,
-      "title": "Amazon",
-      "schema": 1615464108938,
-      "include_regions": [
-        "FR"
-      ],
-      "search_shortcut": true,
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "id": "61f01feb-8243-4fef-b360-733e8c5a150e",
-      "last_modified": 1616085500499
-    },
-    {
-      "url": "https://www.amazon.de/",
-      "order": 25,
-      "title": "Amazon",
-      "schema": 1615385045177,
-      "include_regions": [
-        "DE"
-      ],
-      "search_shortcut": true,
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "id": "6e34a6b9-353f-4df1-91f7-25912d621d23",
-      "last_modified": 1616085500495
-    },
-    {
-      "url": "https://www.amazon.co.uk/",
-      "order": 50,
-      "title": "Amazon",
-      "schema": 1615463950377,
-      "include_regions": [
-        "GB"
-      ],
-      "search_shortcut": true,
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "id": "bd30e776-5537-47e9-b66c-e46d6daccc50",
-      "last_modified": 1616085500491
-    },
-    {
-      "url": "https://www.amazon.com/",
-      "order": 23,
-      "title": "Amazon",
-      "schema": 1615463623635,
-      "include_regions": [
-        "US",
-        "IT",
-        "JP"
-      ],
-      "search_shortcut": true,
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "id": "2bf453fd-b588-4d99-b7e7-0d48387a62f8",
-      "last_modified": 1616085500488
-    },
-    {
-      "url": "https://www.amazon.com/",
-      "order": 50,
-      "title": "Amazon",
-      "schema": 1615377805685,
-      "exclude_regions": [
-        "PL",
-        "RU",
-        "CN",
-        "GB",
-        "FR",
-        "CA",
-        "DE",
-        "US",
-        "IT",
-        "JP"
-      ],
-      "search_shortcut": false,
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "send_attribution_request": false,
-      "id": "78e5e650-e27c-485c-a321-f1ea3d161c34",
-      "last_modified": 1615385045111
-    },
-    {
-      "url": "https://www.google.com/",
-      "order": 0,
-      "title": "Google",
-      "schema": 1611836508660,
-      "exclude_regions": [
-        "BY",
-        "KZ",
-        "RU",
-        "TR",
-        "CN"
-      ],
-      "search_shortcut": true,
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "id": "0ee067b7-c0ca-4cf6-96f8-320c2a1b927b",
-      "last_modified": 1611838808382
-    },
-    {
-      "url": "https://www.baidu.com/",
-      "order": 10,
-      "schema": 1611835746769,
-      "include_regions": [
-        "CN"
-      ],
-      "search_shortcut": true,
-      "id": "1bfdbe7c-570f-494a-b61c-20c5c64dd69d",
-      "last_modified": 1611838808375
-    },
-    {
-      "url": "https://www.ebay.co.uk/",
-      "order": 70,
-      "title": "eBay",
-      "schema": 1610661703633,
-      "include_regions": [
-        "GB"
-      ],
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "id": "2abe6385-2f49-49ca-a4ff-0e635bc2ab7c",
-      "last_modified": 1611076538693
-    },
-    {
-      "url": "https://www.ebay.de/",
-      "order": 26,
-      "title": "eBay",
-      "schema": 1608200991177,
-      "include_regions": [
-        "DE"
-      ],
-      "exclude_experiments": [
-        "ebay-2020-1"
-      ],
-      "id": "61b60614-1285-41c6-a0c1-9fe12f3983c4",
-      "last_modified": 1608239138349
-    },
-    {
-      "url": "https://www.reddit.com/",
-      "order": 25,
-      "title": "Reddit",
-      "schema": 1600861208039,
-      "include_regions": [
-        "US",
-        "CA"
-      ],
-      "id": "cfac4c2a-4788-450d-8e92-ba6b26a1faf1",
-      "last_modified": 1603813848271
-    },
-    {
-      "url": "https://www.reddit.com/",
-      "order": 40,
-      "title": "Reddit",
-      "schema": 1600861199348,
-      "exclude_regions": [
-        "PL",
-        "RU",
-        "FR",
-        "CN",
-        "US",
-        "CA"
-      ],
-      "id": "7a129df1-ec8f-4bf3-a83e-49b90409d1be",
-      "last_modified": 1603813848259
-    },
-    {
-      "url": "https://www.zhihu.com/",
-      "order": 20,
-      "schema": 1599055903630,
-      "include_regions": [
-        "CN"
-      ],
-      "id": "568fca49-6921-4895-997b-60276d6ea000",
-      "last_modified": 1599146777461
-    },
-    {
-      "url": "https://www.ifeng.com/",
-      "order": 30,
-      "schema": 1599055904425,
-      "include_regions": [
-        "CN"
-      ],
-      "id": "8d392627-d892-4b51-96a7-3b2ec47c6cdd",
-      "last_modified": 1599146777456
-    },
-    {
-      "url": "https://weibo.com/",
-      "order": 40,
-      "schema": 1599055905429,
-      "include_regions": [
-        "CN"
-      ],
-      "id": "e707b8ac-e844-4b14-a1bc-ce6968c777d1",
-      "last_modified": 1599146777450
-    },
-    {
-      "url": "https://www.ctrip.com/",
-      "order": 50,
-      "schema": 1599055906460,
-      "include_regions": [
-        "CN"
-      ],
-      "id": "38ac6a09-6d0b-43fc-894b-d214e6d5b684",
-      "last_modified": 1599146777446
-    },
-    {
-      "url": "https://www.iqiyi.com/",
-      "order": 60,
-      "schema": 1599055907301,
-      "include_regions": [
-        "CN"
-      ],
-      "id": "cbb986d2-665d-4967-b74b-41ecdd7c6188",
-      "last_modified": 1599146777441
-    },
-    {
-      "url": "https://allegro.pl/",
-      "order": 25,
-      "schema": 1599055926547,
-      "include_regions": [
-        "PL"
-      ],
-      "id": "dfbd59db-d7d5-4efe-b379-5c3af46e390b",
-      "last_modified": 1599146777436
-    },
-    {
-      "url": "https://www.olx.pl/",
-      "order": 40,
-      "schema": 1599056211503,
-      "include_regions": [
-        "PL"
-      ],
-      "id": "88f3bf01-350c-4fbb-97c9-c2b72b93274c",
-      "last_modified": 1599146777431
-    },
-    {
-      "url": "https://www.wykop.pl/",
-      "order": 50,
-      "schema": 1599056212302,
-      "include_regions": [
-        "PL"
-      ],
-      "id": "7fcb96b5-b24f-4bb0-8b0d-c363ef24973b",
-      "last_modified": 1599146777426
-    },
-    {
-      "url": "https://www.avito.ru/",
-      "order": 12,
-      "schema": 1599056215077,
-      "include_regions": [
-        "RU"
-      ],
-      "id": "92af59ac-7e4f-48b0-b20c-a7a39f0496ee",
-      "last_modified": 1599146777411
-    },
-    {
-      "url": "https://www.aliexpress.com/",
-      "order": 13,
-      "schema": 1599056215899,
-      "include_regions": [
-        "RU"
-      ],
-      "id": "881bc01e-45f8-4c9d-b650-3940f2aa430b",
-      "last_modified": 1599146777406
-    },
-    {
-      "url": "https://www.leboncoin.fr/",
-      "order": 50,
-      "schema": 1599056218470,
-      "include_regions": [
-        "FR"
-      ],
-      "id": "aa4bb35d-ddaf-452f-af32-cbf80d67a135",
-      "last_modified": 1599146777401
-    },
-    {
-      "url": "https://twitter.com/",
-      "order": 60,
-      "title": "Twitter",
-      "schema": 1599056605399,
-      "exclude_regions": [
-        "PL",
-        "RU",
-        "DE",
-        "GB",
-        "CN"
-      ],
-      "id": "4c49a54c-6515-495f-b24d-4aae165479fc",
-      "last_modified": 1599146777375
-    },
-    {
-      "url": "https://www.youtube.com/",
-      "order": 10,
-      "title": "YouTube",
-      "schema": 1599056606419,
-      "exclude_regions": [
-        "CN"
-      ],
-      "id": "12abfe01-c608-4060-8fff-c3df73ff3fc0",
-      "last_modified": 1599146777370
-    },
-    {
-      "url": "https://www.facebook.com/",
-      "order": 20,
-      "title": "Facebook",
-      "schema": 1599056607418,
-      "exclude_regions": [
-        "RU",
-        "CN"
-      ],
-      "id": "19045f47-6dde-43a8-9511-3fc0c4555a09",
-      "last_modified": 1599146777365
-    },
-    {
-      "url": "https://www.bbc.co.uk/",
-      "order": 60,
-      "title": "BBC",
-      "schema": 1599056678589,
-      "include_regions": [
-        "GB"
-      ],
-      "id": "89e3da9e-1d84-4c24-beab-d0692fc7cc45",
-      "last_modified": 1599146777345
-    }
+
   ]
 }
Index: firefox-100.0.2/browser/components/newtab/lib/TopSitesFeed.jsm
===================================================================
--- firefox-100.0.2.orig/browser/components/newtab/lib/TopSitesFeed.jsm
+++ firefox-100.0.2/browser/components/newtab/lib/TopSitesFeed.jsm
@@ -582,6 +582,7 @@ this.TopSitesFeed = class TopSitesFeed {
    * @returns {Boolean} Did we insert any search shortcuts?
    */
   async _maybeInsertSearchShortcuts(plainPinnedSites) {
+/* MagIns - We do not want to pin any search engines for users
     // Only insert shortcuts if the experiment is running
     if (this.store.getState().Prefs.values[SEARCH_SHORTCUTS_EXPERIMENT]) {
       // We don't want to insert shortcuts we've previously inserted
@@ -653,7 +654,7 @@ this.TopSitesFeed = class TopSitesFeed {
         return true;
       }
     }
-
+    */
     return false;
   }
 
Index: firefox-100.0.2/browser/base/content/appmenu-viewcache.inc.xhtml
===================================================================
--- firefox-100.0.2.orig/browser/base/content/appmenu-viewcache.inc.xhtml
+++ firefox-100.0.2/browser/base/content/appmenu-viewcache.inc.xhtml
@@ -543,10 +543,12 @@
 
   <panelview id="PanelUI-fxa" class="PanelUI-subView" descriptionheightworkaround="true">
     <vbox id="PanelUI-fxa-menu" class="panel-subview-body">
+<!-- MagIns - Skip the submenu for FxA and go straight to /member. Original line:
+                     oncommand="gSync.clickFxAMenuHeaderButton(this);" -->
       <toolbarbutton id="fxa-manage-account-button"
                      align="center"
                      class="subviewbutton"
-                     oncommand="gSync.clickFxAMenuHeaderButton(this);">
+                     oncommand="gSync.openFxAManagePageFromFxaMenu(this);">
         <vbox flex="1">
           <label id="fxa-menu-header-title"
                  crop="end"
Index: firefox-100.0.2/browser/base/content/default-bookmarks.html
===================================================================
--- firefox-100.0.2.orig/browser/base/content/default-bookmarks.html
+++ firefox-100.0.2/browser/base/content/default-bookmarks.html
@@ -19,11 +19,11 @@
     <!-- These localization links are not automatically applied to any XHR
          response body and must be applied manually as well.  They are included
          so that viewing the file directly shows the results. -->
-    <link rel="localization" href="branding/brand.ftl"/>
+<!--    <link rel="localization" href="branding/brand.ftl"/>
     <link rel="localization" href="browser/profile/default-bookmarks.ftl"/>
-    <title data-l10n-id="default-bookmarks-title">default-bookmarks-title</title>
+    <title data-l10n-id="default-bookmarks-title">default-bookmarks-title</title>-->
 </head>
-<body>
+<!--<body>
 <h1 data-l10n-id="default-bookmarks-heading">default-bookmarks-heading</h1>
 
 <dl><p>
@@ -55,5 +55,5 @@
         </dl><p>
 #endif
 </dl>
-</body>
+</body>-->
 </html>
