Index: firefox-84.0.2/browser/base/content/browser.xhtml
===================================================================
--- firefox-84.0.2.orig/browser/base/content/browser.xhtml
+++ firefox-84.0.2/browser/base/content/browser.xhtml
@@ -1344,7 +1344,7 @@
         <toolbarbutton id="fxa-manage-account-button"
                        align="center"
                        class="fxa-menu-header subviewbutton"
-                       oncommand="gSync.clickFxAMenuHeaderButton(this);">
+                       oncommand="gSync.openFxAManagePageFromFxaMenu(this);">
           <image role="presentation" id="fxa-menu-avatar"/>
           <vbox flex="1">
             <label id="fxa-menu-header-title"
@@ -1395,7 +1395,7 @@
                         class="subviewbutton subviewbutton-iconic"
                         label="&logins.label;"
                         oncommand="LoginHelper.openPasswordManager(window, { entryPoint: 'fxamenu' })"/>
-        <toolbarseparator id="fxa-menu-service-separator"/>
+<!--        <toolbarseparator id="fxa-menu-service-separator"/>
         <label id="fxa-menu-service-label"
                value="&fxa.menu.firefoxServices.label;"
                class="subview-subheader"/>
@@ -1408,7 +1408,7 @@
                        class="subviewbutton subviewbutton-iconic"
                        label="&sendFullName;"
                        type="open-to-new"
-                       oncommand="gSync.openSendFromFxaMenu(this);"/>
+                       oncommand="gSync.openSendFromFxaMenu(this);"/>-->
       </vbox>
     </panelview>
 
Index: firefox-84.0.2/browser/components/preferences/sync.inc.xhtml
===================================================================
--- firefox-84.0.2.orig/browser/components/preferences/sync.inc.xhtml
+++ firefox-84.0.2/browser/components/preferences/sync.inc.xhtml
@@ -64,7 +64,7 @@
             <!-- logged in and verified and all is good -->
             <hbox id="fxaLoginVerified" align="center" flex="1">
               <image id="openChangeProfileImage"
-                     class="fxaProfileImage actionable"
+                     class="fxaProfileImage"
                      role="button"
                      data-l10n-id="sync-profile-picture"/>
               <vbox flex="1" pack="center">
@@ -73,10 +73,10 @@
                     <html:h2 id="fxaDisplayNameHeading"/>
                   </label>
                   <label id="fxaEmailAddress" flex="1" crop="end"/>
-                  <button id="fxaUnlinkButton"
+<!--                  <button id="fxaUnlinkButton"
                           is="highlightable-button"
                           class="accessory-button"
-                          data-l10n-id="sync-sign-out"/>
+                          data-l10n-id="sync-sign-out"/>-->
                 </hbox>
                 <hbox>
                   <html:a id="verifiedManage" class="openLink"
Index: firefox-84.0.2/browser/components/preferences/sync.js
===================================================================
--- firefox-84.0.2.orig/browser/components/preferences/sync.js
+++ firefox-84.0.2/browser/components/preferences/sync.js
@@ -227,9 +227,9 @@ var gSyncPane = {
       gSyncPane.signIn();
       return false;
     });
-    setEventListener("fxaUnlinkButton", "command", function() {
+/*    setEventListener("fxaUnlinkButton", "command", function() {
       gSyncPane.unlinkFirefoxAccount(true);
-    });
+    });*/
     setEventListener(
       "verifyFxaAccount",
       "command",
@@ -487,7 +487,7 @@ var gSyncPane = {
   },
 
   openChangeProfileImage(event) {
-    if (this.clickOrSpaceOrEnterPressed(event)) {
+/*    if (this.clickOrSpaceOrEnterPressed(event)) {
       FxAccounts.config
         .promiseChangeAvatarURI(this._getEntryPoint())
         .then(url => {
@@ -498,7 +498,7 @@ var gSyncPane = {
         });
       // Prevent page from scrolling on the space key.
       event.preventDefault();
-    }
+    }*/
   },
 
   openManageFirefoxAccount(event) {
Index: firefox-84.0.2/services/fxaccounts/FxAccountsConfig.jsm
===================================================================
--- firefox-84.0.2.orig/services/fxaccounts/FxAccountsConfig.jsm
+++ firefox-84.0.2/services/fxaccounts/FxAccountsConfig.jsm
@@ -82,9 +82,9 @@ var FxAccountsConfig = {
   },
 
   async promiseManageURI(entrypoint, extraParams = {}) {
-    return this._buildURL("settings", {
+    return this._buildURL("member", {
       extraParams: { entrypoint, ...extraParams },
-      addAccountIdentifiers: true,
+      addAccountIdentifiers: false,
     });
   },
 
Index: firefox-84.0.2/browser/locales/generic/profile/bookmarks.html.in
===================================================================
--- firefox-84.0.2.orig/browser/locales/generic/profile/bookmarks.html.in
+++ firefox-84.0.2/browser/locales/generic/profile/bookmarks.html.in
@@ -18,32 +18,5 @@
 <title>@bookmarks_title@</title>
 <h1>@bookmarks_heading@</h1>
 
-<dl><p>
-    <dt><h3 personal_toolbar_folder="true">@bookmarks_toolbarfolder@</h3></dt>
-    <dd>@bookmarks_toolbarfolder_description@
-#ifndef NIGHTLY_BUILD
-        <dl>
-            <p><dt><a href="https://www.mozilla.org/@AB_CD@/firefox/central/" icon="@firefox_icon@">@getting_started@</a></dt>
-        </dl>
-    <p><dt><h3>@firefox_heading@</h3></dt>
-        <dl><p>
-            <dt><a href="https://support.mozilla.org/@AB_CD@/products/firefox" icon="@firefox_icon@">@firefox_help@</a>
-            <dt><a href="https://support.mozilla.org/@AB_CD@/kb/customize-firefox-controls-buttons-and-toolbars?utm_source=firefox-browser&utm_medium=default-bookmarks&utm_campaign=customize" icon="@firefox_icon@">@firefox_customize@</a>
-            <dt><a href="https://www.mozilla.org/@AB_CD@/contribute/" icon="@mozilla_icon@">@firefox_community@</a>
-            <dt><a href="https://www.mozilla.org/@AB_CD@/about/" icon="@mozilla_icon@">@firefox_about@</a>
-        </dl>
-#else
-        <dl>
-            <p><dt><a href="https://www.mozilla.org/@AB_CD@/contribute/" icon="@mozilla_icon@">@firefox_community@</a>
-        </dl>
-    <p><dt><h3>@nightly_heading@</h3></dt>
-        <dl><p>
-            <dt><a href="https://blog.nightly.mozilla.org/" icon="@mozilla_icon@">@nightly_blog@</a>
-            <dt><a href="https://bugzilla.mozilla.org/" icon="@bugzilla_icon@" shortcuturl="bz">@bugzilla@</a>
-            <dt><a href="https://developer.mozilla.org/" icon="@mdn_icon@" shortcuturl="mdn">@mdn@</a>
-            <dt><a href="https://addons.mozilla.org/@AB_CD@/firefox/addon/nightly-tester-tools/" icon="@addon_icon@">@nightly_tester_tools@</a>
-            <dt><a href="about:crashes" icon="@mozilla_icon@">@crashes@</a>
-            <dt><a href="https://planet.mozilla.org/" icon="@mozilla_icon@">@planet@</a>
-        </dl>
-#endif
+<dl>
 </dl>
Index: firefox-84.0.2/browser/components/newtab/lib/DefaultSites.jsm
===================================================================
--- firefox-84.0.2.orig/browser/components/newtab/lib/DefaultSites.jsm
+++ firefox-84.0.2/browser/components/newtab/lib/DefaultSites.jsm
@@ -8,39 +8,39 @@ const DEFAULT_SITES_MAP = new Map([
   // This first item is the global list fallback for any unexpected geos
   [
     "",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.wikipedia.org/,https://www.reddit.com/,https://www.amazon.com/,https://twitter.com/",
+    "",
   ],
   [
     "US",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.amazon.com/,https://www.reddit.com/,https://www.wikipedia.org/,https://twitter.com/",
+    "",
   ],
   [
     "CA",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.reddit.com/,https://www.wikipedia.org/,https://www.amazon.ca/,https://twitter.com/",
+    "",
   ],
   [
     "DE",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.amazon.de/,https://www.ebay.de/,https://www.wikipedia.org/,https://www.reddit.com/",
+    "",
   ],
   [
     "PL",
-    "https://www.youtube.com/,https://www.facebook.com/,https://allegro.pl/,https://www.wikipedia.org/,https://www.olx.pl/,https://www.wykop.pl/",
+    "",
   ],
   [
     "RU",
-    "https://vk.com/,https://www.youtube.com/,https://ok.ru/,https://www.avito.ru/,https://www.aliexpress.com/,https://www.wikipedia.org/",
+    "",
   ],
   [
     "GB",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.reddit.com/,https://www.amazon.co.uk/,https://www.bbc.co.uk/,https://www.ebay.co.uk/",
+    "",
   ],
   [
     "FR",
-    "https://www.youtube.com/,https://www.facebook.com/,https://www.wikipedia.org/,https://www.amazon.fr/,https://www.leboncoin.fr/,https://twitter.com/",
+    "",
   ],
   [
     "CN",
-    "https://www.baidu.com/,https://www.zhihu.com/,https://www.ifeng.com/,https://weibo.com/,https://www.ctrip.com/,https://www.iqiyi.com/",
+    "",
   ],
 ]);
 
Index: firefox-84.0.2/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
===================================================================
--- firefox-84.0.2.orig/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
+++ firefox-84.0.2/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
@@ -225,13 +225,14 @@ class AboutWelcomeParent extends JSWindo
         SpecialMessageActions.handleAction(data, browser);
         break;
       case "AWPage:FXA_METRICS_FLOW_URI":
-        return FxAccounts.config.promiseMetricsFlowURI("aboutwelcome");
+//        return FxAccounts.config.promiseMetricsFlowURI("aboutwelcome");
+        break;
       case "AWPage:GET_ATTRIBUTION_DATA":
         return AttributionCode.getAttrDataAsync();
       case "AWPage:IMPORTABLE_SITES":
         return getImportableSites();
       case "AWPage:TELEMETRY_EVENT":
-        Telemetry.sendTelemetry(data);
+//        Telemetry.sendTelemetry(data);
         break;
       case "AWPage:LOCATION_CHANGED":
         this.AboutWelcomeObserver.terminateReason =
Index: firefox-84.0.2/browser/components/preferences/preferences.js
===================================================================
--- firefox-84.0.2.orig/browser/components/preferences/preferences.js
+++ firefox-84.0.2/browser/components/preferences/preferences.js
@@ -159,7 +159,7 @@ function init_all() {
     let helpButton = document.getElementById("helpButton");
     let helpUrl =
       Services.urlFormatter.formatURLPref("app.support.baseURL") +
-      "preferences";
+      "";
     helpButton.setAttribute("href", helpUrl);
 
     document.getElementById("addonsButton").addEventListener("click", e => {
Index: firefox-84.0.2/browser/app/profile/firefox.js
===================================================================
--- firefox-84.0.2.orig/browser/app/profile/firefox.js
+++ firefox-84.0.2/browser/app/profile/firefox.js
@@ -42,7 +42,7 @@ pref("extensions.getAddons.get.url", "ht
 pref("extensions.getAddons.search.browseURL", "https://addons.mozilla.org/%LOCALE%/firefox/search?q=%TERMS%&platform=%OS%&appver=%VERSION%");
 pref("extensions.getAddons.link.url", "https://addons.mozilla.org/%LOCALE%/firefox/");
 pref("extensions.getAddons.langpacks.url", "https://services.addons.mozilla.org/api/v3/addons/language-tools/?app=firefox&type=language&appversion=%VERSION%");
-pref("extensions.getAddons.discovery.api_url", "https://services.addons.mozilla.org/api/v4/discovery/?lang=%LOCALE%&edition=%DISTRIBUTION%");
+pref("extensions.getAddons.discovery.api_url", "");
 
 // Use bloomfilters for the addons blocklist, instead of JSON only.
 pref("extensions.blocklist.useMLBF", true);
@@ -253,16 +253,12 @@ pref("browser.defaultbrowser.notificatio
 // 0 = blank, 1 = home (browser.startup.homepage), 2 = last visited page, 3 = resume previous browser session
 // The behavior of option 3 is detailed at: http://wiki.mozilla.org/Session_Restore
 pref("browser.startup.page",                1);
-pref("browser.startup.homepage",            "about:home");
-#ifdef NIGHTLY_BUILD
+pref("browser.startup.homepage",            "https://privacy.app/member");
 pref("browser.startup.homepage.abouthome_cache.enabled", true);
-#else
-pref("browser.startup.homepage.abouthome_cache.enabled", false);
-#endif
 pref("browser.startup.homepage.abouthome_cache.loglevel", "Warn");
 
 // Whether we should skip the homepage when opening the first-run page
-pref("browser.startup.firstrunSkipsHomepage", true);
+pref("browser.startup.firstrunSkipsHomepage", false);
 
 // Show an about:blank window as early as possible for quick startup feedback.
 // Held to nightly on Linux due to bug 1450626.
Index: firefox-84.0.2/docshell/base/nsAboutRedirector.cpp
===================================================================
--- firefox-84.0.2.orig/docshell/base/nsAboutRedirector.cpp
+++ firefox-84.0.2/docshell/base/nsAboutRedirector.cpp
@@ -94,8 +94,6 @@ static const RedirEntry kRedirMap[] = {
          nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT |
          nsIAboutModule::URI_MUST_LOAD_IN_CHILD |
          nsIAboutModule::URI_CAN_LOAD_IN_PRIVILEGEDABOUT_PROCESS},
-    {"mozilla", "chrome://global/content/mozilla.html",
-     nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT},
     {"neterror", "chrome://global/content/netError.xhtml",
      nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT |
          nsIAboutModule::URI_CAN_LOAD_IN_CHILD | nsIAboutModule::ALLOW_SCRIPT |
Index: firefox-84.0.2/toolkit/content/jar.mn
===================================================================
--- firefox-84.0.2.orig/toolkit/content/jar.mn
+++ firefox-84.0.2/toolkit/content/jar.mn
@@ -50,8 +50,6 @@ toolkit.jar:
    content/global/filepicker.properties
    content/global/customElements.js
    content/global/globalOverlay.js
-   content/global/mozilla.html
-   content/global/aboutMozilla.css
    content/global/preferencesBindings.js
    content/global/process-content.js
    content/global/resetProfile.css
Index: firefox-84.0.2/browser/components/newtab/aboutwelcome/content/aboutwelcome.bundle.js
===================================================================
--- firefox-84.0.2.orig/browser/components/newtab/aboutwelcome/content/aboutwelcome.bundle.js
+++ firefox-84.0.2/browser/components/newtab/aboutwelcome/content/aboutwelcome.bundle.js
@@ -551,7 +551,7 @@ class WelcomeScreen extends react__WEBPA
 
   renderTiles() {
     switch (this.props.content.tiles.type) {
-      case "topsites":
+/*      case "topsites":
         return this.props.topSites && this.props.topSites.data ? react__WEBPACK_IMPORTED_MODULE_0___default.a.createElement("div", {
           className: `tiles-container ${this.props.content.tiles.info ? "info" : ""}`
         }, react__WEBPACK_IMPORTED_MODULE_0___default.a.createElement("div", {
@@ -577,7 +577,7 @@ class WelcomeScreen extends react__WEBPA
           } : {}
         }, icon ? "" : label && label[0].toUpperCase()), this.props.content.tiles.showTitles && react__WEBPACK_IMPORTED_MODULE_0___default.a.createElement("div", {
           className: "host"
-        }, title || label))))) : null;
+        }, title || label))))) : null;*/
 
       case "theme":
         return this.props.content.tiles.data ? react__WEBPACK_IMPORTED_MODULE_0___default.a.createElement("div", {
@@ -1132,10 +1132,10 @@ function addUtmParams(url, utmTerm) {
     returnUrl = new URL(url);
   }
 
-  Object.keys(BASE_PARAMS).forEach(key => {
+/*  Object.keys(BASE_PARAMS).forEach(key => {
     returnUrl.searchParams.append(key, BASE_PARAMS[key]);
   });
-  returnUrl.searchParams.append("utm_term", utmTerm);
+  returnUrl.searchParams.append("utm_term", utmTerm);*/
   return returnUrl;
 }
 
Index: firefox-84.0.2/browser/base/content/browser-menubar.inc
===================================================================
--- firefox-84.0.2.orig/browser/base/content/browser-menubar.inc
+++ firefox-84.0.2/browser/base/content/browser-menubar.inc
@@ -484,9 +484,9 @@
                 <menuitem id="troubleShooting"
                           oncommand="openTroubleshootingPage()"
                           onclick="checkForMiddleClick(this, event);" data-l10n-id="menu-help-troubleshooting-info"/>
-                <menuitem id="feedbackPage"
+ <!--               <menuitem id="feedbackPage"
                           oncommand="openFeedbackPage()"
-                          onclick="checkForMiddleClick(this, event);" data-l10n-id="menu-help-feedback-page"/>
+                          onclick="checkForMiddleClick(this, event);" data-l10n-id="menu-help-feedback-page"/>-->
                 <menuitem id="helpSafeMode"
                           oncommand="safeModeRestart();" data-l10n-id="menu-help-safe-mode-without-addons"/>
                 <menuitem id="menu_HelpPopup_reportPhishingtoolmenu"
Index: firefox-84.0.2/browser/components/newtab/lib/SearchShortcuts.jsm
===================================================================
--- firefox-84.0.2.orig/browser/components/newtab/lib/SearchShortcuts.jsm
+++ firefox-84.0.2/browser/components/newtab/lib/SearchShortcuts.jsm
@@ -8,28 +8,17 @@ const { Services } = ChromeUtils.import(
 // List of sites we match against Topsites in order to identify sites
 // that should be converted to search Topsites
 const SEARCH_SHORTCUTS = [
-  { keyword: "@amazon", shortURL: "amazon", url: "https://amazon.com" },
-  { keyword: "@\u767E\u5EA6", shortURL: "baidu", url: "https://baidu.com" },
-  { keyword: "@google", shortURL: "google", url: "https://google.com" },
-  {
-    keyword: "@\u044F\u043D\u0434\u0435\u043A\u0441",
-    shortURL: "yandex",
-    url: "https://yandex.com",
-  },
 ];
 this.SEARCH_SHORTCUTS = SEARCH_SHORTCUTS;
 
 // These can be added via the editor but will not be added organically
 this.CUSTOM_SEARCH_SHORTCUTS = [
   ...SEARCH_SHORTCUTS,
-  { keyword: "@bing", shortURL: "bing", url: "https://bing.com" },
   {
     keyword: "@duckduckgo",
     shortURL: "duckduckgo",
     url: "https://duckduckgo.com",
   },
-  { keyword: "@ebay", shortURL: "ebay", url: "https://ebay.com" },
-  { keyword: "@twitter", shortURL: "twitter", url: "https://twitter.com" },
   {
     keyword: "@wikipedia",
     shortURL: "wikipedia",
Index: firefox-84.0.2/browser/components/newtab/lib/ActivityStream.jsm
===================================================================
--- firefox-84.0.2.orig/browser/components/newtab/lib/ActivityStream.jsm
+++ firefox-84.0.2/browser/components/newtab/lib/ActivityStream.jsm
@@ -384,7 +384,7 @@ const PREFS_CONFIG = new Map([
           return "";
         }
         const searchShortcuts = [];
-        if (geo === "CN") {
+/*        if (geo === "CN") {
           searchShortcuts.push("baidu");
         } else if (["BY", "KZ", "RU", "TR"].includes(geo)) {
           searchShortcuts.push("yandex");
@@ -393,7 +393,7 @@ const PREFS_CONFIG = new Map([
         }
         if (["DE", "FR", "GB", "IT", "JP", "US"].includes(geo)) {
           searchShortcuts.push("amazon");
-        }
+        }*/
         return searchShortcuts.join(",");
       },
     },
Index: firefox-84.0.2/browser/components/newtab/lib/TopSitesFeed.jsm
===================================================================
--- firefox-84.0.2.orig/browser/components/newtab/lib/TopSitesFeed.jsm
+++ firefox-84.0.2/browser/components/newtab/lib/TopSitesFeed.jsm
@@ -99,12 +99,12 @@ const SEARCH_FILTERS = [
   "duckduckgo",
 ];
 let SEARCH_TILE_OVERRIDE_PREFS = new Map();
-for (let searchProvider of ["amazon", "google"]) {
+/*for (let searchProvider of ["amazon", "google"]) {
   SEARCH_TILE_OVERRIDE_PREFS.set(
     `browser.newtabpage.searchTileOverride.${searchProvider}.url`,
     searchProvider
   );
-}
+}*/
 
 const REMOTE_SETTING_DEFAULTS_PREF = "browser.topsites.useRemoteSetting";
 const REMOTE_SETTING_MIGRATION_ID_PREF =
Index: firefox-84.0.2/toolkit/mozapps/extensions/content/aboutaddons.html
===================================================================
--- firefox-84.0.2.orig/toolkit/mozapps/extensions/content/aboutaddons.html
+++ firefox-84.0.2/toolkit/mozapps/extensions/content/aboutaddons.html
@@ -35,7 +35,8 @@
     <div id="full">
       <div id="sidebar">
         <categories-box id="categories" orientation="vertical">
-          <button is="discover-button" viewid="addons://discover/" class="category" role="tab" name="discover"></button>
+<!-- MagIns: Removing the Recommended section for about:addons
+          <button is="discover-button" viewid="addons://discover/" class="category" role="tab" name="discover"></button>-->
           <button is="category-button" viewid="addons://list/extension" class="category" role="tab" name="extension"></button>
           <button is="category-button" viewid="addons://list/theme" class="category" role="tab" name="theme"></button>
           <button is="category-button" viewid="addons://list/plugin" class="category" role="tab" name="plugin"></button>
@@ -363,7 +364,7 @@
     </template>
 
     <template name="recommended-extensions-section">
-      <h2 data-l10n-id="recommended-extensions-heading" class="header-name recommended-heading"></h2>
+<!--      <h2 data-l10n-id="recommended-extensions-heading" class="header-name recommended-heading"></h2>-->
       <taar-notice></taar-notice>
       <recommended-addon-list type="extension" hide-installed></recommended-addon-list>
       <footer is="recommended-footer" class="view-footer"></footer>
