Index: mozilla-release/browser/app/profile/firefox.js
===================================================================
--- mozilla-release.orig/browser/app/profile/firefox.js
+++ mozilla-release/browser/app/profile/firefox.js
@@ -1654,7 +1654,9 @@ pref("browser.newtabpage.activity-stream
 pref("browser.newtabpage.activity-stream.discoverystream.spocs.personalized", true);
 
 // User pref to show stories on newtab (feeds.system.topstories has to be set to true as well)
-pref("browser.newtabpage.activity-stream.feeds.section.topstories", true);
+/* MagIns - Show sponsored news stories on the Home page
+pref("browser.newtabpage.activity-stream.feeds.section.topstories", true);*/
+pref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
 
 // The pref controls if search hand-off is enabled for Activity Stream.
 pref("browser.newtabpage.activity-stream.improvesearch.handoffToAwesomebar", true);
@@ -3040,4 +3042,12 @@ pref("dom.event.clipboardevents.enabled"
 //3 - Only. Only use TRR. Never use the native (This mode also requires the bootstrapAddress pref to be set)
 //4 - Reserved (used to be Shadow mode)
 //5 - Off by choice. This is the same as 0 but marks it as done by choice and not done by default.
-pref("network.trr.mode", 5);
\ No newline at end of file
+pref("network.trr.mode", 5);
+
+/* MagIns - On the Home Page, Mozilla by default allows sponsored articles to appear in the
+            "Recommended by Pocket" section. This preference disables sponsored stories.*/
+pref("browser.newtabpage.activity-stream.showSponsored", false);
+
+/* MagIns - On the Home Page, Mozilla by default allows sponsored sites to appear in the "Top
+            Sites" section. This preference disables sponsored sites.*/
+pref("browser.newtabpage.activity-stream.showSponsoredTopSites", false);
\ No newline at end of file
Index: mozilla-release/browser/base/content/browser-context.inc
===================================================================
--- mozilla-release.orig/browser/base/content/browser-context.inc
+++ mozilla-release/browser/base/content/browser-context.inc
@@ -99,9 +99,9 @@
       <menuitem id="context-savelink"
                 data-l10n-id="main-context-menu-save-link"
                 oncommand="gContextMenu.saveLink();"/>
-      <menuitem id="context-savelinktopocket"
+<!--      <menuitem id="context-savelinktopocket"
                 data-l10n-id="main-context-menu-save-link-to-pocket"
-                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>
+                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>-->
       <menuitem id="context-copyemail"
                 data-l10n-id="main-context-menu-copy-email"
                 oncommand="gContextMenu.copyEmail();"/>
@@ -259,9 +259,10 @@
       <menuitem id="context-savepage"
                 data-l10n-id="main-context-menu-page-save"
                 oncommand="gContextMenu.savePageAs();"/>
+<!-- MagIns - Remove "Save Page to Pocket" option from context menu
       <menuitem id="context-pocket"
                 data-l10n-id="main-context-menu-save-to-pocket"
-                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
+                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>-->
 <!-- MagIns - Remove "Send Page to Device" option from context menu
       <menu id="context-sendpagetodevice"
             class="sync-ui-item"
Index: mozilla-release/browser/components/BrowserGlue.sys.mjs
===================================================================
--- mozilla-release.orig/browser/components/BrowserGlue.sys.mjs
+++ mozilla-release/browser/components/BrowserGlue.sys.mjs
@@ -1292,7 +1292,8 @@ BrowserGlue.prototype = {
       lazy.Normandy.init();
     }
 
-    lazy.SaveToPocket.init();
+/* MagIns - Disable Pocket feature
+    lazy.SaveToPocket.init();*/
 
     AboutHomeStartupCache.init();
 
Index: mozilla-release/browser/components/newtab/content-src/lib/link-menu-options.js
===================================================================
--- mozilla-release.orig/browser/components/newtab/content-src/lib/link-menu-options.js
+++ mozilla-release/browser/components/newtab/content-src/lib/link-menu-options.js
@@ -215,7 +215,8 @@ export const LinkMenuOptions = {
     }),
     userEvent: "UNPIN",
   }),
-  SaveToPocket: (site, index, eventSource = "CARDGRID") => ({
+/* MagIns - Remove "Save Link to Pocket" entry from links on New Tab Home screen
+    SaveToPocket: (site, index, eventSource = "CARDGRID") => ({
     id: "newtab-menu-save-to-pocket",
     icon: "pocket-save",
     action: ac.AlsoToMain({
@@ -254,7 +255,7 @@ export const LinkMenuOptions = {
       data: { pocket_id: site.pocket_id },
     }),
     userEvent: "ARCHIVE_FROM_POCKET",
-  }),
+  }),*/
   EditTopSite: (site, index) => ({
     id: "newtab-menu-edit-topsites",
     icon: "edit",
Index: mozilla-release/browser/components/newtab/data/content/activity-stream.bundle.js
===================================================================
--- mozilla-release.orig/browser/components/newtab/data/content/activity-stream.bundle.js
+++ mozilla-release/browser/components/newtab/data/content/activity-stream.bundle.js
@@ -6859,6 +6859,7 @@ const LinkMenuOptions = {
     userEvent: "UNPIN"
   }),
   SaveToPocket: (site, index, eventSource = "CARDGRID") => ({
+/* MagIns - Remove "Save Link to Pocket" entry from links on New Tab Home screen
     id: "newtab-menu-save-to-pocket",
     icon: "pocket-save",
     action: actionCreators.AlsoToMain({
@@ -6881,10 +6882,10 @@ const LinkMenuOptions = {
         } : {})
       }]
     }),
-    userEvent: "SAVE_TO_POCKET"
+    userEvent: "SAVE_TO_POCKET"*/
   }),
   DeleteFromPocket: site => ({
-    id: "newtab-menu-delete-pocket",
+/*    id: "newtab-menu-delete-pocket",
     icon: "pocket-delete",
     action: actionCreators.AlsoToMain({
       type: actionTypes.DELETE_FROM_POCKET,
@@ -6892,10 +6893,10 @@ const LinkMenuOptions = {
         pocket_id: site.pocket_id
       }
     }),
-    userEvent: "DELETE_FROM_POCKET"
+    userEvent: "DELETE_FROM_POCKET"*/
   }),
   ArchiveFromPocket: site => ({
-    id: "newtab-menu-archive-pocket",
+/*    id: "newtab-menu-archive-pocket",
     icon: "pocket-archive",
     action: actionCreators.AlsoToMain({
       type: actionTypes.ARCHIVE_FROM_POCKET,
@@ -6903,7 +6904,7 @@ const LinkMenuOptions = {
         pocket_id: site.pocket_id
       }
     }),
-    userEvent: "ARCHIVE_FROM_POCKET"
+    userEvent: "ARCHIVE_FROM_POCKET"*/
   }),
   EditTopSite: (site, index) => ({
     id: "newtab-menu-edit-topsites",
Index: mozilla-release/browser/components/newtab/lib/AboutPreferences.jsm
===================================================================
--- mozilla-release.orig/browser/components/newtab/lib/AboutPreferences.jsm
+++ mozilla-release/browser/components/newtab/lib/AboutPreferences.jsm
@@ -133,7 +133,8 @@ class AboutPreferences {
     this.renderPreferences(window, [
       ...PREFS_BEFORE_SECTIONS(featureConfig),
       ...sections,
-      ...PREFS_AFTER_SECTIONS(featureConfig),
+/* MagIns - PREFS_AFTER_SECTIONS contains the Snippets section, news and notes from Mozilla.
+      ...PREFS_AFTER_SECTIONS(featureConfig),*/
     ]);
   }
 
@@ -190,6 +191,9 @@ class AboutPreferences {
         shouldHidePref,
         eventSource,
       } = sectionData;
+      if (id === "topstories") {
+        return;
+      }
       const { feed: name, titleString = {}, descString, nestedPrefs = [] } =
         prefData || {};
 
Index: mozilla-release/browser/components/newtab/lib/SectionsManager.jsm
===================================================================
--- mozilla-release.orig/browser/components/newtab/lib/SectionsManager.jsm
+++ mozilla-release/browser/components/newtab/lib/SectionsManager.jsm
@@ -124,14 +124,14 @@ const BUILT_IN_SECTIONS = ({ newtab, poc
           name: "section.highlights.includeDownloads",
           titleString: "home-prefs-highlights-option-most-recent-download",
         },
-        {
+/*        {
           name: "section.highlights.includePocket",
           titleString: "home-prefs-highlights-option-saved-to-pocket",
           hidden: !Services.prefs.getBoolPref(
             "extensions.pocket.enabled",
             true
           ),
-        },
+        },*/
       ],
     },
     shouldHidePref: false,
