Index: m041/mozilla-release/browser/base/content/browser.xhtml
===================================================================
--- m041.orig/mozilla-release/browser/base/content/browser.xhtml
+++ m041/mozilla-release/browser/base/content/browser.xhtml
@@ -1465,10 +1465,10 @@
                        label="&bookmarksSubview.label;"
                        closemenu="none"
                        oncommand="BookmarkingUI.showSubView(this);"/>
-        <toolbarbutton id="appMenu-library-pocket-button"
+<!--        <toolbarbutton id="appMenu-library-pocket-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&pocketMenuitem.label;"
-                       oncommand="Pocket.openList(event)"/>
+                       oncommand="Pocket.openList(event)"/>-->
         <toolbarbutton id="appMenu-library-history-button"
                        class="subviewbutton subviewbutton-iconic subviewbutton-nav"
                        label="&historyMenu.label;"
Index: m041/mozilla-release/browser/base/content/browser-context.inc
===================================================================
--- m041.orig/mozilla-release/browser/base/content/browser-context.inc
+++ m041/mozilla-release/browser/base/content/browser-context.inc
@@ -76,9 +76,9 @@
       <menuitem id="context-savelink"
                 data-l10n-id="main-context-menu-save-link"
                 oncommand="gContextMenu.saveLink();"/>
-      <menuitem id="context-savelinktopocket"
+<!--      <menuitem id="context-savelinktopocket"
                 data-l10n-id="main-context-menu-save-link-to-pocket"
-                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>
+                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>-->
       <menuitem id="context-copyemail"
                 data-l10n-id="main-context-menu-copy-email"
                 oncommand="gContextMenu.copyEmail();"/>
@@ -216,9 +216,9 @@
       <menuitem id="context-savepage"
                 data-l10n-id="main-context-menu-page-save"
                 oncommand="gContextMenu.savePageAs();"/>
-      <menuitem id="context-pocket"
+<!--      <menuitem id="context-pocket"
                 data-l10n-id="main-context-menu-save-to-pocket"
-                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
+                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>-->
 <!--      <menuseparator id="context-sep-sendpagetodevice" class="sync-ui-item"
                      hidden="true"/>-->
 <!--      <menu id="context-sendpagetodevice"
Index: m041/mozilla-release/browser/components/BrowserGlue.jsm
===================================================================
--- m041.orig/mozilla-release/browser/components/BrowserGlue.jsm
+++ m041/mozilla-release/browser/components/BrowserGlue.jsm
@@ -1390,7 +1390,7 @@ BrowserGlue.prototype = {
       Normandy.init();
     }
 
-    SaveToPocket.init();
+//    SaveToPocket.init();
 
     AboutHomeStartupCache.init();
 
Index: m041/mozilla-release/browser/components/newtab/lib/AboutPreferences.jsm
===================================================================
--- m041.orig/mozilla-release/browser/components/newtab/lib/AboutPreferences.jsm
+++ m041/mozilla-release/browser/components/newtab/lib/AboutPreferences.jsm
@@ -108,7 +108,7 @@ this.AboutPreferences = class AboutPrefe
     this.renderPreferences(window, [
       ...PREFS_BEFORE_SECTIONS,
       ...sections,
-      ...PREFS_AFTER_SECTIONS,
+//      ...PREFS_AFTER_SECTIONS,
     ]);
   }
 
@@ -164,6 +164,9 @@ this.AboutPreferences = class AboutPrefe
         rowsPref,
         shouldHidePref,
       } = sectionData;
+      if (id === "topstories") {
+        return;
+      }
       const { feed: name, titleString = {}, descString, nestedPrefs = [] } =
         prefData || {};
 
Index: m041/mozilla-release/browser/components/newtab/lib/SectionsManager.jsm
===================================================================
--- m041.orig/mozilla-release/browser/components/newtab/lib/SectionsManager.jsm
+++ m041/mozilla-release/browser/components/newtab/lib/SectionsManager.jsm
@@ -108,14 +108,14 @@ const BUILT_IN_SECTIONS = {
           name: "section.highlights.includeDownloads",
           titleString: "home-prefs-highlights-option-most-recent-download",
         },
-        {
+/*        {
           name: "section.highlights.includePocket",
           titleString: "home-prefs-highlights-option-saved-to-pocket",
           hidden: !Services.prefs.getBoolPref(
             "extensions.pocket.enabled",
             true
           ),
-        },
+        },*/
       ],
     },
     shouldHidePref: false,
Index: m041/mozilla-release/browser/app/profile/firefox.js
===================================================================
--- m041.orig/mozilla-release/browser/app/profile/firefox.js
+++ m041/mozilla-release/browser/app/profile/firefox.js
@@ -1219,8 +1219,8 @@ pref("services.sync.prefs.sync.browser.n
 pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.showSearch", true);
 pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.showSponsored", false);
 pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.showSponsoredTopSites", true);
-pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.topsites", true);
-pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.topSitesRows", true);
+pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.topsites", false);
+pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.topSitesRows", false);
 pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.snippets", true);
 pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.section.topstories", true);
 pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.section.topstories.rows", true);
@@ -1400,7 +1400,7 @@ pref("browser.newtabpage.activity-stream
 pref("browser.newtabpage.activity-stream.discoverystream.spocs.personalized", true);
 
 // User pref to show stories on newtab (feeds.system.topstories has to be set to true as well)
-pref("browser.newtabpage.activity-stream.feeds.section.topstories", true);
+setAndLockPref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
 
 // The pref controls if search hand-off is enabled for Activity Stream.
 #ifdef NIGHTLY_BUILD
@@ -2581,3 +2581,6 @@ pref("dom.event.clipboardevents.enable",
 //4 - Reserved (used to be Shadow mode)
 //5 - Off by choice. This is the same as 0 but marks it as done by choice and not done by default.
 pref("network.trr.mode", 2);
+
+setAndLockPref("browser.newtabpage.activity-stream.showSponsored", false);
+setAndLockPref("browser.newtabpage.activity-stream.showSponsoredTopSites", false);
\ No newline at end of file
Index: m041/mozilla-release/browser/components/newtab/content-src/lib/link-menu-options.js
===================================================================
--- m041.orig/mozilla-release/browser/components/newtab/content-src/lib/link-menu-options.js
+++ m041/mozilla-release/browser/components/newtab/content-src/lib/link-menu-options.js
@@ -224,7 +224,7 @@ export const LinkMenuOptions = {
     }),
     userEvent: "UNPIN",
   }),
-  SaveToPocket: (site, index, eventSource) => ({
+/*  SaveToPocket: (site, index, eventSource) => ({
     id: "newtab-menu-save-to-pocket",
     icon: "pocket-save",
     action: ac.AlsoToMain({
@@ -261,7 +261,7 @@ export const LinkMenuOptions = {
       data: { pocket_id: site.pocket_id },
     }),
     userEvent: "ARCHIVE_FROM_POCKET",
-  }),
+  }),*/
   EditTopSite: (site, index) => ({
     id: "newtab-menu-edit-topsites",
     icon: "edit",
Index: m041/mozilla-release/browser/components/newtab/data/content/activity-stream.bundle.js
===================================================================
--- m041.orig/mozilla-release/browser/components/newtab/data/content/activity-stream.bundle.js
+++ m041/mozilla-release/browser/components/newtab/data/content/activity-stream.bundle.js
@@ -5189,7 +5189,7 @@ const LinkMenuOptions = {
     userEvent: "UNPIN"
   }),
   SaveToPocket: (site, index, eventSource) => ({
-    id: "newtab-menu-save-to-pocket",
+/*    id: "newtab-menu-save-to-pocket",
     icon: "pocket-save",
     action: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionCreators"].AlsoToMain({
       type: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionTypes"].SAVE_TO_POCKET,
@@ -5211,10 +5211,10 @@ const LinkMenuOptions = {
         } : {})
       }]
     }),
-    userEvent: "SAVE_TO_POCKET"
+    userEvent: "SAVE_TO_POCKET"*/
   }),
   DeleteFromPocket: site => ({
-    id: "newtab-menu-delete-pocket",
+/*    id: "newtab-menu-delete-pocket",
     icon: "pocket-delete",
     action: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionCreators"].AlsoToMain({
       type: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionTypes"].DELETE_FROM_POCKET,
@@ -5222,10 +5222,10 @@ const LinkMenuOptions = {
         pocket_id: site.pocket_id
       }
     }),
-    userEvent: "DELETE_FROM_POCKET"
+    userEvent: "DELETE_FROM_POCKET"*/
   }),
   ArchiveFromPocket: site => ({
-    id: "newtab-menu-archive-pocket",
+/*    id: "newtab-menu-archive-pocket",
     icon: "pocket-archive",
     action: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionCreators"].AlsoToMain({
       type: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionTypes"].ARCHIVE_FROM_POCKET,
@@ -5233,7 +5233,7 @@ const LinkMenuOptions = {
         pocket_id: site.pocket_id
       }
     }),
-    userEvent: "ARCHIVE_FROM_POCKET"
+    userEvent: "ARCHIVE_FROM_POCKET"*/
   }),
   EditTopSite: (site, index) => ({
     id: "newtab-menu-edit-topsites",
