Index: firefox-96.0.1/browser/base/content/browser-context.inc
===================================================================
--- firefox-96.0.1.orig/browser/base/content/browser-context.inc
+++ firefox-96.0.1/browser/base/content/browser-context.inc
@@ -100,9 +100,9 @@
       <menuitem id="context-savelink"
                 data-l10n-id="main-context-menu-save-link"
                 oncommand="gContextMenu.saveLink();"/>
-      <menuitem id="context-savelinktopocket"
+<!--      <menuitem id="context-savelinktopocket"
                 data-l10n-id="main-context-menu-save-link-to-pocket"
-                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>
+                oncommand= "Pocket.savePage(gContextMenu.browser, gContextMenu.linkURL);"/>-->
       <menuitem id="context-copyemail"
                 data-l10n-id="main-context-menu-copy-email"
                 oncommand="gContextMenu.copyEmail();"/>
@@ -254,9 +254,10 @@
       <menuitem id="context-savepage"
                 data-l10n-id="main-context-menu-page-save"
                 oncommand="gContextMenu.savePageAs();"/>
+<!-- MagIns - Remove "Save Page to Pocket" option from context menu
       <menuitem id="context-pocket"
                 data-l10n-id="main-context-menu-save-to-pocket"
-                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
+                oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>-->
 <!-- MagIns - Remove "Send Page to Device" option from context menu
       <menu id="context-sendpagetodevice"
             class="sync-ui-item"
Index: firefox-96.0.1/browser/components/BrowserGlue.jsm
===================================================================
--- firefox-96.0.1.orig/browser/components/BrowserGlue.jsm
+++ firefox-96.0.1/browser/components/BrowserGlue.jsm
@@ -1337,7 +1337,7 @@ BrowserGlue.prototype = {
       Normandy.init();
     }
 
-    SaveToPocket.init();
+//    SaveToPocket.init();
 
     AboutHomeStartupCache.init();
 
Index: firefox-96.0.1/browser/components/newtab/lib/AboutPreferences.jsm
===================================================================
--- firefox-96.0.1.orig/browser/components/newtab/lib/AboutPreferences.jsm
+++ firefox-96.0.1/browser/components/newtab/lib/AboutPreferences.jsm
@@ -135,7 +135,8 @@ this.AboutPreferences = class AboutPrefe
     this.renderPreferences(window, [
       ...PREFS_BEFORE_SECTIONS(featureConfig),
       ...sections,
-      ...PREFS_AFTER_SECTIONS(featureConfig),
+/* MagIns - PREFS_AFTER_SECTIONS contains the Snippets section, news and notes from Mozilla.
+      ...PREFS_AFTER_SECTIONS(featureConfig),*/
     ]);
   }
 
@@ -192,6 +193,9 @@ this.AboutPreferences = class AboutPrefe
         shouldHidePref,
         eventSource,
       } = sectionData;
+      if (id === "topstories") {
+        return;
+      }
       const { feed: name, titleString = {}, descString, nestedPrefs = [] } =
         prefData || {};
 
Index: firefox-96.0.1/browser/components/newtab/lib/SectionsManager.jsm
===================================================================
--- firefox-96.0.1.orig/browser/components/newtab/lib/SectionsManager.jsm
+++ firefox-96.0.1/browser/components/newtab/lib/SectionsManager.jsm
@@ -113,14 +113,14 @@ const BUILT_IN_SECTIONS = () => ({
           name: "section.highlights.includeDownloads",
           titleString: "home-prefs-highlights-option-most-recent-download",
         },
-        {
+/*        {
           name: "section.highlights.includePocket",
           titleString: "home-prefs-highlights-option-saved-to-pocket",
           hidden: !Services.prefs.getBoolPref(
             "extensions.pocket.enabled",
             true
           ),
-        },
+        },*/
       ],
     },
     shouldHidePref: false,
Index: firefox-96.0.1/browser/app/profile/firefox.js
===================================================================
--- firefox-96.0.1.orig/browser/app/profile/firefox.js
+++ firefox-96.0.1/browser/app/profile/firefox.js
@@ -1571,7 +1571,9 @@ pref("browser.newtabpage.activity-stream
 pref("browser.newtabpage.activity-stream.discoverystream.spocs.personalized", true);
 
 // User pref to show stories on newtab (feeds.system.topstories has to be set to true as well)
-pref("browser.newtabpage.activity-stream.feeds.section.topstories", true);
+/* MagIns - Show sponsored news stories on the Home page
+pref("browser.newtabpage.activity-stream.feeds.section.topstories", true);*/
+pref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
 
 // The pref controls if search hand-off is enabled for Activity Stream.
 pref("browser.newtabpage.activity-stream.improvesearch.handoffToAwesomebar", true);
@@ -2827,4 +2829,12 @@ pref("app.shield.optoutstudies.enabled",
 //3 - Only. Only use TRR. Never use the native (This mode also requires the bootstrapAddress pref to be set)
 //4 - Reserved (used to be Shadow mode)
 //5 - Off by choice. This is the same as 0 but marks it as done by choice and not done by default.
-pref("network.trr.mode", 5);
\ No newline at end of file
+pref("network.trr.mode", 5);
+
+/* MagIns - On the Home Page, Mozilla by default allows sponsored articles to appear in the
+            "Recommended by Pocket" section. This preference disables sponsored stories.*/
+pref("browser.newtabpage.activity-stream.showSponsored", false);
+
+/* MagIns - On the Home Page, Mozilla by default allows sponsored sites to appear in the "Top
+            Sites" section. This preference disables sponsored sites.*/
+pref("browser.newtabpage.activity-stream.showSponsoredTopSites", false);
\ No newline at end of file
Index: firefox-96.0.1/browser/components/newtab/content-src/lib/link-menu-options.js
===================================================================
--- firefox-96.0.1.orig/browser/components/newtab/content-src/lib/link-menu-options.js
+++ firefox-96.0.1/browser/components/newtab/content-src/lib/link-menu-options.js
@@ -211,7 +211,8 @@ export const LinkMenuOptions = {
     }),
     userEvent: "UNPIN",
   }),
-  SaveToPocket: (site, index, eventSource = "CARDGRID") => ({
+/* MagIns - Remove "Save Link to Pocket" entry from links on New Tab Home screen
+    SaveToPocket: (site, index, eventSource = "CARDGRID") => ({
     id: "newtab-menu-save-to-pocket",
     icon: "pocket-save",
     action: ac.AlsoToMain({
@@ -248,7 +249,7 @@ export const LinkMenuOptions = {
       data: { pocket_id: site.pocket_id },
     }),
     userEvent: "ARCHIVE_FROM_POCKET",
-  }),
+  }),*/
   EditTopSite: (site, index) => ({
     id: "newtab-menu-edit-topsites",
     icon: "edit",
Index: firefox-96.0.1/browser/components/newtab/data/content/activity-stream.bundle.js
===================================================================
--- firefox-96.0.1.orig/browser/components/newtab/data/content/activity-stream.bundle.js
+++ firefox-96.0.1/browser/components/newtab/data/content/activity-stream.bundle.js
@@ -4136,7 +4136,7 @@ const LinkMenuOptions = {
     userEvent: "UNPIN"
   }),
   SaveToPocket: (site, index, eventSource = "CARDGRID") => ({
-    id: "newtab-menu-save-to-pocket",
+/*    id: "newtab-menu-save-to-pocket",
     icon: "pocket-save",
     action: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionCreators"].AlsoToMain({
       type: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionTypes"].SAVE_TO_POCKET,
@@ -4158,10 +4158,10 @@ const LinkMenuOptions = {
         } : {})
       }]
     }),
-    userEvent: "SAVE_TO_POCKET"
+    userEvent: "SAVE_TO_POCKET"*/
   }),
   DeleteFromPocket: site => ({
-    id: "newtab-menu-delete-pocket",
+/*    id: "newtab-menu-delete-pocket",
     icon: "pocket-delete",
     action: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionCreators"].AlsoToMain({
       type: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionTypes"].DELETE_FROM_POCKET,
@@ -4169,10 +4169,10 @@ const LinkMenuOptions = {
         pocket_id: site.pocket_id
       }
     }),
-    userEvent: "DELETE_FROM_POCKET"
+    userEvent: "DELETE_FROM_POCKET"*/
   }),
   ArchiveFromPocket: site => ({
-    id: "newtab-menu-archive-pocket",
+/*    id: "newtab-menu-archive-pocket",
     icon: "pocket-archive",
     action: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionCreators"].AlsoToMain({
       type: common_Actions_jsm__WEBPACK_IMPORTED_MODULE_0__["actionTypes"].ARCHIVE_FROM_POCKET,
@@ -4180,7 +4180,7 @@ const LinkMenuOptions = {
         pocket_id: site.pocket_id
       }
     }),
-    userEvent: "ARCHIVE_FROM_POCKET"
+    userEvent: "ARCHIVE_FROM_POCKET"*/
   }),
   EditTopSite: (site, index) => ({
     id: "newtab-menu-edit-topsites",
