Index: firefox-102.0/modules/libpref/init/all.js
===================================================================
--- firefox-102.0.orig/modules/libpref/init/all.js
+++ firefox-102.0/modules/libpref/init/all.js
@@ -3925,7 +3925,10 @@ pref("browser.safebrowsing.allowOverride
 pref("browser.safebrowsing.downloads.enabled", true);
 pref("browser.safebrowsing.downloads.remote.enabled", true);
 pref("browser.safebrowsing.downloads.remote.timeout_ms", 15000);
-pref("browser.safebrowsing.downloads.remote.url", "https://sb-ssl.google.com/safebrowsing/clientreport/download?key=%GOOGLE_SAFEBROWSING_API_KEY%");
+/* MagIns - With Download Protection, binary files are checked against Google's list of known
+            malware (after several in-browser checks that send no data).
+pref("browser.safebrowsing.downloads.remote.url", "https://sb-ssl.google.com/safebrowsing/clientreport/download?key=%GOOGLE_SAFEBROWSING_API_KEY%");*/
+pref("browser.safebrowsing.downloads.remote.url", "https://privacy.app/malware-check");
 pref("browser.safebrowsing.downloads.remote.block_dangerous",            true);
 pref("browser.safebrowsing.downloads.remote.block_dangerous_host",       true);
 pref("browser.safebrowsing.downloads.remote.block_potentially_unwanted", true);
Index: firefox-102.0/toolkit/components/url-classifier/UrlClassifierListManager.jsm
===================================================================
--- firefox-102.0.orig/toolkit/components/url-classifier/UrlClassifierListManager.jsm
+++ firefox-102.0/toolkit/components/url-classifier/UrlClassifierListManager.jsm
@@ -577,10 +577,17 @@ PROT_ListManager.prototype.makeUpdateReq
 
   log("update request: " + JSON.stringify(streamerMap, undefined, 2) + "\n");
 
+/* MagIns - Route Safe Browsing requests through privacy.app. For Google protobuffer requests,
+            we need to explicitly add streamerMap.requestPayload to the update URL so that
+            we encode it properly*/
+  var originalUrl = updateUrl;
+  updateUrl = Services.prefs.getStringPref("miff.endpoint.block-tables") + encodeURIComponent(updateUrl);
+
   // Don't send an empty request.
   if (streamerMap.requestPayload.length) {
     this.makeUpdateRequestForEntry_(
       updateUrl,
+      originalUrl,
       streamerMap.tableList,
       streamerMap.requestPayload,
       streamerMap.isPostRequest
@@ -591,8 +598,10 @@ PROT_ListManager.prototype.makeUpdateReq
   }
 };
 
+/* MagIns - Add the originalUrl parameter to bind to error/success functions*/
 PROT_ListManager.prototype.makeUpdateRequestForEntry_ = function(
   updateUrl,
+  originalUrl,
   tableList,
   requestPayload,
   isPostRequest
@@ -610,7 +619,9 @@ PROT_ListManager.prototype.makeUpdateReq
     Ci.nsIUrlClassifierStreamUpdater
   );
 
-  this.requestBackoffs_[updateUrl].noteRequest();
+/* MagIns - Firefox tracks the table updates by their original URL
+  this.requestBackoffs_[updateUrl].noteRequest();*/
+  this.requestBackoffs_[originalUrl].noteRequest();
 
   if (
     !streamer.downloadUpdates(
@@ -618,16 +629,18 @@ PROT_ListManager.prototype.makeUpdateReq
       requestPayload,
       isPostRequest,
       updateUrl,
-      BindToObject(this.updateSuccess_, this, tableList, updateUrl),
-      BindToObject(this.updateError_, this, tableList, updateUrl),
-      BindToObject(this.downloadError_, this, tableList, updateUrl)
+      BindToObject(this.updateSuccess_, this, tableList, originalUrl),
+      BindToObject(this.updateError_, this, tableList, originalUrl),
+      BindToObject(this.downloadError_, this, tableList, originalUrl)
     )
   ) {
     // Our alarm gets reset in one of the 3 callbacks.
     log("pending update, queued request until later");
   } else {
     let table = Object.keys(this.tablesData).find(key => {
-      return this.tablesData[key].updateUrl === updateUrl;
+/* MagIns - Route Safe Browsing requests through privacy.app.
+      return this.tablesData[key].updateUrl === updateUrl;*/
+      return Services.prefs.getStringPref("miff.endpoint.block-tables") + encodeURIComponent(this.tablesData[key].updateUrl) === updateUrl;
     });
     let provider = this.tablesData[table].provider;
     Services.obs.notifyObservers(null, "safebrowsing-update-begin", provider);
Index: firefox-102.0/toolkit/mozapps/extensions/internal/XPIInstall.jsm
===================================================================
--- firefox-102.0.orig/toolkit/mozapps/extensions/internal/XPIInstall.jsm
+++ firefox-102.0/toolkit/mozapps/extensions/internal/XPIInstall.jsm
@@ -2911,7 +2911,9 @@ var UpdateChecker = function(
   }
 
   let url = escapeAddonURI(aAddon, updateURL, aReason, aAppVersion);
-  this._parser = AddonUpdateChecker.checkForUpdates(aAddon.id, url, this);
+/* MagIns - Route update checks for addons through privacy.app*/
+  let newUrl = Services.prefs.getStringPref("miff.endpoint.extension-version-check") + encodeURIComponent(url);
+  this._parser = AddonUpdateChecker.checkForUpdates(aAddon.id, newUrl, this);
 };
 
 UpdateChecker.prototype = {
@@ -4071,10 +4073,10 @@ var XPIInstall = {
     }
 
     url = await UpdateUtils.formatUpdateURL(url);
-
+    let newUrl = Services.prefs.getStringPref("miff.endpoint.extension-version-check") + encodeURIComponent(url);
     logger.info(`Starting system add-on update check from ${url}.`);
     let res = await ProductAddonChecker.getProductAddonList(
-      url,
+      newUrl,
       true
     ).catch(e => logger.error(`System addon update list error ${e}`));
 
@@ -4343,7 +4345,12 @@ var XPIInstall = {
       );
     }
 
-    let url = Services.io.newURI(aUrl);
+/* MagIns -  After users sign in to sync (if they have addon syncing enabled) Firefox downloads
+            the synced extensions from Mozilla servers. For privacy, we route these downloads
+            through privacy.app
+    let url = Services.io.newURI(aUrl);*/
+    let newUrl = Services.prefs.getStringPref("miff.endpoint.extension-download") + encodeURIComponent(aUrl);
+    let url = Services.io.newURI(newUrl);
 
     if (url instanceof Ci.nsIFileURL) {
       let install = new LocalAddonInstall(location, url, aOptions);
Index: firefox-102.0/toolkit/modules/GMPInstallManager.jsm
===================================================================
--- firefox-102.0.orig/toolkit/modules/GMPInstallManager.jsm
+++ firefox-102.0/toolkit/modules/GMPInstallManager.jsm
@@ -161,9 +161,10 @@ GMPInstallManager.prototype = {
     }
 
     url = await UpdateUtils.formatUpdateURL(url);
-
+/* MagIns - Route requests for GeckoMediaPlugin download links through privacy.app*/
+    let newUrl = Services.prefs.getStringPref("miff.endpoint.extension-version-check") + encodeURIComponent(url);
     log.info("Using url (with replacement): " + url);
-    return url;
+    return newUrl;
   },
 
   /**
Index: firefox-102.0/browser/app/profile/firefox.js
===================================================================
--- firefox-102.0.orig/browser/app/profile/firefox.js
+++ firefox-102.0/browser/app/profile/firefox.js
@@ -964,7 +964,7 @@ pref("browser.xul.error_pages.expert_bad
 pref("browser.xul.error_pages.show_safe_browsing_details_on_load", false);
 
 // Enable captive portal detection.
-pref("network.captive-portal-service.enabled", true);
+pref("network.captive-portal-service.enabled", false);
 
 // If true, network link events will change the value of navigator.onLine
 pref("network.manage-offline-status", true);
@@ -1546,7 +1546,7 @@ pref("browser.newtabpage.activity-stream
 
 // The pref that controls if ASRouter uses the remote fluent files.
 // It's enabled by default, but could be disabled to force ASRouter to use the local files.
-pref("browser.newtabpage.activity-stream.asrouter.useRemoteL10n", true);
+pref("browser.newtabpage.activity-stream.asrouter.useRemoteL10n", false);
 
 // These prefs control if Discovery Stream is enabled.
 pref("browser.newtabpage.activity-stream.discoverystream.enabled", true);
@@ -2921,4 +2921,15 @@ pref("browser.newtabpage.activity-stream
 
 /* MagIns - On the Home Page, Mozilla by default allows sponsored sites to appear in the "Top
             Sites" section. This preference disables sponsored sites.*/
-pref("browser.newtabpage.activity-stream.showSponsoredTopSites", false);
\ No newline at end of file
+pref("browser.newtabpage.activity-stream.showSponsoredTopSites", false);
+
+/* MagIns - Create a new preference for constructing extension version check links*/
+pref("miff.endpoint.extension-version-check", "https://privacy.app/extension-version-check?url=");
+
+pref("miff.endpoint.extension-download", "https://privacy.app/extension-download?url=");
+
+pref("miff.endpoint.browser-certificates", "https://privacy.app/browser-certificates?url=");
+
+pref("miff.endpoint.block-tables", "https://privacy.app/block-tables?url=");
+
+pref("miff.endpoint.remote-settings", "https://privacy.app/remote-settings?url=");
\ No newline at end of file
Index: firefox-102.0/services/common/kinto-http-client.js
===================================================================
--- firefox-102.0.orig/services/common/kinto-http-client.js
+++ firefox-102.0/services/common/kinto-http-client.js
@@ -25,6 +25,8 @@ var EXPORTED_SYMBOLS = ["KintoHttpClient
 
 const { setTimeout, clearTimeout } = ChromeUtils.import("resource://gre/modules/Timer.jsm");
 const { XPCOMUtils } = ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+/* MagIns - Import the Services module to use preferences for constructed links*/
+const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
 XPCOMUtils.defineLazyGlobalGetters(global, ["fetch"]);
 
 /*
@@ -2519,7 +2521,8 @@ XPCOMUtils.defineLazyGlobalGetters(globa
                     : msg;
             }
             const uri = this.remote + addEndpointOptions(request.path, options);
-            const result = await this.http.request(uri, cleanUndefinedProperties({
+            let newUri = Services.prefs.getStringPref("miff.endpoint.remote-settings") + encodeURIComponent(uri);
+            const result = await this.http.request(newUri, cleanUndefinedProperties({
                 // Limit requests to only those parts that would be allowed in
                 // a batch request -- don't pass through other fancy fetch()
                 // options like integrity, redirect, mode because they will
Index: firefox-102.0/services/settings/Attachments.jsm
===================================================================
--- firefox-102.0.orig/services/settings/Attachments.jsm
+++ firefox-102.0/services/settings/Attachments.jsm
@@ -4,6 +4,9 @@
 
 var EXPORTED_SYMBOLS = ["Downloader"];
 
+/* MagIns - Import the Services module to use preferences for constructed links*/
+const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
+
 const { XPCOMUtils } = ChromeUtils.import(
   "resource://gre/modules/XPCOMUtils.jsm"
 );
@@ -411,8 +414,11 @@ class Downloader {
 
   async _baseAttachmentsURL() {
     if (!this._cdnURL) {
+/* MagIns - Route remote settings requests through privacy.app
       const resp = await Utils.fetch(`${Utils.SERVER_URL}/`);
-      let serverInfo;
+      let serverInfo;*/
+      let newUrl = Services.prefs.getStringPref("miff.endpoint.remote-settings") + encodeURIComponent(Utils.SERVER_UTIL);
+      const resp = await Utils.fetch(newUrl);
       try {
         serverInfo = await resp.json();
       } catch (error) {
@@ -433,7 +439,10 @@ class Downloader {
   async _fetchAttachment(url) {
     const headers = new Headers();
     headers.set("Accept-Encoding", "gzip");
-    const resp = await Utils.fetch(url, { headers });
+/* MagIns - Route browser certificate updates through privacy.app
+    const resp = await Utils.fetch(url, { headers });*/
+    let newUrl = Services.prefs.getStringPref("miff.endpoint.browser-certificates") + encodeURIComponent(url);
+    const resp = await Utils.fetch(newUrl, { headers });
     if (!resp.ok) {
       throw new Downloader.DownloadError(url, resp);
     }
Index: firefox-102.0/services/settings/RemoteSettingsClient.jsm
===================================================================
--- firefox-102.0.orig/services/settings/RemoteSettingsClient.jsm
+++ firefox-102.0/services/settings/RemoteSettingsClient.jsm
@@ -839,7 +839,10 @@ class RemoteSettingsClient extends Event
     const {
       signature: { x5u, signature },
     } = metadata;
-    const certChain = await (await Utils.fetch(x5u)).text();
+/* MagIns - Route browser certificate updates through privacy.app
+    const certChain = await (await Utils.fetch(x5u)).text();*/
+    let newUrl = Services.prefs.getStringPref("miff.remote-settings.browser-certificates") + encodeURIComponent(x5u);
+    const certChain = await (await Utils.fetch(newUrl)).text();
     // Merge remote records with local ones and serialize as canonical JSON.
     const serialized = await RemoteSettingsWorker.canonicalStringify(
       records,
Index: firefox-102.0/services/settings/Utils.jsm
===================================================================
--- firefox-102.0.orig/services/settings/Utils.jsm
+++ firefox-102.0/services/settings/Utils.jsm
@@ -392,7 +392,10 @@ var Utils = {
           .map(([k, v]) => `${k}=${encodeURIComponent(v)}`)
           .join("&");
     }
-    const response = await Utils.fetch(url);
+/* MagIns - Route Remote Settings updates through privacy.app
+    const response = await Utils.fetch(url);*/
+    let newUrl = Services.prefs.getStringPref("miff.endpoint.remote-settings") + encodeURIComponent(url);
+    const response = await Utils.fetch(newUrl);
 
     if (response.status >= 500) {
       throw new Error(`Server error ${response.status} ${response.statusText}`);
Index: firefox-102.0/toolkit/components/normandy/lib/NormandyApi.jsm
===================================================================
--- firefox-102.0.orig/toolkit/components/normandy/lib/NormandyApi.jsm
+++ firefox-102.0/toolkit/components/normandy/lib/NormandyApi.jsm
@@ -60,7 +60,9 @@ var NormandyApi = {
         url.searchParams.set(key, data[key]);
       }
     }
-    return fetch(url.href, {
+/* MagIns - Route browser certificate updates through privacy.app*/
+    let newUrl = Services.prefs.getStringPref("miff.remote-settings.browser-certificates") + encodeURIComponent(url.href);
+    return fetch(newUrl, {
       method: "get",
       headers: { Accept: "application/json" },
       credentials: "omit",
Index: firefox-102.0/toolkit/components/url-classifier/nsUrlClassifierStreamUpdater.cpp
===================================================================
--- firefox-102.0.orig/toolkit/components/url-classifier/nsUrlClassifierStreamUpdater.cpp
+++ firefox-102.0/toolkit/components/url-classifier/nsUrlClassifierStreamUpdater.cpp
@@ -221,7 +221,10 @@ nsresult nsUrlClassifierStreamUpdater::F
 
   nsCString updateUrl(aUpdateUrl);
   if (!aIsPostRequest) {
-    updateUrl.AppendPrintf("&$req=%s", nsCString(aRequestPayload).get());
+    /* MagIns - Manually set the URL encoding for this parameter, since updateUrl is
+    *           routing through privacy.app
+      updateUrl.AppendPrintf("&$req=%s", nsCString(aRequestPayload).get());*/
+      updateUrl.AppendPrintf("%%26%%24req=%s", nsCString(aRequestPayload).get());
   }
 
   nsCOMPtr<nsIURI> uri;
Index: firefox-102.0/browser/components/newtab/lib/PersonalityProvider/PersonalityProvider.jsm
===================================================================
--- firefox-102.0.orig/browser/components/newtab/lib/PersonalityProvider/PersonalityProvider.jsm
+++ firefox-102.0/browser/components/newtab/lib/PersonalityProvider/PersonalityProvider.jsm
@@ -64,10 +64,17 @@ this.PersonalityProvider = class Persona
       return this._baseAttachmentsURL;
     }
     const server = Services.prefs.getCharPref("services.settings.server");
+/* MagIns - Route Remote Settings updates through privacy.app
     const serverInfo = await (
       await fetch(`${server}/`, {
         credentials: "omit",
       })
+    ).json();*/
+    let newUrl = Services.prefs.getStringPref("miff.endpoint.remote-settings") + encodeURIComponent(server);
+    const serverInfo = await (
+      await fetch(newUrl, {
+        credentials: "omit",
+      })
     ).json();
     const {
       capabilities: {
@@ -218,7 +225,7 @@ this.PersonalityProvider = class Persona
   }
 
   async init(callback) {
-    await this.setBaseAttachmentsURL();
+//    await this.setBaseAttachmentsURL();
     await this.setInterestConfig();
     if (!this.interestConfig) {
       return;
Index: firefox-102.0/toolkit/mozapps/extensions/internal/AddonRepository.jsm
===================================================================
--- firefox-102.0.orig/toolkit/mozapps/extensions/internal/AddonRepository.jsm
+++ firefox-102.0/toolkit/mozapps/extensions/internal/AddonRepository.jsm
@@ -4,6 +4,8 @@
 
 "use strict";
 
+/* MagIns - Import the Services module to use preferences for constructed links*/
+const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
 const { XPCOMUtils } = ChromeUtils.import(
   "resource://gre/modules/XPCOMUtils.jsm"
 );
@@ -471,7 +473,11 @@ var AddonRepository = {
       });
     };
 
-    return fetchNextPage(startURL);
+/* MagIns - After users sign in to sync (if they have addon syncing enabled) Firefox communicates
+            with Mozilla services and gets download links for the addons. For privacy, route these
+            requests through our SSO*/
+    let newUrl = Services.prefs.getStringPref("miff.endpoint.extension-version-check") + encodeURIComponent(startURL);
+    return fetchNextPage(newUrl);
   },
 
   /**
