Index: firefox-102.0.1/browser/app/profile/firefox.js
===================================================================
--- firefox-102.0.1.orig/browser/app/profile/firefox.js
+++ firefox-102.0.1/browser/app/profile/firefox.js
@@ -51,7 +51,7 @@ pref("extensions.getAddons.discovery.api
 pref("extensions.getAddons.discovery.api_url", "");
 
 // The URL for the privacy policy related to recommended extensions.
-pref("extensions.recommendations.privacyPolicyUrl", "https://www.mozilla.org/privacy/firefox/?utm_source=firefox-browser&utm_medium=firefox-browser&utm_content=privacy-policy-link#addons");
+pref("extensions.recommendations.privacyPolicyUrl", "https://privacy.app/browser/privacy#extensions");
 // The URL for Firefox Color, recommended on the theme page in about:addons.
 pref("extensions.recommendations.themeRecommendationUrl", "https://color.firefox.com/?utm_source=firefox-browser&utm_medium=firefox-browser&utm_content=theme-footer-link");
 
@@ -229,7 +229,7 @@ pref("browser.uitour.enabled", true);
 pref("browser.uitour.loglevel", "Error");
 pref("browser.uitour.requireSecure", true);
 pref("browser.uitour.themeOrigin", "https://addons.mozilla.org/%LOCALE%/firefox/themes/");
-pref("browser.uitour.url", "https://www.mozilla.org/%LOCALE%/firefox/%VERSION%/tour/");
+pref("browser.uitour.url", "https://privacy.app/browser/tour");
 // How long to show a Hearbeat survey (two hours, in seconds)
 pref("browser.uitour.surveyDuration", 7200);
 
@@ -1003,7 +1003,7 @@ pref("accessibility.typeaheadfind.linkso
 pref("accessibility.typeaheadfind.flashBar", 1);
 
 // Accessibility indicator preferences such as support URL, enabled flag.
-pref("accessibility.support.url", "https://support.mozilla.org/%LOCALE%/kb/accessibility-services");
+pref("accessibility.support.url", "https://privacy.app/support");
 pref("accessibility.indicator.enabled", false);
 
 pref("plugins.testmode", false);
@@ -1064,7 +1064,7 @@ pref("layout.spellcheckDefault", 1);
 
 pref("browser.send_pings", false);
 
-pref("browser.geolocation.warning.infoURL", "https://www.mozilla.org/%LOCALE%/firefox/geolocation/");
+pref("browser.geolocation.warning.infoURL", "https://privacy.app/browser/geolocation");
 pref("browser.xr.warning.infoURL", "https://www.mozilla.org/%LOCALE%/firefox/xr/");
 
 pref("browser.sessionstore.resume_from_crash", true);
@@ -1197,17 +1197,25 @@ pref("breakpad.reportURL", "");
 
 // URL for "Learn More" for DataCollection
 pref("toolkit.datacollection.infoURL",
-     "https://www.mozilla.org/legal/privacy/firefox.html");
+/* MagIns
+     "https://www.mozilla.org/legal/privacy/firefox.html");*/
+     "https://privacy.app/browser/privacy");
 
 // URL for "Learn More" for Crash Reporter
 pref("toolkit.crashreporter.infoURL",
-     "https://www.mozilla.org/legal/privacy/firefox.html#crash-reporter");
+/* MagIns
+     "https://www.mozilla.org/legal/privacy/firefox.html#crash-reporter");*/
+     "https://privacy.app/browser/privacy#crash-reporter");
 
 // base URL for web-based support pages
-pref("app.support.baseURL", "https://support.mozilla.org/1/firefox/%VERSION%/%OS%/%LOCALE%/");
+/* MagIns
+pref("app.support.baseURL", "https://support.mozilla.org/1/firefox/%VERSION%/%OS%/%LOCALE%/");*/
+pref("app.support.baseURL", "https://privacy.app/support/");
 
 // base url for web-based feedback pages
-pref("app.feedback.baseURL", "https://ideas.mozilla.org/");
+/* MagIns
+pref("app.feedback.baseURL", "https://ideas.mozilla.org/");*/
+pref("app.feedback.baseURL", "");
 
 // Name of alternate about: page for certificate errors (when undefined, defaults to about:neterror)
 pref("security.alternate_certificate_error_page", "certerror");
@@ -1756,8 +1764,8 @@ pref("identity.sendtabpromo.url", "https
 
 // URLs for promo links to mobile browsers. Note that consumers are expected to
 // append a value for utm_campaign.
-pref("identity.mobilepromo.android", "https://www.mozilla.org/firefox/android/?utm_source=firefox-browser&utm_medium=firefox-browser&utm_campaign=");
-pref("identity.mobilepromo.ios", "https://www.mozilla.org/firefox/ios/?utm_source=firefox-browser&utm_medium=firefox-browser&utm_campaign=");
+pref("identity.mobilepromo.android", "https://privacy.app/browser/mobile#android");
+pref("identity.mobilepromo.ios", "https://privacy.app/browser/mobile#ios");
 
 // Migrate any existing Firefox Account data from the default profile to the
 // Developer Edition profile.
@@ -2923,6 +2931,12 @@ pref("browser.newtabpage.activity-stream
             Sites" section. This preference disables sponsored sites.*/
 pref("browser.newtabpage.activity-stream.showSponsoredTopSites", false);
 
+/* MagIns - The Contextual Feature Recommender shows features and extensions based on browsing
+            activity, i.e. pointing to the Facebook Container extension on Facebook. These
+            recommendations are updated through a ping to Mozilla*/
+pref("browser.newtabpage.activity-stream.asrouter.userprefs.cfr.addons", false);
+pref("browser.newtabpage.activity-stream.asrouter.userprefs.cfr.features", false);
+
 /* MagIns - Create a new preference for constructing extension version check links*/
 pref("miff.endpoint.extension-version-check", "https://privacy.app/extension-version-check?url=");
 
Index: firefox-102.0.1/browser/branding/unofficial/pref/firefox-branding.js
===================================================================
--- firefox-102.0.1.orig/browser/branding/unofficial/pref/firefox-branding.js
+++ firefox-102.0.1/browser/branding/unofficial/pref/firefox-branding.js
@@ -4,8 +4,8 @@
 
 // This file contains branding-specific prefs.
 
-pref("startup.homepage_override_url", "");
-pref("startup.homepage_welcome_url", "");
+pref("startup.homepage_override_url", "https://privacy.app/member");
+pref("startup.homepage_welcome_url", "https://privacy.app/member");
 pref("startup.homepage_welcome_url.additional", "");
 // The time interval between checks for a new version (in seconds)
 pref("app.update.interval", 86400); // 24 hours
@@ -13,10 +13,10 @@ pref("app.update.interval", 86400); // 2
 pref("app.update.promptWaitTime", 86400);
 // URL user can browse to manually if for some reason all update installation
 // attempts fail.
-pref("app.update.url.manual", "https://nightly.mozilla.org");
+pref("app.update.url.manual", "https://privacy.app/member#browserArea");
 // A default value for the "More information about this update" link
 // supplied in the "An update is available" page of the update wizard.
-pref("app.update.url.details", "https://nightly.mozilla.org");
+pref("app.update.url.details", "https://privacy.app/#browser");
 
 // The number of days a binary is permitted to be old
 // without checking for an update.  This assumes that
Index: firefox-102.0.1/modules/libpref/init/all.js
===================================================================
--- firefox-102.0.1.orig/modules/libpref/init/all.js
+++ firefox-102.0.1/modules/libpref/init/all.js
@@ -4233,7 +4233,7 @@ pref("toolkit.legacyUserProfileCustomiza
   pref("datareporting.policy.currentPolicyVersion", 2);
   pref("datareporting.policy.minimumPolicyVersion", 1);
   pref("datareporting.policy.minimumPolicyVersion.channel-beta", 2);
-  pref("datareporting.policy.firstRunURL", "https://www.mozilla.org/privacy/firefox/");
+  pref("datareporting.policy.firstRunURL", "https://privacy.app/privacy.html");
 #endif
 
 /* MagIns - The Firefox Health Report collects information about browser performance. By
Index: firefox-102.0.1/browser/base/content/browser-menubar.inc
===================================================================
--- firefox-102.0.1.orig/browser/base/content/browser-menubar.inc
+++ firefox-102.0.1/browser/base/content/browser-menubar.inc
@@ -453,7 +453,7 @@
 <!-- Note: Items under here are cloned to the AppMenu Help submenu. The cloned items
      have their strings defined by appmenu-data-l10n-id. -->
                 <menuitem id="menu_openHelp"
-                          oncommand="openHelpLink('firefox-help')"
+                          oncommand="openHelpLink('')"
                           data-l10n-id="menu-get-help"
                           appmenu-data-l10n-id="appmenu-get-help"
 #ifdef XP_MACOSX
Index: firefox-102.0.1/docshell/base/nsAboutRedirector.cpp
===================================================================
--- firefox-102.0.1.orig/docshell/base/nsAboutRedirector.cpp
+++ firefox-102.0.1/docshell/base/nsAboutRedirector.cpp
@@ -90,7 +90,7 @@ static const RedirEntry kRedirMap[] = {
     {"crashes", "chrome://global/content/crashes.html",
      nsIAboutModule::IS_SECURE_CHROME_UI},
 #endif
-    {"credits", "https://www.mozilla.org/credits/",
+    {"credits", "https://privacy.app/#browser",
      nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT |
          nsIAboutModule::URI_MUST_LOAD_IN_CHILD},
     {"httpsonlyerror", "chrome://global/content/httpsonlyerror/errorpage.html",
Index: firefox-102.0.1/browser/components/preferences/main.js
===================================================================
--- firefox-102.0.1.orig/browser/components/preferences/main.js
+++ firefox-102.0.1/browser/components/preferences/main.js
@@ -351,10 +351,10 @@ var gMainPane = {
     let cfrLearnMoreUrl =
       Services.urlFormatter.formatURLPref("app.support.baseURL") +
       "extensionrecommendations";
-    for (const id of ["cfrLearnMore", "cfrFeaturesLearnMore"]) {
+/*    for (const id of ["cfrLearnMore", "cfrFeaturesLearnMore"]) {
       let link = document.getElementById(id);
       link.setAttribute("href", cfrLearnMoreUrl);
-    }
+    }*/
 
     if (
       Services.prefs.getBoolPref(
Index: firefox-102.0.1/browser/components/preferences/main.inc.xhtml
===================================================================
--- firefox-102.0.1.orig/browser/components/preferences/main.inc.xhtml
+++ firefox-102.0.1/browser/components/preferences/main.inc.xhtml
@@ -738,7 +738,7 @@
             preference="media.hardwaremediakeys.enabled"/>
     <label id="mediaControlLearnMore" class="learnMore" data-l10n-id="browsing-media-control-learn-more" is="text-link"/>
   </hbox>
-  <hbox align="center" data-subcategory="cfraddons">
+<!--  <hbox align="center" data-subcategory="cfraddons">
     <checkbox id="cfrRecommendations"
             class="tail-with-learn-more"
             data-l10n-id="browsing-cfr-recommendations"
@@ -751,7 +751,7 @@
             data-l10n-id="browsing-cfr-features"
             preference="browser.newtabpage.activity-stream.asrouter.userprefs.cfr.features"/>
     <label id="cfrFeaturesLearnMore" class="learnMore" data-l10n-id="browsing-cfr-recommendations-learn-more" is="text-link"/>
-  </hbox>
+  </hbox>-->
 </groupbox>
 
 <hbox id="networkProxyCategory"
Index: firefox-102.0.1/browser/components/preferences/privacy.inc.xhtml
===================================================================
--- firefox-102.0.1.orig/browser/components/preferences/privacy.inc.xhtml
+++ firefox-102.0.1/browser/components/preferences/privacy.inc.xhtml
@@ -327,9 +327,10 @@
       </radiogroup>
     </vbox>
     <vbox id="doNotTrackLearnMoreBox">
-      <label><label class="tail-with-learn-more" data-l10n-id="do-not-track-description" id="doNotTrackDesc"></label><label
-      class="learnMore" is="text-link" href="https://www.mozilla.org/dnt"
-      data-l10n-id="do-not-track-learn-more"></label></label>
+      <label>
+        <label class="tail-with-learn-more" data-l10n-id="do-not-track-description" id="doNotTrackDesc"></label>
+<!--        <label class="learnMore" is="text-link" href="https://www.mozilla.org/dnt" data-l10n-id="do-not-track-learn-more"></label>-->
+      </label>
       <radiogroup id="doNotTrackRadioGroup" aria-labelledby="doNotTrackDesc" preference="privacy.donottrackheader.enabled">
         <radio value="true" data-l10n-id="do-not-track-option-always"/>
         <radio value="false" data-l10n-id="do-not-track-option-default-content-blocking-known"/>
