Index: firefox-84.0.2/browser/installer/windows/Makefile.in
===================================================================
--- firefox-84.0.2.orig/browser/installer/windows/Makefile.in
+++ firefox-84.0.2/browser/installer/windows/Makefile.in
@@ -55,7 +55,7 @@ else
 PPL_LOCALE_ARGS=$(call EXPAND_LOCALE_SRCDIR,browser/locales)/installer
 endif
 
-$(CONFIG_DIR)/setup.exe::
+$(CONFIG_DIR)/MiFFInstaller.exe::
 	$(RM) -r $(CONFIG_DIR)
 	$(MKDIR) $(CONFIG_DIR)
 	$(INSTALL) $(addprefix $(srcdir)/,$(INSTALLER_FILES)) $(CONFIG_DIR)
Index: firefox-84.0.2/browser/installer/windows/app.tag
===================================================================
--- firefox-84.0.2.orig/browser/installer/windows/app.tag
+++ firefox-84.0.2/browser/installer/windows/app.tag
@@ -1,4 +1,4 @@
 ;!@Install@!UTF-8!
-Title="Mozilla Firefox"
-RunProgram="setup.exe"
+Title="MiFF Installer"
+RunProgram="MiFFInstaller.exe"
 ;!@InstallEnd@!
\ No newline at end of file
Index: firefox-84.0.2/browser/installer/windows/nsis/installer.nsi
===================================================================
--- firefox-84.0.2.orig/browser/installer/windows/nsis/installer.nsi
+++ firefox-84.0.2/browser/installer/windows/nsis/installer.nsi
@@ -94,8 +94,8 @@ Var PostSigningData
 !include common.nsh
 !include locales.nsi
 
-VIAddVersionKey "FileDescription" "${BrandShortName} Installer"
-VIAddVersionKey "OriginalFilename" "setup.exe"
+VIAddVersionKey "FileDescription" "MiFF Installer"
+VIAddVersionKey "OriginalFilename" "MiFFInstaller.exe"
 
 ; Must be inserted before other macros that use logging
 !insertmacro _LoggingCommon
@@ -149,12 +149,12 @@ VIAddVersionKey "OriginalFilename" "setu
 !insertmacro OnEndCommon
 !insertmacro PreDirectoryCommon
 
-Name "${BrandFullName}"
-OutFile "setup.exe"
+Name "MiFF"
+OutFile "MiFFInstaller.exe"
 !ifdef HAVE_64BIT_BUILD
-  InstallDir "$PROGRAMFILES64\${BrandFullName}\"
+  InstallDir "$PROGRAMFILES64\MiFF\"
 !else
-  InstallDir "$PROGRAMFILES32\${BrandFullName}\"
+  InstallDir "$PROGRAMFILES32\MiFF\"
 !endif
 ShowInstDetails nevershow
 
@@ -245,7 +245,7 @@ Section "-InstallStartCleanup"
   SetDetailsPrint none
 
   SetOutPath "$INSTDIR"
-  ${StartInstallLog} "${BrandFullName}" "${AB_CD}" "${AppVersion}" "${GREVersion}"
+  ${StartInstallLog} "MiFF" "${AB_CD}" "${AppVersion}" "${GREVersion}"
 
   StrCpy $R9 "true"
   StrCpy $PreventRebootRequired "false"
Index: firefox-84.0.2/toolkit/mozapps/installer/windows/nsis/makensis.mk
===================================================================
--- firefox-84.0.2.orig/toolkit/mozapps/installer/windows/nsis/makensis.mk
+++ firefox-84.0.2/toolkit/mozapps/installer/windows/nsis/makensis.mk
@@ -51,7 +51,7 @@ CUSTOM_UI = \
 	nsisui.exe \
 	$(NULL)
 
-$(CONFIG_DIR)/setup.exe::
+$(CONFIG_DIR)/MiFFInstaller.exe::
 	$(INSTALL) $(addprefix $(MOZILLA_DIR)/toolkit/mozapps/installer/windows/nsis/,$(TOOLKIT_NSIS_FILES)) $(CONFIG_DIR)
 	$(INSTALL) $(addprefix $(MOZILLA_DIR)/other-licenses/nsis/Plugins/,$(CUSTOM_NSIS_PLUGINS)) $(CONFIG_DIR)
 	$(INSTALL) $(addprefix $(MOZILLA_DIR)/other-licenses/nsis/,$(CUSTOM_UI)) $(CONFIG_DIR)
@@ -61,7 +61,7 @@ ifdef MOZ_STUB_INSTALLER
 endif
 
 ifdef ZIP_IN
-installer:: $(CONFIG_DIR)/setup.exe $(ZIP_IN)
+installer:: $(CONFIG_DIR)/MiFFInstaller.exe $(ZIP_IN)
 	@echo 'Packaging $(WIN32_INSTALLER_OUT).'
 	$(NSINSTALL) -D '$(ABS_DIST)/$(PKG_INST_PATH)'
 	$(PYTHON3) $(MOZILLA_DIR)/mach repackage installer \
@@ -69,7 +69,7 @@ installer:: $(CONFIG_DIR)/setup.exe $(ZI
 	  --package-name '$(MOZ_PKG_DIR)' \
 	  --package '$(ZIP_IN)' \
 	  --tag $(topsrcdir)/$(MOZ_BUILD_APP)/installer/windows/app.tag \
-	  --setupexe $(CONFIG_DIR)/setup.exe \
+	  --setupexe $(CONFIG_DIR)/MiFFInstaller.exe \
 	  --sfx-stub $(SFX_MODULE) \
 	  $(USE_UPX)
 ifdef MOZ_STUB_INSTALLER
Index: firefox-84.0.2/browser/locales/en-US/browser/preferences/preferences.ftl
===================================================================
--- firefox-84.0.2.orig/browser/locales/en-US/browser/preferences/preferences.ftl
+++ firefox-84.0.2/browser/locales/en-US/browser/preferences/preferences.ftl
@@ -590,7 +590,7 @@ home-restore-defaults =
 # "Firefox" should be treated as a brand and kept in English,
 # while "Home" and "(Default)" can be localized.
 home-mode-choice-default =
-    .label = Firefox Home (Default)
+    .label = Home (Default)
 
 home-mode-choice-custom =
     .label = Custom URLs…
@@ -619,8 +619,8 @@ choose-bookmark =
 
 ## Home Section - Firefox Home Content Customization
 
-home-prefs-content-header = Firefox Home Content
-home-prefs-content-description = Choose what content you want on your Firefox Home screen.
+home-prefs-content-header = Home Content
+home-prefs-content-description = Choose what content you want on your Home screen.
 
 home-prefs-search-header =
     .label = Web Search
@@ -779,7 +779,7 @@ sync-signedout-account-signin2 =
 #
 # They can be moved within the sentence as needed to adapt
 # to your language, but should not be changed or translated.
-sync-mobile-promo = Download Firefox for <img data-l10n-name="android-icon"/> <a data-l10n-name="android-link">Android</a> or <img data-l10n-name="ios-icon"/> <a data-l10n-name="ios-link">iOS</a> to sync with your mobile device.
+sync-mobile-promo = Mobile versions of MiFF in development
 
 ## Firefox Account - Signed in
 
@@ -886,7 +886,7 @@ sync-engine-creditcards =
 
 sync-engine-addons =
     .label = Add-ons
-    .tooltiptext = Extensions and themes for Firefox desktop
+    .tooltiptext = Extensions and themes for MiFF desktop
     .accesskey = A
 
 sync-engine-prefs =
Index: firefox-84.0.2/browser/locales/en-US/browser/aboutLogins.ftl
===================================================================
--- firefox-84.0.2.orig/browser/locales/en-US/browser/aboutLogins.ftl
+++ firefox-84.0.2/browser/locales/en-US/browser/aboutLogins.ftl
@@ -62,11 +62,11 @@ about-logins-list-item-vulnerable-passwo
 about-logins-login-intro-heading-logged-out = Looking for your saved logins? Set up { -sync-brand-short-name } or Import Them.
 about-logins-login-intro-heading-logged-in = No synced logins found.
 login-intro-description = If you saved your logins to { -brand-product-name } on a different device, here’s how to get them here:
-login-intro-instruction-fxa = Create or sign in to your { -fxaccount-brand-name } on the device where your logins are saved
+login-intro-instruction-fxa = Sign in to privacy.app on the device where your logins are saved
 login-intro-instruction-fxa-settings = Make sure you’ve selected the Logins checkbox in { -sync-brand-short-name } Settings
 about-logins-intro-instruction-help = Visit <a data-l10n-name="help-link">{ -lockwise-brand-short-name } Support</a> for more help
 about-logins-intro-import = If your logins are saved in another browser, you can <a data-l10n-name="import-link">import them into { -lockwise-brand-short-name }</a>
-about-logins-intro-import2 = If your logins are saved outside of { -brand-product-name }, you can <a data-l10n-name="import-browser-link">import them from another browser</a> or <a data-l10n-name="import-file-link">from a file</a>
+about-logins-intro-import2 = If your logins are saved outside of MiFF, you can <a data-l10n-name="import-browser-link">import them from another browser</a> or <a data-l10n-name="import-file-link">from a file</a>
 
 ## Login
 
Index: firefox-84.0.2/browser/components/privatebrowsing/content/aboutPrivateBrowsing.html
===================================================================
--- firefox-84.0.2.orig/browser/components/privatebrowsing/content/aboutPrivateBrowsing.html
+++ firefox-84.0.2/browser/components/privatebrowsing/content/aboutPrivateBrowsing.html
@@ -58,7 +58,7 @@
       </div>
     </div>
 
-    <p id="private-browsing-vpn-text" class="vpn-promo" data-l10n-id="about-private-browsing-need-more-privacy"></p>
-    <a id="private-browsing-vpn-link" class="vpn-promo" data-l10n-id="about-private-browsing-turn-on-vpn"></a>
+<!--    <p id="private-browsing-vpn-text" class="vpn-promo" data-l10n-id="about-private-browsing-need-more-privacy"></p>
+    <a id="private-browsing-vpn-link" class="vpn-promo" data-l10n-id="about-private-browsing-turn-on-vpn"></a>-->
   </body>
 </html>
Index: firefox-84.0.2/browser/locales/en-US/browser/menubar.ftl
===================================================================
--- firefox-84.0.2.orig/browser/locales/en-US/browser/menubar.ftl
+++ firefox-84.0.2/browser/locales/en-US/browser/menubar.ftl
@@ -192,7 +192,7 @@ menu-tools-addons =
     .label = Add-ons
     .accesskey = A
 menu-tools-fxa-sign-in =
-    .label = Sign In To { -brand-product-name }…
+    .label = Sign in to { -fxaccount-brand-name }…
     .accesskey = g
 menu-tools-turn-on-sync =
     .label = Turn on { -sync-brand-short-name }…
@@ -201,7 +201,7 @@ menu-tools-sync-now =
     .label = Sync Now
     .accesskey = S
 menu-tools-fxa-re-auth =
-    .label = Reconnect to { -brand-product-name }…
+    .label = Reconnect to { -fxaccount-brand-name }…
     .accesskey = R
 menu-tools-web-developer =
     .label = Web Developer
Index: firefox-84.0.2/browser/branding/aurora/locales/en-US/brand.ftl
===================================================================
--- firefox-84.0.2.orig/browser/branding/aurora/locales/en-US/brand.ftl
+++ firefox-84.0.2/browser/branding/aurora/locales/en-US/brand.ftl
@@ -16,11 +16,11 @@
 ## For further details, consult:
 ## https://mozilla-l10n.github.io/styleguides/mozilla_general/#brands-copyright-and-trademark
 
--brand-shorter-name = Firefox
--brand-short-name = Firefox Developer Edition
--brand-full-name = Firefox Developer Edition
+-brand-shorter-name = MiFF
+-brand-short-name = MiFF
+-brand-full-name = MiFF
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
+-brand-product-name = MiFF
+-vendor-short-name = Magnusson Institute
 trademarkInfo = { " " }
Index: firefox-84.0.2/browser/branding/nightly/locales/en-US/brand.ftl
===================================================================
--- firefox-84.0.2.orig/browser/branding/nightly/locales/en-US/brand.ftl
+++ firefox-84.0.2/browser/branding/nightly/locales/en-US/brand.ftl
@@ -16,11 +16,11 @@
 ## For further details, consult:
 ## https://mozilla-l10n.github.io/styleguides/mozilla_general/#brands-copyright-and-trademark
 
--brand-shorter-name = Nightly
--brand-short-name = Nightly
--brand-full-name = Firefox Nightly
+-brand-shorter-name = MiFF
+-brand-short-name = MiFF
+-brand-full-name = MiFF
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
+-brand-product-name = MiFF
+-vendor-short-name = Magnusson Institute
 trademarkInfo = { " " }
Index: firefox-84.0.2/browser/branding/official/locales/en-US/brand.ftl
===================================================================
--- firefox-84.0.2.orig/browser/branding/official/locales/en-US/brand.ftl
+++ firefox-84.0.2/browser/branding/official/locales/en-US/brand.ftl
@@ -16,11 +16,11 @@
 ## For further details, consult:
 ## https://mozilla-l10n.github.io/styleguides/mozilla_general/#brands-copyright-and-trademark
 
--brand-shorter-name = Firefox
--brand-short-name = Firefox
--brand-full-name = Mozilla Firefox
+-brand-shorter-name = MiFF
+-brand-short-name = MiFF
+-brand-full-name = MiFF
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
+-brand-product-name = MiFF
+-vendor-short-name = Magnusson Institute
 trademarkInfo = Firefox and the Firefox logos are trademarks of the Mozilla Foundation.
Index: firefox-84.0.2/browser/branding/unofficial/locales/en-US/brand.ftl
===================================================================
--- firefox-84.0.2.orig/browser/branding/unofficial/locales/en-US/brand.ftl
+++ firefox-84.0.2/browser/branding/unofficial/locales/en-US/brand.ftl
@@ -16,11 +16,11 @@
 ## For further details, consult:
 ## https://mozilla-l10n.github.io/styleguides/mozilla_general/#brands-copyright-and-trademark
 
--brand-shorter-name = Nightly
--brand-short-name = Nightly
--brand-full-name = Nightly
+-brand-shorter-name = MiFF
+-brand-short-name = MiFF
+-brand-full-name = MiFF
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
+-brand-product-name = MiFF
+-vendor-short-name = Magnusson Institute
 trademarkInfo = { " " }
Index: firefox-84.0.2/browser/branding/official/branding.nsi
===================================================================
--- firefox-84.0.2.orig/browser/branding/official/branding.nsi
+++ firefox-84.0.2/browser/branding/official/branding.nsi
@@ -8,12 +8,12 @@
 
 # BrandFullNameInternal is used for some registry and file system values
 # instead of BrandFullName and typically should not be modified.
-!define BrandFullNameInternal "Mozilla Firefox"
-!define BrandFullName         "Mozilla Firefox"
-!define CompanyName           "Mozilla Corporation"
-!define URLInfoAbout          "https://www.mozilla.org"
+!define BrandFullNameInternal "MiFF"
+!define BrandFullName         "MiFF"
+!define CompanyName           "Magnusson Institute"
+!define URLInfoAbout          "https://privacy.app"
 !define URLUpdateInfo         "https://www.mozilla.org/firefox/${AppVersion}/releasenotes"
-!define HelpLink              "https://support.mozilla.org"
+!define HelpLink              "https://privacy.app/support"
 
 ; The OFFICIAL define is a workaround to support different urls for Release and
 ; Beta since they share the same branding when building with other branches that
Index: firefox-84.0.2/browser/branding/unofficial/branding.nsi
===================================================================
--- firefox-84.0.2.orig/browser/branding/unofficial/branding.nsi
+++ firefox-84.0.2/browser/branding/unofficial/branding.nsi
@@ -8,11 +8,11 @@
 
 # BrandFullNameInternal is used for some registry and file system values
 # instead of BrandFullName and typically should not be modified.
-!define BrandFullNameInternal "Mozilla Developer Preview"
-!define BrandFullName         "Mozilla Developer Preview"
-!define CompanyName           "mozilla.org"
-!define URLInfoAbout          "https://www.mozilla.org"
-!define HelpLink              "https://support.mozilla.org"
+!define BrandFullNameInternal "MiFF"
+!define BrandFullName         "MiFF"
+!define CompanyName           "Magnusson Institute"
+!define URLInfoAbout          "https://privacy.app"
+!define HelpLink              "https://privacy.app/support"
 
 !define URLStubDownloadX86 "https://download.mozilla.org/?os=win&lang=${AB_CD}&product=firefox-latest"
 !define URLStubDownloadAMD64 "https://download.mozilla.org/?os=win64&lang=${AB_CD}&product=firefox-latest"
Index: firefox-84.0.2/browser/branding/aurora/branding.nsi
===================================================================
--- firefox-84.0.2.orig/browser/branding/aurora/branding.nsi
+++ firefox-84.0.2/browser/branding/aurora/branding.nsi
@@ -8,12 +8,12 @@
 
 # BrandFullNameInternal is used for some registry and file system values
 # instead of BrandFullName and typically should not be modified.
-!define BrandFullNameInternal "Firefox Developer Edition"
-!define BrandShortName        "Firefox Developer Edition"
-!define BrandFullName         "Firefox Developer Edition"
-!define CompanyName           "mozilla.org"
-!define URLInfoAbout          "https://www.mozilla.org"
-!define HelpLink              "https://support.mozilla.org"
+!define BrandFullNameInternal "MiFF"
+!define BrandShortName        "MiFF"
+!define BrandFullName         "MiFF"
+!define CompanyName           "Magnusson Institute"
+!define URLInfoAbout          "https://privacy.app"
+!define HelpLink              "https://privacy.app/support"
 
 !define URLStubDownloadX86 "https://download.mozilla.org/?os=win&lang=${AB_CD}&product=firefox-devedition-latest"
 !define URLStubDownloadAMD64 "https://download.mozilla.org/?os=win64&lang=${AB_CD}&product=firefox-devedition-latest"
Index: firefox-84.0.2/browser/branding/nightly/branding.nsi
===================================================================
--- firefox-84.0.2.orig/browser/branding/nightly/branding.nsi
+++ firefox-84.0.2/browser/branding/nightly/branding.nsi
@@ -8,11 +8,11 @@
 
 # BrandFullNameInternal is used for some registry and file system values
 # instead of BrandFullName and typically should not be modified.
-!define BrandFullNameInternal "Nightly"
-!define BrandFullName         "Firefox Nightly"
-!define CompanyName           "mozilla.org"
-!define URLInfoAbout          "https://www.mozilla.org"
-!define HelpLink              "https://support.mozilla.org"
+!define BrandFullNameInternal "MiFF"
+!define BrandFullName         "MiFF"
+!define CompanyName           "Magnusson Institute"
+!define URLInfoAbout          "https://privacy.app"
+!define HelpLink              "https://privacy.app/support"
 
 !define URLStubDownloadX86 "https://download.mozilla.org/?os=win&lang=${AB_CD}&product=firefox-nightly-latest"
 !define URLStubDownloadAMD64 "https://download.mozilla.org/?os=win64&lang=${AB_CD}&product=firefox-nightly-latest"
Index: firefox-84.0.2/browser/branding/aurora/locales/en-US/brand.dtd
===================================================================
--- firefox-84.0.2.orig/browser/branding/aurora/locales/en-US/brand.dtd
+++ firefox-84.0.2/browser/branding/aurora/locales/en-US/brand.dtd
@@ -2,10 +2,11 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
-<!ENTITY  brandShorterName      "Firefox">
-<!ENTITY  brandShortName        "Firefox Developer Edition">
-<!ENTITY  brandFullName         "Firefox Developer Edition">
+<!ENTITY  brandShorterName      "MiFF">
+<!ENTITY  brandShortName        "MiFF">
+<!ENTITY  brandFullName         "MiFF">
 <!-- LOCALIZATION NOTE (brandProductName):
    This brand name can be used in messages where the product name needs to
    remain unchanged across different versions (Nightly, Beta, etc.). -->
-<!ENTITY  brandProductName      "Firefox">
+<!ENTITY  brandProductName      "MiFF">
+<!ENTITY  brandSSOName          "privacy.app">
Index: firefox-84.0.2/browser/branding/aurora/locales/en-US/brand.properties
===================================================================
--- firefox-84.0.2.orig/browser/branding/aurora/locales/en-US/brand.properties
+++ firefox-84.0.2/browser/branding/aurora/locales/en-US/brand.properties
@@ -2,13 +2,13 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-brandShorterName=Firefox
-brandShortName=Firefox Developer Edition
-brandFullName=Firefox Developer Edition
+brandShorterName=MiFF
+brandShortName=MiFF
+brandFullName=MiFF
 # LOCALIZATION NOTE(brandProductName):
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
-brandProductName=Firefox
-vendorShortName=Mozilla
+brandProductName=MiFF
+vendorShortName=Magnusson Institute
 
 syncBrandShortName=Sync
Index: firefox-84.0.2/browser/branding/nightly/locales/en-US/brand.dtd
===================================================================
--- firefox-84.0.2.orig/browser/branding/nightly/locales/en-US/brand.dtd
+++ firefox-84.0.2/browser/branding/nightly/locales/en-US/brand.dtd
@@ -2,10 +2,11 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
-<!ENTITY  brandShorterName      "Nightly">
-<!ENTITY  brandShortName        "Nightly">
-<!ENTITY  brandFullName         "Firefox Nightly">
+<!ENTITY  brandShorterName      "MiFF">
+<!ENTITY  brandShortName        "MiFF">
+<!ENTITY  brandFullName         "MiFF">
 <!-- LOCALIZATION NOTE (brandProductName):
    This brand name can be used in messages where the product name needs to
    remain unchanged across different versions (Nightly, Beta, etc.). -->
-<!ENTITY  brandProductName      "Firefox">
+<!ENTITY  brandProductName      "MiFF">
+<!ENTITY  brandSSOName          "privacy.app">
Index: firefox-84.0.2/browser/branding/nightly/locales/en-US/brand.properties
===================================================================
--- firefox-84.0.2.orig/browser/branding/nightly/locales/en-US/brand.properties
+++ firefox-84.0.2/browser/branding/nightly/locales/en-US/brand.properties
@@ -2,13 +2,13 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-brandShorterName=Nightly
-brandShortName=Nightly
-brandFullName=Firefox Nightly
+brandShorterName=MiFF
+brandShortName=MiFF
+brandFullName=MiFF
 # LOCALIZATION NOTE(brandProductName):
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
-brandProductName=Firefox
-vendorShortName=Mozilla
+brandProductName=MiFF
+vendorShortName=Magnusson Institute
 
 syncBrandShortName=Sync
Index: firefox-84.0.2/browser/branding/official/locales/en-US/brand.dtd
===================================================================
--- firefox-84.0.2.orig/browser/branding/official/locales/en-US/brand.dtd
+++ firefox-84.0.2/browser/branding/official/locales/en-US/brand.dtd
@@ -2,10 +2,11 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
-<!ENTITY  brandShorterName      "Firefox">
-<!ENTITY  brandShortName        "Firefox">
-<!ENTITY  brandFullName         "Mozilla Firefox">
+<!ENTITY  brandShorterName      "MiFF">
+<!ENTITY  brandShortName        "MiFF">
+<!ENTITY  brandFullName         "MiFF">
 <!-- LOCALIZATION NOTE (brandProductName):
    This brand name can be used in messages where the product name needs to
    remain unchanged across different versions (Nightly, Beta, etc.). -->
-<!ENTITY  brandProductName      "Firefox">
+<!ENTITY  brandProductName      "MiFF">
+<!ENTITY  brandSSOName          "privacy.app">
Index: firefox-84.0.2/browser/branding/official/locales/en-US/brand.properties
===================================================================
--- firefox-84.0.2.orig/browser/branding/official/locales/en-US/brand.properties
+++ firefox-84.0.2/browser/branding/official/locales/en-US/brand.properties
@@ -2,13 +2,13 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-brandShorterName=Firefox
-brandShortName=Firefox
-brandFullName=Mozilla Firefox
+brandShorterName=MiFF
+brandShortName=MiFF
+brandFullName=MiFF
 # LOCALIZATION NOTE(brandProductName):
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
-brandProductName=Firefox
-vendorShortName=Mozilla
+brandProductName=MiFF
+vendorShortName=Magnusson Institute
 
 syncBrandShortName=Sync
Index: firefox-84.0.2/browser/branding/unofficial/locales/en-US/brand.dtd
===================================================================
--- firefox-84.0.2.orig/browser/branding/unofficial/locales/en-US/brand.dtd
+++ firefox-84.0.2/browser/branding/unofficial/locales/en-US/brand.dtd
@@ -2,10 +2,11 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
-<!ENTITY  brandShorterName      "Nightly">
-<!ENTITY  brandShortName        "Nightly">
-<!ENTITY  brandFullName         "Nightly">
+<!ENTITY  brandShorterName      "MiFF">
+<!ENTITY  brandShortName        "MiFF">
+<!ENTITY  brandFullName         "MiFF">
 <!-- LOCALIZATION NOTE (brandProductName):
    This brand name can be used in messages where the product name needs to
    remain unchanged across different versions (Nightly, Beta, etc.). -->
-<!ENTITY  brandProductName      "Firefox">
+<!ENTITY  brandProductName      "MiFF">
+<!ENTITY  brandSSOName          "privacy.app">
Index: firefox-84.0.2/browser/branding/unofficial/locales/en-US/brand.properties
===================================================================
--- firefox-84.0.2.orig/browser/branding/unofficial/locales/en-US/brand.properties
+++ firefox-84.0.2/browser/branding/unofficial/locales/en-US/brand.properties
@@ -2,13 +2,13 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-brandShorterName=Nightly
-brandShortName=Nightly
-brandFullName=Nightly
+brandShorterName=MiFF
+brandShortName=MiFF
+brandFullName=MiFF
 # LOCALIZATION NOTE(brandProductName):
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
-brandProductName=Firefox
-vendorShortName=Mozilla
+brandProductName=MiFF
+vendorShortName=Magnusson Institute
 
 syncBrandShortName=Sync
Index: firefox-84.0.2/browser/app/module.ver
===================================================================
--- firefox-84.0.2.orig/browser/app/module.ver
+++ firefox-84.0.2/browser/app/module.ver
@@ -5,4 +5,4 @@ WIN32_MODULE_PRODUCTVERSION_STRING=@MOZ_
 WIN32_MODULE_TRADEMARKS=Firefox is a Trademark of The Mozilla Foundation.
 WIN32_MODULE_DESCRIPTION=@MOZ_APP_DISPLAYNAME@
 WIN32_MODULE_PRODUCTNAME=@MOZ_APP_DISPLAYNAME@
-WIN32_MODULE_NAME=@MOZ_APP_DISPLAYNAME@
+WIN32_MODULE_NAME=MiFF
Index: firefox-84.0.2/browser/installer/windows/nsis/defines.nsi.in
===================================================================
--- firefox-84.0.2.orig/browser/installer/windows/nsis/defines.nsi.in
+++ firefox-84.0.2/browser/installer/windows/nsis/defines.nsi.in
@@ -21,23 +21,23 @@
 !endif
 
 # These defines should match application.ini settings
-!define AppName               "Firefox"
+!define AppName               "MiFF"
 !define AppVersion            "@APP_VERSION@"
 !define GREVersion            @MOZILLA_VERSION@
 !define AB_CD                 "@AB_CD@"
 
-!define FileMainEXE           "@MOZ_APP_NAME@.exe"
+!define FileMainEXE           "miff.exe"
 !define MainWindowClass       "MozillaWindowClass"
 !define DialogWindowClass     "MozillaDialogClass"
-!define DDEApplication        "Firefox"
-!define AppRegName            "Firefox"
+!define DDEApplication        "MiFF"
+!define AppRegName            "MiFF"
 
-!define BrandProductName      "Firefox"
+!define BrandProductName      "MiFF"
 !ifndef DEV_EDITION
-!define BrandShortName        "@MOZ_APP_DISPLAYNAME@"
+!define BrandShortName        "MiFF"
 !endif
 !ifndef BrandFullName
-!define BrandFullName         "${BrandFullNameInternal}"
+!define BrandFullName         "MiFF"
 !endif
 
 !define CERTIFICATE_NAME            "Mozilla Corporation"
Index: firefox-84.0.2/browser/locales/en-US/chrome/browser/browser.dtd
===================================================================
--- firefox-84.0.2.orig/browser/locales/en-US/chrome/browser/browser.dtd
+++ firefox-84.0.2/browser/locales/en-US/chrome/browser/browser.dtd
@@ -43,7 +43,7 @@ this container is a toolbar. This avoids
 <!ENTITY fxa.menu.syncSettings2.label "&syncBrand.shortName.label; Settings">
 <!ENTITY fxa.menu.connectAnotherDevice2.label "Connect Another Device…">
 <!ENTITY fxa.menu.firefoxAccount "&syncBrand.fxAccount.label;">
-<!ENTITY fxa.menu.signin.label "Sign in to &brandProductName;">
+<!ENTITY fxa.menu.signin.label "Sign in to &brandSSOName;">
 <!ENTITY fxa.menu.turnOnSync.label "Turn on &syncBrand.shortName.label;">
 <!ENTITY fxa.menu.firefoxServices.label "&brandProductName; Services">
 <!ENTITY fxa.menu.account.label "Account">
Index: firefox-84.0.2/browser/locales/en-US/chrome/browser/accounts.properties
===================================================================
--- firefox-84.0.2.orig/browser/locales/en-US/chrome/browser/accounts.properties
+++ firefox-84.0.2/browser/locales/en-US/chrome/browser/accounts.properties
@@ -19,7 +19,7 @@ verificationNotSentBody = We are unable
 # LOCALIZATION NOTE (deviceConnDisconnTitle) - the title shown in a notification
 # when either this device or another device has connected to, or disconnected
 # from, a Firefox Account.
-deviceConnDisconnTitle = Firefox Account
+deviceConnDisconnTitle = privacy.app
 
 # LOCALIZATION NOTE (otherDeviceConnectedBody, otherDeviceConnectedBody.noDeviceName)
 # These strings are used in a notification shown when a new device joins the
@@ -126,4 +126,4 @@ account.finishAccountSetup = Finish Acco
 # LOCALIZATION NOTE (account.reconnectToFxA):
 # Used as the FxA toolbar menu item title when the user
 # needs to reconnect their account.
-account.reconnectToFxA = Reconnect to Firefox Account
+account.reconnectToFxA = Reconnect to privacy.app
Index: firefox-84.0.2/browser/locales/en-US/browser/branding/sync-brand.ftl
===================================================================
--- firefox-84.0.2.orig/browser/locales/en-US/browser/branding/sync-brand.ftl
+++ firefox-84.0.2/browser/locales/en-US/browser/branding/sync-brand.ftl
@@ -6,8 +6,8 @@
 
 # “Sync” can be localized, “Firefox” must be treated as a brand,
 # and kept in English.
--sync-brand-name = Firefox Sync
+-sync-brand-name = MiFF Sync
 
 # “Account” can be localized, “Firefox” must be treated as a brand,
 # and kept in English.
--fxaccount-brand-name = Firefox Account
+-fxaccount-brand-name = privacy.app
Index: firefox-84.0.2/browser/locales/en-US/chrome/browser/syncBrand.dtd
===================================================================
--- firefox-84.0.2.orig/browser/locales/en-US/chrome/browser/syncBrand.dtd
+++ firefox-84.0.2/browser/locales/en-US/chrome/browser/syncBrand.dtd
@@ -3,4 +3,4 @@
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
 <!ENTITY syncBrand.shortName.label  "Sync">
-<!ENTITY syncBrand.fxAccount.label  "Firefox Account">
+<!ENTITY syncBrand.fxAccount.label  "Magnusson Institute Account">
