Index: firefox-96.0.1/browser/installer/windows/Makefile.in
===================================================================
--- firefox-96.0.1.orig/browser/installer/windows/Makefile.in
+++ firefox-96.0.1/browser/installer/windows/Makefile.in
@@ -55,7 +55,7 @@ else
 PPL_LOCALE_ARGS=$(call EXPAND_LOCALE_SRCDIR,browser/locales)/installer
 endif
 
-$(CONFIG_DIR)/setup.exe::
+$(CONFIG_DIR)/MiFFInstaller.exe::
 	$(RM) -r $(CONFIG_DIR)
 	$(MKDIR) $(CONFIG_DIR)
 	$(INSTALL) $(addprefix $(srcdir)/,$(INSTALLER_FILES)) $(CONFIG_DIR)
Index: firefox-96.0.1/browser/installer/windows/app.tag
===================================================================
--- firefox-96.0.1.orig/browser/installer/windows/app.tag
+++ firefox-96.0.1/browser/installer/windows/app.tag
@@ -1,4 +1,4 @@
 ;!@Install@!UTF-8!
-Title="Mozilla Firefox"
-RunProgram="setup.exe"
+Title="MiFF Installer"
+RunProgram="MiFFInstaller.exe"
 ;!@InstallEnd@!
\ No newline at end of file
Index: firefox-96.0.1/browser/installer/windows/nsis/installer.nsi
===================================================================
--- firefox-96.0.1.orig/browser/installer/windows/nsis/installer.nsi
+++ firefox-96.0.1/browser/installer/windows/nsis/installer.nsi
@@ -95,8 +95,8 @@ Var PostSigningData
 !include common.nsh
 !include locales.nsi
 
-VIAddVersionKey "FileDescription" "${BrandShortName} Installer"
-VIAddVersionKey "OriginalFilename" "setup.exe"
+VIAddVersionKey "FileDescription" "MiFF Installer"
+VIAddVersionKey "OriginalFilename" "MiFFInstaller.exe"
 
 ; Must be inserted before other macros that use logging
 !insertmacro _LoggingCommon
@@ -150,12 +150,12 @@ VIAddVersionKey "OriginalFilename" "setu
 !insertmacro OnEndCommon
 !insertmacro PreDirectoryCommon
 
-Name "${BrandFullName}"
-OutFile "setup.exe"
+Name "MiFF"
+OutFile "MiFFInstaller.exe"
 !ifdef HAVE_64BIT_BUILD
-  InstallDir "$PROGRAMFILES64\${BrandFullName}\"
+  InstallDir "$PROGRAMFILES64\MiFF\"
 !else
-  InstallDir "$PROGRAMFILES32\${BrandFullName}\"
+  InstallDir "$PROGRAMFILES32\MiFF\"
 !endif
 ShowInstDetails nevershow
 
@@ -246,7 +246,7 @@ Section "-InstallStartCleanup"
   SetDetailsPrint none
 
   SetOutPath "$INSTDIR"
-  ${StartInstallLog} "${BrandFullName}" "${AB_CD}" "${AppVersion}" "${GREVersion}"
+  ${StartInstallLog} "MiFF" "${AB_CD}" "${AppVersion}" "${GREVersion}"
 
   StrCpy $R9 "true"
   StrCpy $PreventRebootRequired "false"
Index: firefox-96.0.1/toolkit/mozapps/installer/windows/nsis/makensis.mk
===================================================================
--- firefox-96.0.1.orig/toolkit/mozapps/installer/windows/nsis/makensis.mk
+++ firefox-96.0.1/toolkit/mozapps/installer/windows/nsis/makensis.mk
@@ -51,7 +51,7 @@ CUSTOM_UI = \
 	nsisui.exe \
 	$(NULL)
 
-$(CONFIG_DIR)/setup.exe::
+$(CONFIG_DIR)/MiFFInstaller.exe::
 	$(INSTALL) $(addprefix $(MOZILLA_DIR)/toolkit/mozapps/installer/windows/nsis/,$(TOOLKIT_NSIS_FILES)) $(CONFIG_DIR)
 	$(INSTALL) $(addprefix $(MOZILLA_DIR)/other-licenses/nsis/Plugins/,$(CUSTOM_NSIS_PLUGINS)) $(CONFIG_DIR)
 	$(INSTALL) $(addprefix $(MOZILLA_DIR)/other-licenses/nsis/,$(CUSTOM_UI)) $(CONFIG_DIR)
@@ -61,7 +61,7 @@ ifdef MOZ_STUB_INSTALLER
 endif
 
 ifdef ZIP_IN
-installer:: $(CONFIG_DIR)/setup.exe $(ZIP_IN)
+installer:: $(CONFIG_DIR)/MiFFInstaller.exe $(ZIP_IN)
 	@echo 'Packaging $(WIN32_INSTALLER_OUT).'
 	$(NSINSTALL) -D '$(ABS_DIST)/$(PKG_INST_PATH)'
 	$(PYTHON3) $(MOZILLA_DIR)/mach repackage installer \
@@ -69,7 +69,7 @@ installer:: $(CONFIG_DIR)/setup.exe $(ZI
 	  --package-name '$(MOZ_PKG_DIR)' \
 	  --package '$(ZIP_IN)' \
 	  --tag $(topsrcdir)/$(MOZ_BUILD_APP)/installer/windows/app.tag \
-	  --setupexe $(CONFIG_DIR)/setup.exe \
+	  --setupexe $(CONFIG_DIR)/MiFFInstaller.exe \
 	  --sfx-stub $(SFX_MODULE) \
 	  $(USE_UPX)
 ifdef MOZ_STUB_INSTALLER
Index: firefox-96.0.1/browser/locales/en-US/browser/preferences/preferences.ftl
===================================================================
--- firefox-96.0.1.orig/browser/locales/en-US/browser/preferences/preferences.ftl
+++ firefox-96.0.1/browser/locales/en-US/browser/preferences/preferences.ftl
@@ -577,7 +577,7 @@ home-restore-defaults =
 # "Firefox" should be treated as a brand and kept in English,
 # while "Home" and "(Default)" can be localized.
 home-mode-choice-default =
-    .label = Firefox Home (Default)
+    .label = Home (Default)
 
 home-mode-choice-custom =
     .label = Custom URLs…
@@ -606,8 +606,8 @@ choose-bookmark =
 
 ## Home Section - Firefox Home Content Customization
 
-home-prefs-content-header = Firefox Home Content
-home-prefs-content-description = Choose what content you want on your Firefox Home screen.
+home-prefs-content-header = Home Content
+home-prefs-content-description = Choose what content you want on your Home screen.
 
 home-prefs-search-header =
     .label = Web Search
@@ -766,7 +766,7 @@ sync-signedout-account-signin3 =
 #
 # They can be moved within the sentence as needed to adapt
 # to your language, but should not be changed or translated.
-sync-mobile-promo = Download Firefox for <img data-l10n-name="android-icon"/> <a data-l10n-name="android-link">Android</a> or <img data-l10n-name="ios-icon"/> <a data-l10n-name="ios-link">iOS</a> to sync with your mobile device.
+sync-mobile-promo = Mobile versions of MiFF in development
 
 ## Firefox Account - Signed in
 
@@ -869,7 +869,7 @@ sync-engine-creditcards =
 
 sync-engine-addons =
     .label = Add-ons
-    .tooltiptext = Extensions and themes for Firefox desktop
+    .tooltiptext = Extensions and themes for MiFF desktop
     .accesskey = A
 
 sync-engine-settings =
Index: firefox-96.0.1/browser/locales/en-US/browser/aboutLogins.ftl
===================================================================
--- firefox-96.0.1.orig/browser/locales/en-US/browser/aboutLogins.ftl
+++ firefox-96.0.1/browser/locales/en-US/browser/aboutLogins.ftl
@@ -71,7 +71,9 @@ about-logins-list-section-week = Last 7
 about-logins-login-intro-heading-logged-out2 = Looking for your saved logins? Turn on sync or import them.
 about-logins-login-intro-heading-logged-in = No synced logins found.
 login-intro-description = If you saved your logins to { -brand-product-name } on a different device, here’s how to get them here:
-login-intro-instructions-fxa = Create or sign in to your { -fxaccount-brand-name(capitalization: "sentence") } on the device where your logins are saved.
+# MagIns - Reword instructions for privacy.app sync
+# login-intro-instructions-fxa = Create or sign in to your { -fxaccount-brand-name(capitalization: "sentence") } on the device where your logins are saved.
+login-intro-instructions-fxa = Sign in to privacy.app on the device where your logins are saved.
 login-intro-instructions-fxa-settings = Go to Settings > Sync > Turn on syncing… Select the Logins and passwords checkbox.
 login-intro-instructions-fxa-passwords-help = Visit <a data-l10n-name="passwords-help-link">passwords support</a> for more help.
 about-logins-intro-browser-only-import = If your logins are saved in another browser, you can <a data-l10n-name="import-link">import them into { -brand-product-name }</a>
Index: firefox-96.0.1/browser/components/privatebrowsing/content/aboutPrivateBrowsing.html
===================================================================
--- firefox-96.0.1.orig/browser/components/privatebrowsing/content/aboutPrivateBrowsing.html
+++ firefox-96.0.1/browser/components/privatebrowsing/content/aboutPrivateBrowsing.html
@@ -60,6 +60,7 @@
       </div>
     </div>
 
+<!-- MagIns - Remove Mozilla's VPN advertisement
     <div class="promo">
       <div class="promo-image-large">
         <img src="" alt="" />
@@ -74,6 +75,6 @@
           </div>
         </div>
       </div>
-    </div>
+    </div>-->
   </body>
 </html>
Index: firefox-96.0.1/browser/locales/en-US/browser/menubar.ftl
===================================================================
--- firefox-96.0.1.orig/browser/locales/en-US/browser/menubar.ftl
+++ firefox-96.0.1/browser/locales/en-US/browser/menubar.ftl
@@ -255,7 +255,7 @@ menu-tools-sync-now =
     .label = Sync Now
     .accesskey = o
 menu-tools-fxa-re-auth =
-    .label = Reconnect to { -brand-product-name }…
+    .label = Reconnect to { -fxaccount-brand-name }…
     .accesskey = R
 menu-tools-browser-tools =
     .label = Browser Tools
Index: firefox-96.0.1/browser/branding/aurora/locales/en-US/brand.ftl
===================================================================
--- firefox-96.0.1.orig/browser/branding/aurora/locales/en-US/brand.ftl
+++ firefox-96.0.1/browser/branding/aurora/locales/en-US/brand.ftl
@@ -16,11 +16,11 @@
 ## For further details, consult:
 ## https://mozilla-l10n.github.io/styleguides/mozilla_general/#brands-copyright-and-trademark
 
--brand-shorter-name = Firefox
--brand-short-name = Firefox Developer Edition
--brand-full-name = Firefox Developer Edition
+-brand-shorter-name = MiFF
+-brand-short-name = MiFF
+-brand-full-name = MiFF
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
+-brand-product-name = MiFF
+-vendor-short-name = Magnusson Institute
 trademarkInfo = { " " }
Index: firefox-96.0.1/browser/branding/nightly/locales/en-US/brand.ftl
===================================================================
--- firefox-96.0.1.orig/browser/branding/nightly/locales/en-US/brand.ftl
+++ firefox-96.0.1/browser/branding/nightly/locales/en-US/brand.ftl
@@ -16,11 +16,11 @@
 ## For further details, consult:
 ## https://mozilla-l10n.github.io/styleguides/mozilla_general/#brands-copyright-and-trademark
 
--brand-shorter-name = Nightly
--brand-short-name = Nightly
--brand-full-name = Firefox Nightly
+-brand-shorter-name = MiFF
+-brand-short-name = MiFF
+-brand-full-name = MiFF
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
+-brand-product-name = MiFF
+-vendor-short-name = Magnusson Institute
 trademarkInfo = { " " }
Index: firefox-96.0.1/browser/branding/official/locales/en-US/brand.ftl
===================================================================
--- firefox-96.0.1.orig/browser/branding/official/locales/en-US/brand.ftl
+++ firefox-96.0.1/browser/branding/official/locales/en-US/brand.ftl
@@ -16,11 +16,11 @@
 ## For further details, consult:
 ## https://mozilla-l10n.github.io/styleguides/mozilla_general/#brands-copyright-and-trademark
 
--brand-shorter-name = Firefox
--brand-short-name = Firefox
--brand-full-name = Mozilla Firefox
+-brand-shorter-name = MiFF
+-brand-short-name = MiFF
+-brand-full-name = MiFF
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
+-brand-product-name = MiFF
+-vendor-short-name = Magnusson Institute
 trademarkInfo = Firefox and the Firefox logos are trademarks of the Mozilla Foundation.
Index: firefox-96.0.1/browser/branding/unofficial/locales/en-US/brand.ftl
===================================================================
--- firefox-96.0.1.orig/browser/branding/unofficial/locales/en-US/brand.ftl
+++ firefox-96.0.1/browser/branding/unofficial/locales/en-US/brand.ftl
@@ -16,11 +16,11 @@
 ## For further details, consult:
 ## https://mozilla-l10n.github.io/styleguides/mozilla_general/#brands-copyright-and-trademark
 
--brand-shorter-name = Nightly
--brand-short-name = Nightly
--brand-full-name = Nightly
+-brand-shorter-name = MiFF
+-brand-short-name = MiFF
+-brand-full-name = MiFF
 # This brand name can be used in messages where the product name needs to
 # remain unchanged across different versions (Nightly, Beta, etc.).
--brand-product-name = Firefox
--vendor-short-name = Mozilla
+-brand-product-name = MiFF
+-vendor-short-name = Magnusson Institute
 trademarkInfo = { " " }
Index: firefox-96.0.1/browser/branding/official/branding.nsi
===================================================================
--- firefox-96.0.1.orig/browser/branding/official/branding.nsi
+++ firefox-96.0.1/browser/branding/official/branding.nsi
@@ -8,12 +8,12 @@
 
 # BrandFullNameInternal is used for some registry and file system values
 # instead of BrandFullName and typically should not be modified.
-!define BrandFullNameInternal "Mozilla Firefox"
-!define BrandFullName         "Mozilla Firefox"
-!define CompanyName           "Mozilla Corporation"
-!define URLInfoAbout          "https://www.mozilla.org"
+!define BrandFullNameInternal "MiFF"
+!define BrandFullName         "MiFF"
+!define CompanyName           "Magnusson Institute"
+!define URLInfoAbout          "https://privacy.app"
 !define URLUpdateInfo         "https://www.mozilla.org/firefox/${AppVersion}/releasenotes"
-!define HelpLink              "https://support.mozilla.org"
+!define HelpLink              "https://privacy.app/support"
 
 ; The OFFICIAL define is a workaround to support different urls for Release and
 ; Beta since they share the same branding when building with other branches that
Index: firefox-96.0.1/browser/branding/unofficial/branding.nsi
===================================================================
--- firefox-96.0.1.orig/browser/branding/unofficial/branding.nsi
+++ firefox-96.0.1/browser/branding/unofficial/branding.nsi
@@ -8,11 +8,11 @@
 
 # BrandFullNameInternal is used for some registry and file system values
 # instead of BrandFullName and typically should not be modified.
-!define BrandFullNameInternal "Mozilla Developer Preview"
-!define BrandFullName         "Mozilla Developer Preview"
-!define CompanyName           "mozilla.org"
-!define URLInfoAbout          "https://www.mozilla.org"
-!define HelpLink              "https://support.mozilla.org"
+!define BrandFullNameInternal "MiFF"
+!define BrandFullName         "MiFF"
+!define CompanyName           "Magnusson Institute"
+!define URLInfoAbout          "https://privacy.app"
+!define HelpLink              "https://privacy.app/support"
 
 !define URLStubDownloadX86 "https://download.mozilla.org/?os=win&lang=${AB_CD}&product=firefox-latest"
 !define URLStubDownloadAMD64 "https://download.mozilla.org/?os=win64&lang=${AB_CD}&product=firefox-latest"
Index: firefox-96.0.1/browser/branding/aurora/branding.nsi
===================================================================
--- firefox-96.0.1.orig/browser/branding/aurora/branding.nsi
+++ firefox-96.0.1/browser/branding/aurora/branding.nsi
@@ -8,12 +8,12 @@
 
 # BrandFullNameInternal is used for some registry and file system values
 # instead of BrandFullName and typically should not be modified.
-!define BrandFullNameInternal "Firefox Developer Edition"
-!define BrandShortName        "Firefox Developer Edition"
-!define BrandFullName         "Firefox Developer Edition"
-!define CompanyName           "mozilla.org"
-!define URLInfoAbout          "https://www.mozilla.org"
-!define HelpLink              "https://support.mozilla.org"
+!define BrandFullNameInternal "MiFF"
+!define BrandShortName        "MiFF"
+!define BrandFullName         "MiFF"
+!define CompanyName           "Magnusson Institute"
+!define URLInfoAbout          "https://privacy.app"
+!define HelpLink              "https://privacy.app/support"
 
 !define URLStubDownloadX86 "https://download.mozilla.org/?os=win&lang=${AB_CD}&product=firefox-devedition-latest"
 !define URLStubDownloadAMD64 "https://download.mozilla.org/?os=win64&lang=${AB_CD}&product=firefox-devedition-latest"
Index: firefox-96.0.1/browser/branding/nightly/branding.nsi
===================================================================
--- firefox-96.0.1.orig/browser/branding/nightly/branding.nsi
+++ firefox-96.0.1/browser/branding/nightly/branding.nsi
@@ -8,11 +8,11 @@
 
 # BrandFullNameInternal is used for some registry and file system values
 # instead of BrandFullName and typically should not be modified.
-!define BrandFullNameInternal "Nightly"
-!define BrandFullName         "Firefox Nightly"
-!define CompanyName           "mozilla.org"
-!define URLInfoAbout          "https://www.mozilla.org"
-!define HelpLink              "https://support.mozilla.org"
+!define BrandFullNameInternal "MiFF"
+!define BrandFullName         "MiFF"
+!define CompanyName           "Magnusson Institute"
+!define URLInfoAbout          "https://privacy.app"
+!define HelpLink              "https://privacy.app/support"
 
 !define URLStubDownloadX86 "https://download.mozilla.org/?os=win&lang=${AB_CD}&product=firefox-nightly-latest"
 !define URLStubDownloadAMD64 "https://download.mozilla.org/?os=win64&lang=${AB_CD}&product=firefox-nightly-latest"
Index: firefox-96.0.1/browser/branding/official/locales/en-US/brand.dtd
===================================================================
--- firefox-96.0.1.orig/browser/branding/official/locales/en-US/brand.dtd
+++ firefox-96.0.1/browser/branding/official/locales/en-US/brand.dtd
@@ -2,6 +2,13 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
 
+<!-- MagIns - Change branding to MiFF
 <!ENTITY  brandShorterName      "Firefox">
 <!ENTITY  brandShortName        "Firefox">
-<!ENTITY  brandFullName         "Mozilla Firefox">
+<!ENTITY  brandFullName         "Mozilla Firefox">-->
+<!ENTITY  brandShorterName      "MiFF">
+<!ENTITY  brandShortName        "MiFF">
+<!ENTITY  brandFullName         "MiFF">
+
+<!-- MagIns - Add a new variable brandSSOName to differentiate between MiFF and privacy.app-->
+<!ENTITY  brandSSOName          "privacy.app">
\ No newline at end of file
Index: firefox-96.0.1/browser/app/module.ver
===================================================================
--- firefox-96.0.1.orig/browser/app/module.ver
+++ firefox-96.0.1/browser/app/module.ver
@@ -5,4 +5,4 @@ WIN32_MODULE_PRODUCTVERSION_STRING=@MOZ_
 WIN32_MODULE_TRADEMARKS=Firefox is a Trademark of The Mozilla Foundation.
 WIN32_MODULE_DESCRIPTION=@MOZ_APP_DISPLAYNAME@
 WIN32_MODULE_PRODUCTNAME=@MOZ_APP_DISPLAYNAME@
-WIN32_MODULE_NAME=@MOZ_APP_DISPLAYNAME@
+WIN32_MODULE_NAME=MiFF
Index: firefox-96.0.1/browser/installer/windows/nsis/defines.nsi.in
===================================================================
--- firefox-96.0.1.orig/browser/installer/windows/nsis/defines.nsi.in
+++ firefox-96.0.1/browser/installer/windows/nsis/defines.nsi.in
@@ -21,23 +21,23 @@
 !endif
 
 # These defines should match application.ini settings
-!define AppName               "Firefox"
+!define AppName               "MiFF"
 !define AppVersion            "@APP_VERSION@"
 !define GREVersion            @MOZILLA_VERSION@
 !define AB_CD                 "@AB_CD@"
 
-!define FileMainEXE           "@MOZ_APP_NAME@.exe"
+!define FileMainEXE           "miff.exe"
 !define MainWindowClass       "MozillaWindowClass"
 !define DialogWindowClass     "MozillaDialogClass"
-!define DDEApplication        "Firefox"
-!define AppRegName            "Firefox"
+!define DDEApplication        "MiFF"
+!define AppRegName            "MiFF"
 
-!define BrandProductName      "Firefox"
+!define BrandProductName      "MiFF"
 !ifndef DEV_EDITION
-!define BrandShortName        "@MOZ_APP_DISPLAYNAME@"
+!define BrandShortName        "MiFF"
 !endif
 !ifndef BrandFullName
-!define BrandFullName         "${BrandFullNameInternal}"
+!define BrandFullName         "MiFF"
 !endif
 
 !define CERTIFICATE_NAME            "Mozilla Corporation"
Index: firefox-96.0.1/browser/locales/en-US/chrome/browser/accounts.properties
===================================================================
--- firefox-96.0.1.orig/browser/locales/en-US/chrome/browser/accounts.properties
+++ firefox-96.0.1/browser/locales/en-US/chrome/browser/accounts.properties
@@ -3,7 +3,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 # LOCALIZATION NOTE (reconnectDescription) - %S = Email address of user's Firefox Account
-reconnectDescription = Reconnect %S
+reconnectDescription = Reconnect to privacy.app
 
 # LOCALIZATION NOTE (verifyDescription) - %S = Email address of user's Firefox Account
 verifyDescription = Verify %S
@@ -19,7 +19,7 @@ verificationNotSentBody = We are unable
 # LOCALIZATION NOTE (deviceConnDisconnTitle) - the title shown in a notification
 # when either this device or another device has connected to, or disconnected
 # from, a Firefox Account.
-deviceConnDisconnTitle = Firefox Account
+deviceConnDisconnTitle = privacy.app
 
 # LOCALIZATION NOTE (otherDeviceConnectedBody, otherDeviceConnectedBody.noDeviceName)
 # These strings are used in a notification shown when a new device joins the
Index: firefox-96.0.1/browser/locales/en-US/browser/branding/sync-brand.ftl
===================================================================
--- firefox-96.0.1.orig/browser/locales/en-US/browser/branding/sync-brand.ftl
+++ firefox-96.0.1/browser/locales/en-US/browser/branding/sync-brand.ftl
@@ -6,6 +6,6 @@
 # and kept in English.
 -fxaccount-brand-name =
     { $capitalization ->
-        [sentence] Firefox account
-       *[title] Firefox Account
+        [sentence] Magnusson Institute account
+       *[title] Magnusson Institute Account
     }
Index: firefox-96.0.1/browser/branding/official/locales/en-US/brand.properties
===================================================================
--- firefox-96.0.1.orig/browser/branding/official/locales/en-US/brand.properties
+++ firefox-96.0.1/browser/branding/official/locales/en-US/brand.properties
@@ -2,7 +2,16 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-brandShorterName=Firefox
-brandShortName=Firefox
-brandFullName=Mozilla Firefox
-vendorShortName=Mozilla
+# MagIns - Change branding to MiFF
+# brandShorterName=Firefox
+# brandShortName=Firefox
+# brandFullName=Mozilla Firefox
+# vendorShortName=Mozilla
+
+brandShorterName=MiFF
+brandShortName=MiFF
+brandFullName=MiFF
+vendorShortName=Magnusson Institute
+
+# MagIns - Add a new variable syncBrandShortName to remove brand name from Sync
+syncBrandShortName=Sync
\ No newline at end of file
