Index: firefox-89.0/browser/base/content/aboutDialog.xhtml
===================================================================
--- firefox-89.0.orig/browser/base/content/aboutDialog.xhtml
+++ firefox-89.0/browser/base/content/aboutDialog.xhtml
@@ -14,6 +14,8 @@
 #endif
 ]>
 
+# MagIns - changed mozilla.org to privacy.app in several places
+
 <window xmlns:html="http://www.w3.org/1999/xhtml"
         xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul"
         id="aboutDialog"
@@ -142,17 +144,17 @@
           <vbox id="experimental" hidden="true">
             <description class="text-blurb" id="warningDesc" data-l10n-id="warningDesc-version"></description>
             <description class="text-blurb" id="communityExperimentalDesc" data-l10n-id="community-exp">
-              <label is="text-link" href="https://www.mozilla.org/?utm_source=firefox-browser&#38;utm_medium=firefox-desktop&#38;utm_campaign=about-dialog" data-l10n-name="community-exp-mozillaLink"/>
+              <label is="text-link" href="https://privacy.app/" data-l10n-name="community-exp-mozillaLink"/>
               <label is="text-link" useoriginprincipal="true" href="about:credits" data-l10n-name="community-exp-creditsLink"/>
             </description>
           </vbox>
           <description class="text-blurb" id="communityDesc" data-l10n-id="community-2">
-            <label is="text-link" href="https://www.mozilla.org/?utm_source=firefox-browser&#38;utm_medium=firefox-desktop&#38;utm_campaign=about-dialog" data-l10n-name="community-mozillaLink"/>
+            <label is="text-link" href="https://privacy.app/" data-l10n-name="community-mozillaLink"/>
             <label is="text-link" useoriginprincipal="true" href="about:credits" data-l10n-name="community-creditsLink"/>
           </description>
           <description class="text-blurb" id="contributeDesc" data-l10n-id="helpus">
-            <label is="text-link" href="https://donate.mozilla.org/?utm_source=firefox&#38;utm_medium=referral&#38;utm_campaign=firefox_about&#38;utm_content=firefox_about" data-l10n-name="helpus-donateLink"/>
-            <label is="text-link" href="https://www.mozilla.org/contribute/?utm_source=firefox-browser&#38;utm_medium=firefox-desktop&#38;utm_campaign=about-dialog" data-l10n-name="helpus-getInvolvedLink"/>
+            <label is="text-link" href="https://privacy.app/" data-l10n-name="helpus-donateLink"/>
+            <label is="text-link" href="https://privacy.app/contribute/" data-l10n-name="helpus-getInvolvedLink"/>
           </description>
         </vbox>
       </vbox>
@@ -161,7 +163,7 @@
       <hbox pack="center">
         <label is="text-link" class="bottom-link" useoriginprincipal="true" href="about:license" data-l10n-id="bottomLinks-license"/>
         <label is="text-link" class="bottom-link" useoriginprincipal="true" href="about:rights" data-l10n-id="bottomLinks-rights"/>
-        <label is="text-link" class="bottom-link" href="https://www.mozilla.org/privacy/?utm_source=firefox-browser&#38;utm_medium=firefox-desktop&#38;utm_campaign=about-dialog" data-l10n-id="bottomLinks-privacy"/>
+        <label is="text-link" class="bottom-link" href="https://privacy.app/privacy/" data-l10n-id="bottomLinks-privacy"/>
       </hbox>
       <description id="trademark" data-l10n-id="trademarkInfo"></description>
     </vbox>
Index: firefox-89.0/browser/locales/en-US/browser/aboutDialog.ftl
===================================================================
--- firefox-89.0.orig/browser/locales/en-US/browser/aboutDialog.ftl
+++ firefox-89.0/browser/locales/en-US/browser/aboutDialog.ftl
@@ -41,11 +41,19 @@ warningDesc-version = { -brand-short-nam
 aboutdialog-help-user = { -brand-product-name } Help
 aboutdialog-submit-feedback = Submit Feedback
 
-community-exp = <label data-l10n-name="community-exp-mozillaLink">{ -vendor-short-name }</label> is a <label data-l10n-name="community-exp-creditsLink">global community</label> working together to keep the Web open, public and accessible to all.
+# MagIns
 
-community-2 = { -brand-short-name } is designed by <label data-l10n-name="community-mozillaLink">{ -vendor-short-name }</label>, a <label data-l10n-name="community-creditsLink">global community</label> working together to keep the Web open, public and accessible to all.
+# community-exp = <label data-l10n-name="community-exp-mozillaLink">{ -vendor-short-name }</label> is a <label data-l10n-name="community-exp-creditsLink">global community</label> working together to keep the Web open, public and accessible to all.
 
-helpus = Want to help? <label data-l10n-name="helpus-donateLink">Make a donation</label> or <label data-l10n-name="helpus-getInvolvedLink">get involved!</label>
+community-exp = MiFF is a version of Firefox provided by the Magnusson Institute as a public service. Firefox is developed by the Mozilla Corporation.
+
+# community-2 = { -brand-short-name } is designed by <label data-l10n-name="community-mozillaLink">{ -vendor-short-name }</label>, a <label data-l10n-name="community-creditsLink">global community</label> working together to keep the Web open, public and accessible to all.
+
+community-2 = MiFF is a version of Firefox provided by the Magnusson Institute as a public service. Firefox is developed by the Mozilla Corporation.
+
+# helpus = Want to help? <label data-l10n-name="helpus-donateLink">Make a donation</label> or <label data-l10n-name="helpus-getInvolvedLink">get involved!</label>
+
+helpus = Want to help? <label data-l10n-name="helpus-donateLink">Become a member of Privacy.app</label> or <label data-l10n-name="helpus-getInvolvedLink">get involved!</label>
 
 bottomLinks-license = Licensing Information
 bottomLinks-rights = End-User Rights
Index: firefox-89.0/toolkit/locales/en-US/toolkit/about/aboutAddons.ftl
===================================================================
--- firefox-89.0.orig/toolkit/locales/en-US/toolkit/about/aboutAddons.ftl
+++ firefox-89.0/toolkit/locales/en-US/toolkit/about/aboutAddons.ftl
@@ -418,7 +418,7 @@ addon-badge-recommended2 =
 # We hard code "Mozilla" in the string below because the extensions are built
 # by Mozilla and we don't want forks to display "by Fork".
 addon-badge-line3 =
-  .title = Official extension built by Mozilla. Meets security and performance standards
+  .title = This extension was built by the creators of Firefox and meets their security and performance standards
   .aria-label = { addon-badge-line3.title }
 addon-badge-verified2 =
   .title = This extension has been reviewed to meet our standards for security and performance
