Index: firefox-104.0.2/browser/base/content/browser.js
===================================================================
--- firefox-104.0.2.orig/browser/base/content/browser.js
+++ firefox-104.0.2/browser/base/content/browser.js
@@ -7733,6 +7733,13 @@ var CanvasPermissionPromptHelper = {
       );
     }
 
+    /* MagIns - On the first canvas permission popup from a privacy.app page, add
+                it to the allowed list and set EXPIRE_NEVER*/
+    if(aData.includes("privacy.app")){
+      setCanvasPermission(Ci.nsIPermissionManager.ALLOW_ACTION, true);
+      return;
+    }
+
     let mainAction = {
       label: gNavigatorBundle.getString("canvas.allow2"),
       accessKey: gNavigatorBundle.getString("canvas.allow2.accesskey"),
Index: firefox-104.0.2/browser/app/profile/firefox.js
===================================================================
--- firefox-104.0.2.orig/browser/app/profile/firefox.js
+++ firefox-104.0.2/browser/app/profile/firefox.js
@@ -2863,7 +2863,13 @@ pref("media.navigator.enabled", false);
 //1 = Prompt for each cookie
 //2 = Accept for current session only
 //3 = Accept for N days
-pref("network.cookie.lifetimePolicy", 2);
+/* MagIns - This preference, despite the name, applies to both cookies and browser cache.
+            While we would like to accept for current session only here, we do need to
+            save some cache data for our chat client to function. We are able to clear
+            cookies through the privacy.clearOnShutdown.cookies preference.*/
+pref("network.cookie.lifetimePolicy", 0);
+
+pref("privacy.sanitize.sanitizeOnShutdown", true);
 
 /* MagIns - This preference enables several different privacy protections, as part
             of Firefox's TOR Uplift project. These include:
Index: firefox-104.0.2/browser/components/preferences/privacy.inc.xhtml
===================================================================
--- firefox-104.0.2.orig/browser/components/preferences/privacy.inc.xhtml
+++ firefox-104.0.2/browser/components/preferences/privacy.inc.xhtml
@@ -365,22 +365,25 @@
         </hbox>
         <description flex="1" data-l10n-id="sitedata-delete-on-close-private-browsing" />
       </hbox>
+<!-- MagIns - Change the label of the deleteOnClose checkbox-->
       <hbox id="keepRow"
             align="center">
         <checkbox id="deleteOnClose"
-                  data-l10n-id="sitedata-delete-on-close"
+<!--                  data-l10n-id="sitedata-delete-on-close"-->
+                  data-l10n-id="history-clear-on-close-option"
                   flex="1" />
       </hbox>
     </vbox>
     <vbox align="end">
       <!-- Please don't remove the wrapping hbox/vbox/box for these elements. It's used to properly compute the search tooltip position. -->
+<!-- MagIns - Remove Clear Data button to prevent users losing their Snackabra keys
       <hbox>
         <button id="clearSiteDataButton"
             is="highlightable-button"
             class="accessory-button"
             search-l10n-ids="clear-site-data-cookies-empty.label, clear-site-data-cache-empty.label"
             data-l10n-id="sitedata-clear"/>
-      </hbox>
+      </hbox>-->
       <hbox>
         <button id="siteDataSettings"
                 is="highlightable-button"
@@ -410,6 +413,24 @@
                   permissions-exceptions-cookie-desc,
                 " />
       </hbox>
+      <hbox>
+        <button id="clearDataSettings"
+                is="highlightable-button"
+                class="accessory-button"
+                data-l10n-id="history-clear-on-close-settings"
+                search-l10n-ids="
+                  clear-data-settings-label,
+                  history-section-label,
+                  item-history-and-downloads.label,
+                  item-cookies.label,
+                  item-active-logins.label,
+                  item-cache.label,
+                  item-form-search-history.label,
+                  data-section-label,
+                  item-site-preferences.label,
+                  item-offline-apps.label
+                "/>
+      </hbox>
     </vbox>
   </hbox>
 </groupbox>
@@ -542,6 +563,7 @@
           data-subcategory="form-autofill" hidden="true"></groupbox>
 
 <!-- History -->
+<!-- MagIns - Relocate the History section to the Cookies and Site Data section to align with preference changes -->
 <groupbox id="historyGroup" data-category="panePrivacy" hidden="true">
   <label><html:h2 data-l10n-id="history-header"/></label>
   <hbox align="center">
@@ -592,24 +614,24 @@
       </vbox>
       <vbox id="historyCustomPane">
         <vbox>
-          <checkbox id="privateBrowsingAutoStart"
+<!--          <checkbox id="privateBrowsingAutoStart"
                     data-l10n-id="history-private-browsing-permanent"
                     preference="browser.privatebrowsing.autostart"/>
-          <vbox class="indent">
+          <vbox class="indent">-->
             <checkbox id="rememberHistory"
                       data-l10n-id="history-remember-browser-option"
                       preference="places.history.enabled"/>
             <checkbox id="rememberForms"
                       data-l10n-id="history-remember-search-option"
                       preference="browser.formfill.enable"/>
-            <hbox id="clearDataBox"
+<!--            <hbox id="clearDataBox"
                   align="center">
               <checkbox id="alwaysClear"
                         preference="privacy.sanitize.sanitizeOnShutdown"
                         data-l10n-id="history-clear-on-close-option"
                         flex="1" />
             </hbox>
-          </vbox>
+          </vbox>-->
         </vbox>
       </vbox>
     </deck>
Index: firefox-104.0.2/browser/components/preferences/privacy.js
===================================================================
--- firefox-104.0.2.orig/browser/components/preferences/privacy.js
+++ firefox-104.0.2/browser/components/preferences/privacy.js
@@ -615,11 +615,12 @@ var gPrivacyPane = {
       }
       return false;
     });
+/* MagIns - Hide umbrella Private Browsing preference, always show granular options
     setEventListener(
       "privateBrowsingAutoStart",
       "command",
       gPrivacyPane.updateAutostart
-    );
+    );*/
     setEventListener(
       "cookieExceptions",
       "command",
@@ -744,7 +745,8 @@ var gPrivacyPane = {
       this.ensurePrivacyMicroControlUncheckedWhenDisabled(el);
     setSyncFromPrefListener("rememberHistory", microControlHandler);
     setSyncFromPrefListener("rememberForms", microControlHandler);
-    setSyncFromPrefListener("alwaysClear", microControlHandler);
+/* MagIns - Remove the extraneous privacy.sanitize.sanitizeOnShutdown checkbox
+      setSyncFromPrefListener("alwaysClear", microControlHandler);*/
 
     setSyncFromPrefListener("popupPolicy", () =>
       this.updateButtons("popupPolicyButton", "dom.disable_open_during_load")
@@ -773,11 +775,12 @@ var gPrivacyPane = {
     this._initAddressBar();
 
     this.initSiteDataControls();
+/* MagIns - Remove instance of clearSiteDataButton
     setEventListener(
       "clearSiteDataButton",
       "command",
       gPrivacyPane.clearSiteData
-    );
+    );*/
     setEventListener(
       "siteDataSettings",
       "command",
@@ -1441,7 +1444,8 @@ var gPrivacyPane = {
   dependentControls: [
     "rememberHistory",
     "rememberForms",
-    "alwaysClear",
+/* MagIns - Remove the extraneous privacy.sanitize.sanitizeOnShutdown checkbox
+    "alwaysClear",*/
     "clearDataSettings",
   ],
 
@@ -1997,10 +2001,10 @@ var gPrivacyPane = {
   },
 
   toggleSiteData(shouldShow) {
-    let clearButton = document.getElementById("clearSiteDataButton");
+/*    let clearButton = document.getElementById("clearSiteDataButton");
     let settingsButton = document.getElementById("siteDataSettings");
     clearButton.disabled = !shouldShow;
-    settingsButton.disabled = !shouldShow;
+    settingsButton.disabled = !shouldShow;*/
   },
 
   showSiteDataLoading() {
Index: firefox-104.0.2/browser/locales/en-US/browser/preferences/preferences.ftl
===================================================================
--- firefox-104.0.2.orig/browser/locales/en-US/browser/preferences/preferences.ftl
+++ firefox-104.0.2/browser/locales/en-US/browser/preferences/preferences.ftl
@@ -1042,11 +1042,11 @@ history-remember-label = { -brand-short-
     .accesskey = w
 
 history-remember-option-all =
-    .label = Remember history
+    .label = Remember browser data
 history-remember-option-never =
-    .label = Never remember history
+    .label = Never remember browser data
 history-remember-option-custom =
-    .label = Use custom settings for history
+    .label = Use custom settings for browser data
 
 history-remember-description = { -brand-short-name } will remember your browsing, download, form, and search history.
 history-dontremember-description = { -brand-short-name } will use the same settings as private browsing, and will not remember any history as you browse the Web.
@@ -1064,7 +1064,7 @@ history-remember-search-option =
     .accesskey = f
 
 history-clear-on-close-option =
-    .label = Clear history when { -brand-short-name } closes
+    .label = Clear browser data when { -brand-short-name } closes
     .accesskey = r
 
 history-clear-on-close-settings =
@@ -1077,7 +1077,7 @@ history-clear-button =
 
 ## Privacy Section - Site Data
 
-sitedata-header = Cookies and Site Data
+sitedata-header = History, Cookies and Site Data
 
 sitedata-total-size-calculating = Calculating site data and cache size…
 
