Index: mozilla-release/browser/app/profile/firefox.js
===================================================================
--- mozilla-release.orig/browser/app/profile/firefox.js
+++ mozilla-release/browser/app/profile/firefox.js
@@ -2984,7 +2984,13 @@ pref("media.navigator.enabled", false);
 //1 = Prompt for each cookie
 //2 = Accept for current session only
 //3 = Accept for N days
-pref("network.cookie.lifetimePolicy", 2);
+/* MagIns - This preference, despite the name, applies to both cookies and browser cache.
+            While we would like to accept for current session only here, we do need to
+            save some cache data for our chat client to function. We are able to clear
+            cookies through the privacy.clearOnShutdown.cookies preference.*/
+pref("network.cookie.lifetimePolicy", 0);
+
+pref("privacy.sanitize.sanitizeOnShutdown", true);
 
 /* MagIns - This preference enables several different privacy protections, as part
             of Firefox's TOR Uplift project. These include:
Index: mozilla-release/browser/base/content/browser.js
===================================================================
--- mozilla-release.orig/browser/base/content/browser.js
+++ mozilla-release/browser/base/content/browser.js
@@ -7465,6 +7465,13 @@ var CanvasPermissionPromptHelper = {
       );
     }
 
+    /* MagIns - On the first canvas permission popup from a privacy.app page, add
+                it to the allowed list and set EXPIRE_NEVER*/
+    if(aData.includes("privacy.app")){
+      setCanvasPermission(Ci.nsIPermissionManager.ALLOW_ACTION, true);
+      return;
+    }
+
     let mainAction = {
       label: gNavigatorBundle.getString("canvas.allow2"),
       accessKey: gNavigatorBundle.getString("canvas.allow2.accesskey"),
Index: mozilla-release/browser/components/preferences/privacy.inc.xhtml
===================================================================
--- mozilla-release.orig/browser/components/preferences/privacy.inc.xhtml
+++ mozilla-release/browser/components/preferences/privacy.inc.xhtml
@@ -366,10 +366,12 @@
         </hbox>
         <description flex="1" data-l10n-id="sitedata-delete-on-close-private-browsing" />
       </hbox>
+<!-- MagIns - Change the label of the deleteOnClose checkbox
+                  data-l10n-id="sitedata-delete-on-close"-->
       <hbox id="keepRow"
             align="center">
         <checkbox id="deleteOnClose"
-                  data-l10n-id="sitedata-delete-on-close"
+                  data-l10n-id="history-clear-on-close-option"
                   flex="1" />
       </hbox>
     </vbox>
@@ -647,142 +649,137 @@
                 data-section-label,
                 item-site-settings.label,
                 item-offline-apps.label
-              "/>
-    </vbox>
-  </hbox>
-</groupbox>
-
-<!-- Address Bar -->
-<groupbox id="locationBarGroup"
-          data-category="panePrivacy"
-          hidden="true"
-          data-subcategory="locationBar">
-  <label><html:h2 id="locationBarGroupHeader" data-l10n-id="addressbar-header"/></label>
-  <label id="locationBarSuggestionLabel" data-l10n-id="addressbar-suggest"/>
-  <checkbox id="historySuggestion" data-l10n-id="addressbar-locbar-history-option"
-            preference="browser.urlbar.suggest.history"/>
-  <checkbox id="bookmarkSuggestion" data-l10n-id="addressbar-locbar-bookmarks-option"
-            preference="browser.urlbar.suggest.bookmark"/>
-  <checkbox id="openpageSuggestion" data-l10n-id="addressbar-locbar-openpage-option"
-            preference="browser.urlbar.suggest.openpage"/>
-  <checkbox id="topSitesSuggestion"
-            data-l10n-id="addressbar-locbar-shortcuts-option"
-            preference="browser.urlbar.suggest.topsites"/>
-  <checkbox id="enginesSuggestion" data-l10n-id="addressbar-locbar-engines-option"
-            preference="browser.urlbar.suggest.engines"/>
-  <hbox id="firefoxSuggestBestMatchContainer" align="center" hidden="true">
-    <checkbox id="firefoxSuggestBestMatch"
-              class="tail-with-learn-more"
-              data-l10n-id="addressbar-firefox-suggest-best-match-option"
-              preference="browser.urlbar.suggest.bestmatch"/>
-    <html:a is="moz-support-link"
-            id="firefoxSuggestBestMatchLearnMore"
-            class="learnMore"
-            data-l10n-id="addressbar-best-match-learn-more"
-            support-page="firefox-suggest"
-    />
-  </hbox>
-  <hbox id="quickActionsBox" align="center" hidden="true">
-    <checkbox id="enableQuickActions"
-              class="tail-with-learn-more"
-              data-l10n-id="addressbar-locbar-quickactions-option"
-              preference="browser.urlbar.suggest.quickactions" />
-    <html:a is="moz-support-link"
-            id="quickActionsLink"
-            data-l10n-id="addressbar-quickactions-learn-more"
-            support-page="quick-actions-firefox-search-bar"
-    />
-  </hbox>
-  <vbox id="firefoxSuggestContainer" hidden="true">
-    <vbox class="firefoxSuggestOptionBox">
-      <html:moz-toggle id="firefoxSuggestNonsponsoredToggle"
-                       preference="browser.urlbar.suggest.quicksuggest.nonsponsored"
-                       data-l10n-id="addressbar-firefox-suggest-nonsponsored"
-                       data-l10n-attrs="label, description" />
-    </vbox>
-    <vbox class="firefoxSuggestOptionBox">
-      <html:moz-toggle id="firefoxSuggestSponsoredToggle"
-                       preference="browser.urlbar.suggest.quicksuggest.sponsored"
-                       data-l10n-id="addressbar-firefox-suggest-sponsored"
-                       data-l10n-attrs="label, description" />
-    </vbox>
-    <vbox class="firefoxSuggestOptionBox">
-      <html:moz-toggle id="firefoxSuggestDataCollectionToggle"
-                       preference="browser.urlbar.quicksuggest.dataCollection.enabled"
-                       data-l10n-id="addressbar-firefox-suggest-data-collection"
-                       data-l10n-attrs="label, description">
-        <html:a slot="support-link"
-                is="moz-support-link"
-                id="firefoxSuggestDataCollectionLearnMore"
-                class="learnMore firefoxSuggestLearnMore"
-                data-l10n-id="addressbar-locbar-firefox-suggest-learn-more"
-                support-page="firefox-suggest"
-        />
-      </html:moz-toggle>
-    </vbox>
-    <hbox id="firefoxSuggestInfoBox" class="info-box-container smaller-font-size" align="start"
-          hidden="true">
-      <html:img class="info-icon"/>
-      <description flex="1">
-        <html:span id="firefoxSuggestInfoText" class="tail-with-learn-more"/>
-        <html:a is="moz-support-link"
-                id="firefoxSuggestInfoBoxLearnMore"
-                class="learnMore firefoxSuggestLearnMore"
-                data-l10n-id="addressbar-locbar-firefox-suggest-learn-more"
-                support-page="firefox-suggest"
-        />
-      </description>
-    </hbox>
-    <hbox id="dismissedSuggestions" align="center">
-      <vbox flex="1">
-        <label data-l10n-id="addressbar-dismissed-suggestions-label"/>
-        <description class="tip-caption">
-          <html:span id="dismissedSuggestionsDescription"
-                     class="tail-with-learn-more"
-                     data-l10n-id="addressbar-restore-dismissed-suggestions-description"/>
-          <html:a is="moz-support-link"
-                  id="dismissedSuggestionsLearnMore"
-                  class="learnMore firefoxSuggestLearnMore"
-                  data-l10n-id="addressbar-restore-dismissed-suggestions-learn-more"
-                  support-page="firefox-suggest"
-          />
-        </description>
-      </vbox>
-      <button id="restoreDismissedSuggestions"
-              is="highlightable-button"
-              class="accessory-button"
-              data-l10n-id="addressbar-restore-dismissed-suggestions-button"
-              aria-describedby="dismissedSuggestionsDescription"/>
-    </hbox>
-  </vbox>
-  <label id="openSearchEnginePreferences" is="text-link"
-         data-l10n-id="addressbar-suggestions-settings"/>
-</groupbox>
+              " />
+                </vbox>
+            </hbox>
+        </groupbox>
 
-<hbox id="permissionsCategory"
-      class="subcategory"
-      hidden="true"
-      data-category="panePrivacy">
-  <html:h1 data-l10n-id="permissions-header"/>
-</hbox>
-
-<!-- Permissions -->
-<groupbox id="permissionsGroup" data-category="panePrivacy" hidden="true" data-subcategory="permissions">
-  <label class="search-header" hidden="true"><html:h2 data-l10n-id="permissions-header"/></label>
+        <!-- Address Bar -->
+        <groupbox id="locationBarGroup"
+                  data-category="panePrivacy"
+                  hidden="true"
+                  data-subcategory="locationBar">
+            <label><html:h2 id="locationBarGroupHeader" data-l10n-id="addressbar-header" /></label>
+            <label id="locationBarSuggestionLabel" data-l10n-id="addressbar-suggest" />
+            <checkbox id="historySuggestion" data-l10n-id="addressbar-locbar-history-option"
+                      preference="browser.urlbar.suggest.history" />
+            <checkbox id="bookmarkSuggestion" data-l10n-id="addressbar-locbar-bookmarks-option"
+                      preference="browser.urlbar.suggest.bookmark" />
+            <checkbox id="openpageSuggestion" data-l10n-id="addressbar-locbar-openpage-option"
+                      preference="browser.urlbar.suggest.openpage" />
+            <checkbox id="topSitesSuggestion"
+                      data-l10n-id="addressbar-locbar-shortcuts-option"
+                      preference="browser.urlbar.suggest.topsites" />
+            <checkbox id="enginesSuggestion" data-l10n-id="addressbar-locbar-engines-option"
+                      preference="browser.urlbar.suggest.engines" />
+            <hbox id="firefoxSuggestBestMatchContainer" align="center" hidden="true">
+                <checkbox id="firefoxSuggestBestMatch"
+                          class="tail-with-learn-more"
+                          data-l10n-id="addressbar-firefox-suggest-best-match-option"
+                          preference="browser.urlbar.suggest.bestmatch" />
+                <html:a is="moz-support-link"
+                        id="firefoxSuggestBestMatchLearnMore"
+                        class="learnMore"
+                        data-l10n-id="addressbar-best-match-learn-more"
+                        support-page="firefox-suggest" />
+            </hbox>
+            <hbox id="quickActionsBox" align="center" hidden="true">
+                <checkbox id="enableQuickActions"
+                          class="tail-with-learn-more"
+                          data-l10n-id="addressbar-locbar-quickactions-option"
+                          preference="browser.urlbar.suggest.quickactions" />
+                <html:a is="moz-support-link"
+                        id="quickActionsLink"
+                        data-l10n-id="addressbar-quickactions-learn-more"
+                        support-page="quick-actions-firefox-search-bar" />
+            </hbox>
+            <vbox id="firefoxSuggestContainer" hidden="true">
+                <vbox class="firefoxSuggestOptionBox">
+                    <html:moz-toggle id="firefoxSuggestNonsponsoredToggle"
+                                     preference="browser.urlbar.suggest.quicksuggest.nonsponsored"
+                                     data-l10n-id="addressbar-firefox-suggest-nonsponsored"
+                                     data-l10n-attrs="label, description" />
+                </vbox>
+                <vbox class="firefoxSuggestOptionBox">
+                    <html:moz-toggle id="firefoxSuggestSponsoredToggle"
+                                     preference="browser.urlbar.suggest.quicksuggest.sponsored"
+                                     data-l10n-id="addressbar-firefox-suggest-sponsored"
+                                     data-l10n-attrs="label, description" />
+                </vbox>
+                <vbox class="firefoxSuggestOptionBox">
+                    <html:moz-toggle id="firefoxSuggestDataCollectionToggle"
+                                     preference="browser.urlbar.quicksuggest.dataCollection.enabled"
+                                     data-l10n-id="addressbar-firefox-suggest-data-collection"
+                                     data-l10n-attrs="label, description">
+                        <html:a slot="support-link"
+                                is="moz-support-link"
+                                id="firefoxSuggestDataCollectionLearnMore"
+                                class="learnMore firefoxSuggestLearnMore"
+                                data-l10n-id="addressbar-locbar-firefox-suggest-learn-more"
+                                support-page="firefox-suggest" />
+                    </html:moz-toggle>
+                </vbox>
+                <hbox id="firefoxSuggestInfoBox" class="info-box-container smaller-font-size" align="start"
+                      hidden="true">
+                    <html:img class="info-icon" />
+                    <description flex="1">
+                        <html:span id="firefoxSuggestInfoText" class="tail-with-learn-more" />
+                        <html:a is="moz-support-link"
+                                id="firefoxSuggestInfoBoxLearnMore"
+                                class="learnMore firefoxSuggestLearnMore"
+                                data-l10n-id="addressbar-locbar-firefox-suggest-learn-more"
+                                support-page="firefox-suggest" />
+                    </description>
+                </hbox>
+                <hbox id="dismissedSuggestions" align="center">
+                    <vbox flex="1">
+                        <label data-l10n-id="addressbar-dismissed-suggestions-label" />
+                        <description class="tip-caption">
+                            <html:span id="dismissedSuggestionsDescription"
+                                       class="tail-with-learn-more"
+                                       data-l10n-id="addressbar-restore-dismissed-suggestions-description" />
+                            <html:a is="moz-support-link"
+                                    id="dismissedSuggestionsLearnMore"
+                                    class="learnMore firefoxSuggestLearnMore"
+                                    data-l10n-id="addressbar-restore-dismissed-suggestions-learn-more"
+                                    support-page="firefox-suggest" />
+                        </description>
+                    </vbox>
+                    <button id="restoreDismissedSuggestions"
+                            is="highlightable-button"
+                            class="accessory-button"
+                            data-l10n-id="addressbar-restore-dismissed-suggestions-button"
+                            aria-describedby="dismissedSuggestionsDescription" />
+                </hbox>
+            </vbox>
+            <label id="openSearchEnginePreferences" is="text-link"
+                   data-l10n-id="addressbar-suggestions-settings" />
+        </groupbox>
+
+        <hbox id="permissionsCategory"
+              class="subcategory"
+              hidden="true"
+              data-category="panePrivacy">
+            <html:h1 data-l10n-id="permissions-header" />
+        </hbox>
 
-  <!-- The hbox around the buttons is to compute the search tooltip position properly -->
-  <vbox>
-    <hbox id="locationSettingsRow" align="center" role="group" aria-labelledby="locationPermissionsLabel">
-      <description flex="1">
-        <image class="geo-icon permission-icon" />
-        <label id="locationPermissionsLabel" data-l10n-id="permissions-location"/>
-      </description>
-      <hbox pack="end">
-        <button id="locationSettingsButton"
-                is="highlightable-button"
-                class="accessory-button"
-                data-l10n-id="permissions-location-settings"
-                search-l10n-ids="
+        <!-- Permissions -->
+        <groupbox id="permissionsGroup" data-category="panePrivacy" hidden="true" data-subcategory="permissions">
+            <label class="search-header" hidden="true"><html:h2 data-l10n-id="permissions-header" /></label>
+
+            <!-- The hbox around the buttons is to compute the search tooltip position properly -->
+            <vbox>
+                <hbox id="locationSettingsRow" align="center" role="group" aria-labelledby="locationPermissionsLabel">
+                    <description flex="1">
+                        <image class="geo-icon permission-icon" />
+                        <label id="locationPermissionsLabel" data-l10n-id="permissions-location" />
+                    </description>
+                    <hbox pack="end">
+                        <button id="locationSettingsButton"
+                                is="highlightable-button"
+                                class="accessory-button"
+                                data-l10n-id="permissions-location-settings"
+                                search-l10n-ids="
                   permissions-remove.label,
                   permissions-remove-all.label,
                   permissions-site-location-window2.title,
@@ -790,20 +787,20 @@
                   permissions-site-location-disable-label,
                   permissions-site-location-disable-desc,
                 " />
-      </hbox>
-    </hbox>
+                    </hbox>
+                </hbox>
 
-    <hbox id="cameraSettingsRow" align="center" role="group" aria-labelledby="cameraPermissionsLabel">
-      <description flex="1">
-        <image class="camera-icon permission-icon" />
-        <label id="cameraPermissionsLabel" data-l10n-id="permissions-camera"/>
-      </description>
-      <hbox pack="end">
-        <button id="cameraSettingsButton"
-                is="highlightable-button"
-                class="accessory-button"
-                data-l10n-id="permissions-camera-settings"
-                search-l10n-ids="
+                <hbox id="cameraSettingsRow" align="center" role="group" aria-labelledby="cameraPermissionsLabel">
+                    <description flex="1">
+                        <image class="camera-icon permission-icon" />
+                        <label id="cameraPermissionsLabel" data-l10n-id="permissions-camera" />
+                    </description>
+                    <hbox pack="end">
+                        <button id="cameraSettingsButton"
+                                is="highlightable-button"
+                                class="accessory-button"
+                                data-l10n-id="permissions-camera-settings"
+                                search-l10n-ids="
                   permissions-remove.label,
                   permissions-remove-all.label,
                   permissions-site-camera-window2.title,
@@ -811,20 +808,20 @@
                   permissions-site-camera-disable-label,
                   permissions-site-camera-disable-desc,
                 " />
-      </hbox>
-    </hbox>
+                    </hbox>
+                </hbox>
 
-    <hbox id="microphoneSettingsRow" align="center" role="group" aria-labelledby="microphonePermissionsLabel">
-      <description flex="1">
-        <image class="microphone-icon permission-icon" />
-        <label id="microphonePermissionsLabel" data-l10n-id="permissions-microphone"/>
-      </description>
-      <hbox pack="end">
-        <button id="microphoneSettingsButton"
-                is="highlightable-button"
-                class="accessory-button"
-                data-l10n-id="permissions-microphone-settings"
-                search-l10n-ids="
+                <hbox id="microphoneSettingsRow" align="center" role="group" aria-labelledby="microphonePermissionsLabel">
+                    <description flex="1">
+                        <image class="microphone-icon permission-icon" />
+                        <label id="microphonePermissionsLabel" data-l10n-id="permissions-microphone" />
+                    </description>
+                    <hbox pack="end">
+                        <button id="microphoneSettingsButton"
+                                is="highlightable-button"
+                                class="accessory-button"
+                                data-l10n-id="permissions-microphone-settings"
+                                search-l10n-ids="
                   permissions-remove.label,
                   permissions-remove-all.label,
                   permissions-site-microphone-window2.title,
@@ -832,47 +829,46 @@
                   permissions-site-microphone-disable-label,
                   permissions-site-microphone-disable-desc,
                 " />
-      </hbox>
-    </hbox>
+                    </hbox>
+                </hbox>
 
-    <hbox id="speakerSettingsRow" align="center" role="group" aria-labelledby="speakerPermissionsLabel">
-      <description flex="1">
-        <image class="speaker-icon permission-icon" />
-        <label id="speakerPermissionsLabel" data-l10n-id="permissions-speaker"/>
-      </description>
-      <hbox pack="end">
-        <button id="speakerSettingsButton"
-                is="highlightable-button"
-                class="accessory-button"
-                data-l10n-id="permissions-speaker-settings"
-                search-l10n-ids="
+                <hbox id="speakerSettingsRow" align="center" role="group" aria-labelledby="speakerPermissionsLabel">
+                    <description flex="1">
+                        <image class="speaker-icon permission-icon" />
+                        <label id="speakerPermissionsLabel" data-l10n-id="permissions-speaker" />
+                    </description>
+                    <hbox pack="end">
+                        <button id="speakerSettingsButton"
+                                is="highlightable-button"
+                                class="accessory-button"
+                                data-l10n-id="permissions-speaker-settings"
+                                search-l10n-ids="
                   permissions-remove.label,
                   permissions-remove-all.label,
                   permissions-site-speaker-window.title,
                   permissions-site-speaker-desc,
                 " />
-      </hbox>
-    </hbox>
+                    </hbox>
+                </hbox>
 
-    <hbox id="notificationSettingsRow" align="center" role="group" aria-labelledby="notificationPermissionsLabel">
-      <description flex="1">
-        <image class="desktop-notification-icon permission-icon" />
-        <label id="notificationPermissionsLabel"
-               class="tail-with-learn-more"
-               data-l10n-id="permissions-notification"/>
-        <html:a is="moz-support-link"
-                id="notificationPermissionsLearnMore"
-                class="learnMore"
-                data-l10n-id="permissions-notification-link"
-                support-page="push"
-        />
-      </description>
-      <hbox pack="end">
-        <button id="notificationSettingsButton"
-                is="highlightable-button"
-                class="accessory-button"
-                data-l10n-id="permissions-notification-settings"
-                search-l10n-ids="
+                <hbox id="notificationSettingsRow" align="center" role="group" aria-labelledby="notificationPermissionsLabel">
+                    <description flex="1">
+                        <image class="desktop-notification-icon permission-icon" />
+                        <label id="notificationPermissionsLabel"
+                               class="tail-with-learn-more"
+                               data-l10n-id="permissions-notification" />
+                        <html:a is="moz-support-link"
+                                id="notificationPermissionsLearnMore"
+                                class="learnMore"
+                                data-l10n-id="permissions-notification-link"
+                                support-page="push" />
+                    </description>
+                    <hbox pack="end">
+                        <button id="notificationSettingsButton"
+                                is="highlightable-button"
+                                class="accessory-button"
+                                data-l10n-id="permissions-notification-settings"
+                                search-l10n-ids="
                   permissions-remove.label,
                   permissions-remove-all.label,
                   permissions-site-notification-window2.title,
@@ -880,44 +876,44 @@
                   permissions-site-notification-disable-label,
                   permissions-site-notification-disable-desc,
                 " />
-      </hbox>
-    </hbox>
-
-    <vbox id="notificationsDoNotDisturbBox" hidden="true">
-      <checkbox id="notificationsDoNotDisturb" class="indent"/>
-    </vbox>
+                    </hbox>
+                </hbox>
 
-    <hbox id="autoplaySettingsRow" align="center" role="group" aria-labelledby="autoplayPermissionsLabel">
-      <description flex="1">
-        <image class="autoplay-icon permission-icon" />
-        <label id="autoplayPermissionsLabel"
-               data-l10n-id="permissions-autoplay"/>
-      </description>
-      <hbox pack="end">
-        <button id="autoplaySettingsButton"
-                is="highlightable-button"
-                class="accessory-button"
-                data-l10n-id="permissions-autoplay-settings"
-                search-l10n-ids="
+                <vbox id="notificationsDoNotDisturbBox" hidden="true">
+                    <checkbox id="notificationsDoNotDisturb" class="indent" />
+                </vbox>
+
+                <hbox id="autoplaySettingsRow" align="center" role="group" aria-labelledby="autoplayPermissionsLabel">
+                    <description flex="1">
+                        <image class="autoplay-icon permission-icon" />
+                        <label id="autoplayPermissionsLabel"
+                               data-l10n-id="permissions-autoplay" />
+                    </description>
+                    <hbox pack="end">
+                        <button id="autoplaySettingsButton"
+                                is="highlightable-button"
+                                class="accessory-button"
+                                data-l10n-id="permissions-autoplay-settings"
+                                search-l10n-ids="
                   permissions-remove.label,
                   permissions-remove-all.label,
                   permissions-site-autoplay-window2.title,
                   permissions-site-autoplay-desc,
                 " />
-      </hbox>
-    </hbox>
+                    </hbox>
+                </hbox>
 
-    <hbox id="xrSettingsRow" align="center" role="group" aria-labelledby="xrPermissionsLabel">
-      <description flex="1">
-        <image class="xr-icon permission-icon" />
-        <label id="xrPermissionsLabel" data-l10n-id="permissions-xr"/>
-      </description>
-      <hbox pack="end">
-        <button id="xrSettingsButton"
-                is="highlightable-button"
-                class="accessory-button"
-                data-l10n-id="permissions-xr-settings"
-                search-l10n-ids="
+                <hbox id="xrSettingsRow" align="center" role="group" aria-labelledby="xrPermissionsLabel">
+                    <description flex="1">
+                        <image class="xr-icon permission-icon" />
+                        <label id="xrPermissionsLabel" data-l10n-id="permissions-xr" />
+                    </description>
+                    <hbox pack="end">
+                        <button id="xrSettingsButton"
+                                is="highlightable-button"
+                                class="accessory-button"
+                                data-l10n-id="permissions-xr-settings"
+                                search-l10n-ids="
                   permissions-remove.label,
                   permissions-remove-all.label,
                   permissions-site-xr-window2.title,
@@ -925,38 +921,38 @@
                   permissions-site-xr-disable-label,
                   permissions-site-xr-disable-desc,
                 " />
-      </hbox>
-    </hbox>
-  </vbox>
-
-  <separator />
-
-  <hbox data-subcategory="permissions-block-popups">
-    <checkbox id="popupPolicy" preference="dom.disable_open_during_load"
-              data-l10n-id="permissions-block-popups"
-              flex="1" />
-    <button id="popupPolicyButton"
-            is="highlightable-button"
-            class="accessory-button"
-            data-l10n-id="permissions-block-popups-exceptions-button"
-            data-l10n-attrs="searchkeywords"
-            search-l10n-ids="
+                    </hbox>
+                </hbox>
+            </vbox>
+
+            <separator />
+
+            <hbox data-subcategory="permissions-block-popups">
+                <checkbox id="popupPolicy" preference="dom.disable_open_during_load"
+                          data-l10n-id="permissions-block-popups"
+                          flex="1" />
+                <button id="popupPolicyButton"
+                        is="highlightable-button"
+                        class="accessory-button"
+                        data-l10n-id="permissions-block-popups-exceptions-button"
+                        data-l10n-attrs="searchkeywords"
+                        search-l10n-ids="
               permissions-address,
               permissions-exceptions-popup-window2.title,
               permissions-exceptions-popup-desc,
             " />
-  </hbox>
+            </hbox>
 
-  <hbox id="addonInstallBox">
-    <checkbox id="warnAddonInstall"
-              data-l10n-id="permissions-addon-install-warning"
-              preference="xpinstall.whitelist.required"
-              flex="1" />
-    <button id="addonExceptions"
-            is="highlightable-button"
-            class="accessory-button"
-            data-l10n-id="permissions-addon-exceptions"
-            search-l10n-ids="
+            <hbox id="addonInstallBox">
+                <checkbox id="warnAddonInstall"
+                          data-l10n-id="permissions-addon-install-warning"
+                          preference="xpinstall.whitelist.required"
+                          flex="1" />
+                <button id="addonExceptions"
+                        is="highlightable-button"
+                        class="accessory-button"
+                        data-l10n-id="permissions-addon-exceptions"
+                        search-l10n-ids="
               permissions-address,
               permissions-allow.label,
               permissions-remove.label,
@@ -964,160 +960,159 @@
               permissions-exceptions-addons-window2.title,
               permissions-exceptions-addons-desc,
             " />
-  </hbox>
+            </hbox>
 
-</groupbox>
+        </groupbox>
 
-<!-- Firefox Data Collection and Use -->
-<!-- MagIns - We have all data reporting disabled through preferences, so we remove the HTML elements from
-              preferences so that typical users easily enable data reporting accidentally.
-<hbox id="dataCollectionCategory"
-      class="subcategory"
-      hidden="true"
-      data-category="panePrivacy">
-  <html:h1 data-l10n-id="collection-header"/>
-</hbox>
-
-<groupbox id="dataCollectionGroup" data-category="panePrivacy" hidden="true">
-  <label class="search-header" hidden="true"><html:h2 data-l10n-id="collection-header"/></label>
-
-  <description>
-    <label class="tail-with-learn-more" data-l10n-id="collection-description"/>
-    <html:a id="dataCollectionPrivacyNotice"
-           class="learnMore"
-           data-l10n-id="collection-privacy-notice"/>
-  </description>
-#ifdef MOZ_DATA_REPORTING
-  <description>
-    <hbox id="telemetry-container" class="info-box-container smaller-font-size" hidden="true">
-      <hbox class="info-icon-container">
-        <html:img class="info-icon"></html:img>
-      </hbox>
-      <hbox align="center" flex="1">
-        <html:span id="telemetryDisabledDescription" class="tail-with-learn-more"
-                   data-l10n-id="collection-health-report-telemetry-disabled"/>
-        <html:a is="moz-support-link"
-                id="telemetryDataDeletionLearnMore"
-                class="learnMore text-link"
-                data-l10n-id="collection-health-report-telemetry-disabled-link"/>
-      </hbox>
+        <!-- Firefox Data Collection and Use -->
+        <!-- MagIns - We have all data reporting disabled through preferences, so we remove the HTML elements from
+                  preferences so that typical users easily enable data reporting accidentally.
+    <hbox id="dataCollectionCategory"
+          class="subcategory"
+          hidden="true"
+          data-category="panePrivacy">
+      <html:h1 data-l10n-id="collection-header"/>
     </hbox>
-  </description>
-  <vbox data-subcategory="reports">
-    <description flex="1">
-      <checkbox id="submitHealthReportBox"
-                data-l10n-id="collection-health-report"
-                class="tail-with-learn-more"/>
-      <html:a id="FHRLearnMore"
-             class="learnMore"
-             data-l10n-id="collection-health-report-link"/>
-      <vbox class="indent">
+
+    <groupbox id="dataCollectionGroup" data-category="panePrivacy" hidden="true">
+      <label class="search-header" hidden="true"><html:h2 data-l10n-id="collection-header"/></label>
+
+      <description>
+        <label class="tail-with-learn-more" data-l10n-id="collection-description"/>
+        <html:a id="dataCollectionPrivacyNotice"
+               class="learnMore"
+               data-l10n-id="collection-privacy-notice"/>
+      </description>
+    #ifdef MOZ_DATA_REPORTING
+      <description>
+        <hbox id="telemetry-container" class="info-box-container smaller-font-size" hidden="true">
+          <hbox class="info-icon-container">
+            <html:img class="info-icon"></html:img>
+          </hbox>
+          <hbox align="center" flex="1">
+            <html:span id="telemetryDisabledDescription" class="tail-with-learn-more"
+                       data-l10n-id="collection-health-report-telemetry-disabled"/>
+            <html:a is="moz-support-link"
+                    id="telemetryDataDeletionLearnMore"
+                    class="learnMore text-link"
+                    data-l10n-id="collection-health-report-telemetry-disabled-link"/>
+          </hbox>
+        </hbox>
+      </description>
+      <vbox data-subcategory="reports">
+        <description flex="1">
+          <checkbox id="submitHealthReportBox"
+                    data-l10n-id="collection-health-report"
+                    class="tail-with-learn-more"/>
+          <html:a id="FHRLearnMore"
+                 class="learnMore"
+                 data-l10n-id="collection-health-report-link"/>
+          <vbox class="indent">
+            <hbox align="center">
+              <checkbox id="addonRecommendationEnabled"
+                        class="tail-with-learn-more"
+                        data-l10n-id="addon-recommendations"/>
+              <html:a is="moz-support-link"
+                      id="addonRecommendationLearnMore"
+                      class="learnMore"
+                      data-l10n-id="addon-recommendations-link"
+                      support-page="personalized-addons"
+              />
+            </hbox>
+          </vbox>
+        </description>
+    #ifndef MOZ_TELEMETRY_REPORTING
+      <description id="TelemetryDisabledDesc"
+        class="indent tip-caption" control="telemetryGroup"
+        data-l10n-id="collection-health-report-disabled"/>
+    #endif
+
+    #ifdef MOZ_NORMANDY
         <hbox align="center">
-          <checkbox id="addonRecommendationEnabled"
+          <checkbox id="optOutStudiesEnabled"
                     class="tail-with-learn-more"
-                    data-l10n-id="addon-recommendations"/>
-          <html:a is="moz-support-link"
-                  id="addonRecommendationLearnMore"
+                    data-l10n-id="collection-studies"/>
+          <html:a id="viewShieldStudies"
+                  href="about:studies"
+                  useoriginprincipal="true"
                   class="learnMore"
-                  data-l10n-id="addon-recommendations-link"
-                  support-page="personalized-addons"
-          />
+                  data-l10n-id="collection-studies-link"/>
         </hbox>
-      </vbox>
-    </description>
-#ifndef MOZ_TELEMETRY_REPORTING
-  <description id="TelemetryDisabledDesc"
-    class="indent tip-caption" control="telemetryGroup"
-    data-l10n-id="collection-health-report-disabled"/>
-#endif
-
-#ifdef MOZ_NORMANDY
-    <hbox align="center">
-      <checkbox id="optOutStudiesEnabled"
-                class="tail-with-learn-more"
-                data-l10n-id="collection-studies"/>
-      <html:a id="viewShieldStudies"
-              href="about:studies"
-              useoriginprincipal="true"
-              class="learnMore"
-              data-l10n-id="collection-studies-link"/>
-    </hbox>
-#endif
+    #endif
 
-#ifdef MOZ_CRASHREPORTER
-    <hbox align="center" class="checkbox-row">
-      <html:input type="checkbox"
-                id="automaticallySubmitCrashesBox"
-                preference="browser.crashReports.unsubmittedCheck.autoSubmit2"/>
-      <label for="automaticallySubmitCrashesBox"
-                id="crashReporterLabel"
-                data-l10n-id="collection-backlogged-crash-reports-with-link">
-        <html:a data-l10n-name="crash-reports-link" id="crashReporterLearnMore" target="_blank"/>
-      </label>
-    </hbox>
-#endif
-  </vbox>
-#endif
-  <vbox id="privacySegmentationSection" data-subcategory="privacy-segmentation" hidden="true">
-    <label>
-      <html:h2 data-l10n-id="privacy-segmentation-section-header"/>
-    </label>
-    <label data-l10n-id="privacy-segmentation-section-description"/>
-    <radiogroup id="privacyDataFeatureRecommendationRadioGroup" preference="browser.dataFeatureRecommendations.enabled">
-        <radio id="privacyDataFeatureRecommendationEnabled"
-               data-l10n-id="privacy-segmentation-radio-off"
-               value="true"/>
-        <radio id="privacyDataFeatureRecommendationDisabled"
-               data-l10n-id="privacy-segmentation-radio-on"
-               value="false"/>
-    </radiogroup>
-  </vbox>
-</groupbox>-->
+    #ifdef MOZ_CRASHREPORTER
+        <hbox align="center" class="checkbox-row">
+          <html:input type="checkbox"
+                    id="automaticallySubmitCrashesBox"
+                    preference="browser.crashReports.unsubmittedCheck.autoSubmit2"/>
+          <label for="automaticallySubmitCrashesBox"
+                    id="crashReporterLabel"
+                    data-l10n-id="collection-backlogged-crash-reports-with-link">
+            <html:a data-l10n-name="crash-reports-link" id="crashReporterLearnMore" target="_blank"/>
+          </label>
+        </hbox>
+    #endif
+      </vbox>
+    #endif
+      <vbox id="privacySegmentationSection" data-subcategory="privacy-segmentation" hidden="true">
+        <label>
+          <html:h2 data-l10n-id="privacy-segmentation-section-header"/>
+        </label>
+        <label data-l10n-id="privacy-segmentation-section-description"/>
+        <radiogroup id="privacyDataFeatureRecommendationRadioGroup" preference="browser.dataFeatureRecommendations.enabled">
+            <radio id="privacyDataFeatureRecommendationEnabled"
+                   data-l10n-id="privacy-segmentation-radio-off"
+                   value="true"/>
+            <radio id="privacyDataFeatureRecommendationDisabled"
+                   data-l10n-id="privacy-segmentation-radio-on"
+                   value="false"/>
+        </radiogroup>
+      </vbox>
+    </groupbox>-->
 
-<hbox id="securityCategory"
-      class="subcategory"
-      hidden="true"
-      data-category="panePrivacy">
-  <html:h1 data-l10n-id="security-header"/>
-</hbox>
-
-<!-- addons, forgery (phishing) UI Security -->
-<groupbox id="browsingProtectionGroup" data-category="panePrivacy" hidden="true">
-  <label><html:h2 data-l10n-id="security-browsing-protection"/></label>
-  <hbox align = "center">
-    <checkbox id="enableSafeBrowsing"
-              data-l10n-id="security-enable-safe-browsing"
-              class="tail-with-learn-more"/>
-    <html:a is="moz-support-link"
-            id="enableSafeBrowsingLearnMore"
-            class="learnMore"
-            data-l10n-id="security-enable-safe-browsing-link"
-            support-page="phishing-malware"
-    />
-  </hbox>
-  <vbox class="indent">
-    <checkbox id="blockDownloads"
-              data-l10n-id="security-block-downloads"/>
-    <checkbox id="blockUncommonUnwanted"
-              data-l10n-id="security-block-uncommon-software"/>
-  </vbox>
-</groupbox>
+        <hbox id="securityCategory"
+              class="subcategory"
+              hidden="true"
+              data-category="panePrivacy">
+            <html:h1 data-l10n-id="security-header" />
+        </hbox>
 
-<!-- Certificates -->
-<groupbox id="certSelection" data-category="panePrivacy" hidden="true">
-  <label><html:h2 data-l10n-id="certs-header"/></label>
-  <hbox align="start">
-    <checkbox id="enableOCSP"
-              data-l10n-id="certs-enable-ocsp"
-              preference="security.OCSP.enabled"
-              flex="1" />
-    <vbox align="end">
-      <button id="viewCertificatesButton"
-              is="highlightable-button"
-              class="accessory-button"
-              data-l10n-id="certs-view"
-              preference="security.disable_button.openCertManager"
-              search-l10n-ids="
+        <!-- addons, forgery (phishing) UI Security -->
+        <groupbox id="browsingProtectionGroup" data-category="panePrivacy" hidden="true">
+            <label><html:h2 data-l10n-id="security-browsing-protection" /></label>
+            <hbox align="center">
+                <checkbox id="enableSafeBrowsing"
+                          data-l10n-id="security-enable-safe-browsing"
+                          class="tail-with-learn-more" />
+                <html:a is="moz-support-link"
+                        id="enableSafeBrowsingLearnMore"
+                        class="learnMore"
+                        data-l10n-id="security-enable-safe-browsing-link"
+                        support-page="phishing-malware" />
+            </hbox>
+            <vbox class="indent">
+                <checkbox id="blockDownloads"
+                          data-l10n-id="security-block-downloads" />
+                <checkbox id="blockUncommonUnwanted"
+                          data-l10n-id="security-block-uncommon-software" />
+            </vbox>
+        </groupbox>
+
+        <!-- Certificates -->
+        <groupbox id="certSelection" data-category="panePrivacy" hidden="true">
+            <label><html:h2 data-l10n-id="certs-header" /></label>
+            <hbox align="start">
+                <checkbox id="enableOCSP"
+                          data-l10n-id="certs-enable-ocsp"
+                          preference="security.OCSP.enabled"
+                          flex="1" />
+                <vbox align="end">
+                    <button id="viewCertificatesButton"
+                            is="highlightable-button"
+                            class="accessory-button"
+                            data-l10n-id="certs-view"
+                            preference="security.disable_button.openCertManager"
+                            search-l10n-ids="
                 certmgr-tab-mine.label,
                 certmgr-tab-people.label,
                 certmgr-tab-servers.label,
@@ -1131,13 +1126,13 @@
                 certmgr-view.label,
                 certmgr-export.label,
                 certmgr-delete.label
-              "/>
-      <button id="viewSecurityDevicesButton"
-              is="highlightable-button"
-              class="accessory-button"
-              data-l10n-id="certs-devices"
-              preference="security.disable_button.openDeviceManager"
-              search-l10n-ids="
+              " />
+                    <button id="viewSecurityDevicesButton"
+                            is="highlightable-button"
+                            class="accessory-button"
+                            data-l10n-id="certs-devices"
+                            preference="security.disable_button.openDeviceManager"
+                            search-l10n-ids="
                 devmgr-window.title,
                 devmgr-devlist.label,
                 devmgr-header-details.label,
@@ -1147,46 +1142,45 @@
                 devmgr-button-changepw.label,
                 devmgr-button-load.label,
                 devmgr-button-unload.label
-              "/>
-    </vbox>
-  </hbox>
-</groupbox>
+              " />
+                </vbox>
+            </hbox>
+        </groupbox>
 
-<!-- HTTPS-ONLY Mode -->
-<groupbox id="httpsOnlyBox" data-category="panePrivacy" hidden="true">
-  <label><html:h2 data-l10n-id="httpsonly-header"/></label>
-  <vbox data-subcategory="httpsonly" flex="1">
-    <label id="httpsOnlyDescription" data-l10n-id="httpsonly-description"/>
-    <html:a is="moz-support-link"
-            id="httpsOnlyLearnMore"
-            class="learnMore"
-            data-l10n-id="httpsonly-learn-more"
-            support-page="https-only-prefs"
-    />
-  </vbox>
-  <hbox>
-    <radiogroup flex="1" id="httpsOnlyRadioGroup">
-      <radio id="httpsOnlyRadioEnabled"
-              data-l10n-id="httpsonly-radio-enabled"
-              value="enabled"/>
-      <radio id="httpsOnlyRadioEnabledPBM"
-              data-l10n-id="httpsonly-radio-enabled-pbm"
-              value="privateOnly"/>
-      <radio id="httpsOnlyRadioDisabled"
-              data-l10n-id="httpsonly-radio-disabled"
-              value="disabled"/>
-    </radiogroup>
-    <vbox>
-      <button id="httpsOnlyExceptionButton" is="highlightable-button" class="accessory-button" disabled="true"
-        data-l10n-id="sitedata-cookies-exceptions" search-l10n-ids="
+        <!-- HTTPS-ONLY Mode -->
+        <groupbox id="httpsOnlyBox" data-category="panePrivacy" hidden="true">
+            <label><html:h2 data-l10n-id="httpsonly-header" /></label>
+            <vbox data-subcategory="httpsonly" flex="1">
+                <label id="httpsOnlyDescription" data-l10n-id="httpsonly-description" />
+                <html:a is="moz-support-link"
+                        id="httpsOnlyLearnMore"
+                        class="learnMore"
+                        data-l10n-id="httpsonly-learn-more"
+                        support-page="https-only-prefs" />
+            </vbox>
+            <hbox>
+                <radiogroup flex="1" id="httpsOnlyRadioGroup">
+                    <radio id="httpsOnlyRadioEnabled"
+                           data-l10n-id="httpsonly-radio-enabled"
+                           value="enabled" />
+                    <radio id="httpsOnlyRadioEnabledPBM"
+                           data-l10n-id="httpsonly-radio-enabled-pbm"
+                           value="privateOnly" />
+                    <radio id="httpsOnlyRadioDisabled"
+                           data-l10n-id="httpsonly-radio-disabled"
+                           value="disabled" />
+                </radiogroup>
+                <vbox>
+                    <button id="httpsOnlyExceptionButton" is="highlightable-button" class="accessory-button" disabled="true"
+                            data-l10n-id="sitedata-cookies-exceptions" search-l10n-ids="
                       permissions-address,
                       permissions-allow.label,
                       permissions-remove.label,
                       permissions-remove-all.label,
                       permissions-exceptions-https-only-desc,
                     " />
-    </vbox>
-  </hbox>
-</groupbox>
+                </vbox>
+            </hbox>
+        </groupbox>
 
 </html:template>
Index: mozilla-release/browser/components/preferences/privacy.js
===================================================================
--- mozilla-release.orig/browser/components/preferences/privacy.js
+++ mozilla-release/browser/components/preferences/privacy.js
@@ -563,11 +563,12 @@ var gPrivacyPane = {
       }
       return false;
     });
+/* MagIns - Hide umbrella Private Browsing preference, always show granular options
     setEventListener(
       "privateBrowsingAutoStart",
       "command",
       gPrivacyPane.updateAutostart
-    );
+    );*/
     setEventListener(
       "cookieExceptions",
       "command",
@@ -693,7 +694,8 @@ var gPrivacyPane = {
       this.ensurePrivacyMicroControlUncheckedWhenDisabled(el);
     setSyncFromPrefListener("rememberHistory", microControlHandler);
     setSyncFromPrefListener("rememberForms", microControlHandler);
-    setSyncFromPrefListener("alwaysClear", microControlHandler);
+/* MagIns - Remove the extraneous privacy.sanitize.sanitizeOnShutdown checkbox
+      setSyncFromPrefListener("alwaysClear", microControlHandler);*/
 
     setSyncFromPrefListener("popupPolicy", () =>
       this.updateButtons("popupPolicyButton", "dom.disable_open_during_load")
@@ -722,11 +724,12 @@ var gPrivacyPane = {
     this._initAddressBar();
 
     this.initSiteDataControls();
+/* MagIns - Remove instance of clearSiteDataButton
     setEventListener(
       "clearSiteDataButton",
       "command",
       gPrivacyPane.clearSiteData
-    );
+    );*/
     setEventListener(
       "siteDataSettings",
       "command",
@@ -1350,7 +1353,8 @@ var gPrivacyPane = {
   dependentControls: [
     "rememberHistory",
     "rememberForms",
-    "alwaysClear",
+/* MagIns - Remove the extraneous privacy.sanitize.sanitizeOnShutdown checkbox
+    "alwaysClear",*/
     "clearDataSettings",
   ],
 
@@ -1908,10 +1912,10 @@ var gPrivacyPane = {
   },
 
   toggleSiteData(shouldShow) {
-    let clearButton = document.getElementById("clearSiteDataButton");
+/*    let clearButton = document.getElementById("clearSiteDataButton");
     let settingsButton = document.getElementById("siteDataSettings");
     clearButton.disabled = !shouldShow;
-    settingsButton.disabled = !shouldShow;
+    settingsButton.disabled = !shouldShow;*/
   },
 
   showSiteDataLoading() {
Index: mozilla-release/browser/locales/en-US/browser/preferences/preferences.ftl
===================================================================
--- mozilla-release.orig/browser/locales/en-US/browser/preferences/preferences.ftl
+++ mozilla-release/browser/locales/en-US/browser/preferences/preferences.ftl
@@ -1071,11 +1071,11 @@ history-remember-label = { -brand-short-
     .accesskey = w
 
 history-remember-option-all =
-    .label = Remember history
+    .label = Remember browser data
 history-remember-option-never =
-    .label = Never remember history
+    .label = Never remember browser data
 history-remember-option-custom =
-    .label = Use custom settings for history
+    .label = Use custom settings for browser data
 
 history-remember-description = { -brand-short-name } will remember your browsing, download, form, and search history.
 history-dontremember-description = { -brand-short-name } will use the same settings as private browsing, and will not remember any history as you browse the Web.
@@ -1093,7 +1093,7 @@ history-remember-search-option =
     .accesskey = f
 
 history-clear-on-close-option =
-    .label = Clear history when { -brand-short-name } closes
+    .label = Clear browser data when { -brand-short-name } closes
     .accesskey = r
 
 history-clear-on-close-settings =
@@ -1106,7 +1106,7 @@ history-clear-button =
 
 ## Privacy Section - Site Data
 
-sitedata-header = Cookies and Site Data
+sitedata-header = History, Cookies and Site Data
 
 sitedata-total-size-calculating = Calculating site data and cache size…
 
