Index: firefox-96.0.1/browser/config/version.txt
===================================================================
--- firefox-96.0.1.orig/browser/config/version.txt
+++ firefox-96.0.1/browser/config/version.txt
@@ -1 +1 @@
-96.0.1
+96.0.1.1
Index: firefox-96.0.1/browser/config/version_display.txt
===================================================================
--- firefox-96.0.1.orig/browser/config/version_display.txt
+++ firefox-96.0.1/browser/config/version_display.txt
@@ -1 +1 @@
-96.0.1
+96.0.1.1
Index: firefox-96.0.1/browser/config/linux_mozconfig
===================================================================
--- /dev/null
+++ firefox-96.0.1/browser/config/linux_mozconfig
@@ -0,0 +1,12 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This file specifies the build flags for Firefox.  You can use it by adding:
+#  . $topsrcdir/browser/config/mozconfig
+# to the top of your mozconfig file.
+
+ac_add_options --enable-application=browser
+ac_add_options --with-google-safebrowsing-api-keyfile="/home/mi-linux/miff-dev/GoogleSB.txt"
+
+export MOZILLA_OFFICIAL=1
Index: firefox-96.0.1/browser/config/win_mozconfig
===================================================================
--- /dev/null
+++ firefox-96.0.1/browser/config/win_mozconfig
@@ -0,0 +1,12 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This file specifies the build flags for Firefox.  You can use it by adding:
+#  . $topsrcdir/browser/config/mozconfig
+# to the top of your mozconfig file.
+
+ac_add_options --enable-application=browser
+ac_add_options --with-google-safebrowsing-api-keyfile="C:/mozilla-source/GoogleSB.txt"
+
+export MOZILLA_OFFICIAL=1
